<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>01_Running-RNAseq_pipeline</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="01_running_nfcoreRNAseq_pipeline_files/libs/clipboard/clipboard.min.js"></script>
<script src="01_running_nfcoreRNAseq_pipeline_files/libs/quarto-html/quarto.js"></script>
<script src="01_running_nfcoreRNAseq_pipeline_files/libs/quarto-html/popper.min.js"></script>
<script src="01_running_nfcoreRNAseq_pipeline_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="01_running_nfcoreRNAseq_pipeline_files/libs/quarto-html/anchor.min.js"></script>
<link href="01_running_nfcoreRNAseq_pipeline_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="01_running_nfcoreRNAseq_pipeline_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="01_running_nfcoreRNAseq_pipeline_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="01_running_nfcoreRNAseq_pipeline_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="01_running_nfcoreRNAseq_pipeline_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#github-repo" id="toc-github-repo" class="nav-link active" data-scroll-target="#github-repo">Github repo</a></li>
  <li><a href="#using-the-nf-core-pipeline-to-perform-preprocessing-of-the-rna-seq-fastqs" id="toc-using-the-nf-core-pipeline-to-perform-preprocessing-of-the-rna-seq-fastqs" class="nav-link" data-scroll-target="#using-the-nf-core-pipeline-to-perform-preprocessing-of-the-rna-seq-fastqs">Using the nf-core pipeline to perform preprocessing of the RNA-seq FASTQs</a>
  <ul class="collapse">
  <li><a href="#creation-and-setting-up-of-conda-environment" id="toc-creation-and-setting-up-of-conda-environment" class="nav-link" data-scroll-target="#creation-and-setting-up-of-conda-environment">Creation and setting up of conda environment</a></li>
  <li><a href="#test-nfcore-rnaseq-run" id="toc-test-nfcore-rnaseq-run" class="nav-link" data-scroll-target="#test-nfcore-rnaseq-run">Test nfcore rnaseq run</a></li>
  <li><a href="#samplesheet" id="toc-samplesheet" class="nav-link" data-scroll-target="#samplesheet">Samplesheet</a></li>
  <li><a href="#nf-core-run-with-the-liver-and-heart-samples" id="toc-nf-core-run-with-the-liver-and-heart-samples" class="nav-link" data-scroll-target="#nf-core-run-with-the-liver-and-heart-samples">nf-core run with the liver and heart samples</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">01_Running-RNAseq_pipeline</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Aim:</strong> To execute the bioinformatics pipeline that can be used to analyse RNA sequencing data obtained from organisms with a reference genome and annotation (mouse in case). It takes a samplesheet and FASTQ files as input, performs quality control (QC), trimming, alignment, and quantification to produce a gene expression matrix and extensive QC report.</p>
<section id="github-repo" class="level2">
<h2 class="anchored" data-anchor-id="github-repo">Github repo</h2>
<p>I first created a github repo in my personal account and staged the necessary files to be tracked and pushed to Github repo. The github link for the project can be found here: <a href="https://github.com/adityamiyer-personal/Bulk-RNA-seq-analysis" class="uri">https://github.com/adityamiyer-personal/Bulk-RNA-seq-analysis</a>.</p>
</section>
<section id="using-the-nf-core-pipeline-to-perform-preprocessing-of-the-rna-seq-fastqs" class="level2">
<h2 class="anchored" data-anchor-id="using-the-nf-core-pipeline-to-perform-preprocessing-of-the-rna-seq-fastqs">Using the nf-core pipeline to perform preprocessing of the RNA-seq FASTQs</h2>
<p>nf-core RNA-seq allows us to perform the basic processing of RNA-seq reads in a reproducing fashion. I first set up a conda environment to run nf-core RNAseq pipeline. More information about this open-source pipeline can be found at <a href="https://nf-co.re/rnaseq/3.14.0/" class="uri">https://nf-co.re/rnaseq/3.14.0/</a>.</p>
<section id="creation-and-setting-up-of-conda-environment" class="level3">
<h3 class="anchored" data-anchor-id="creation-and-setting-up-of-conda-environment">Creation and setting up of conda environment</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Creation of conda environment</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>conda create <span class="sc">-</span>n bulkrnaseq</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Activation of conda environment</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>conda activate bulkrnaseq</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#install nextflow</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>c bioconda nextflow</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>c conda<span class="sc">-</span>forge openjdk<span class="ot">=</span><span class="dv">11</span>.<span class="dv">0</span>.<span class="fl">9.1</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>c conda<span class="sc">-</span>forge singularity<span class="sc">==</span><span class="dv">3</span>.<span class="fl">8.6</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Setting the channel priortiy list required for Nextflow</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>add channels defaults</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>add channels bioconda</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>add channels conda<span class="sc">-</span>forge</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>set channel_priority disabled</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>n base conda<span class="sc">-</span>libmamba<span class="sc">-</span>solver</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="test-nfcore-rnaseq-run" class="level3">
<h3 class="anchored" data-anchor-id="test-nfcore-rnaseq-run">Test nfcore rnaseq run</h3>
<p>After setting up the conda environment, I performed a test run to check if the config and local machine support the test run or not. I have used <strong>3.14.0</strong> version of the pipeline therefore went ahead with that.</p>
<p>The script used to perform the test run in the conda environment is given below:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#in the conda environment, run</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile test,conda \</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>test_output<span class="sc">/</span> \</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>test_workdir<span class="sc">/</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="samplesheet" class="level3">
<h3 class="anchored" data-anchor-id="samplesheet">Samplesheet</h3>
<p>As given in the guidelines, the samplesheet follows the guidelines as given here <a href="https://nf-co.re/rnaseq/3.14.0/docs/usage/" class="uri">https://nf-co.re/rnaseq/3.14.0/docs/usage/</a> and <a href="https://nf-co.re/rnaseq/3.14.0/parameters" class="uri">https://nf-co.re/rnaseq/3.14.0/parameters</a> . I set the strandedness value to <strong>auto</strong>, the pipeline will sub-sample the input FastQ files to 1 million reads, use Salmon Quant to automatically infer the strandedness, and then propagate this information through the rest of the pipeline.&nbsp;</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"~/Documents/BiostateAI/samplesheet/samplesheet.csv"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>) <span class="sc">%&gt;%</span> knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 8%">
<col style="width: 42%">
<col style="width: 42%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">sample</th>
<th style="text-align: left;">fastq_1</th>
<th style="text-align: left;">fastq_2</th>
<th style="text-align: left;">strandedness</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Liver_ZT0_REP1</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT0_1_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT0_1_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="even">
<td style="text-align: left;">Liver_ZT0_REP2</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT0_2_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT0_2_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Liver_ZT12_REP1</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT12_1_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT12_1_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="even">
<td style="text-align: left;">Liver_ZT12_REP2</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT12_2_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Liver_ZT12_2_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Heart_ZT0_REP1</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT0_1_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT0_1_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="even">
<td style="text-align: left;">Heart_ZT0_REP2</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT0_2_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT0_2_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Heart_ZT12_REP1</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT12_1_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT12_1_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
<tr class="even">
<td style="text-align: left;">Heart_ZT12_REP2</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT12_2_R1.fastq.gz</td>
<td style="text-align: left;">/Users/adityamiyer/Documents/BiostateAI/data/fastqs/Heart_ZT12_2_R2.fastq.gz</td>
<td style="text-align: left;">forward</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="nf-core-run-with-the-liver-and-heart-samples" class="level3">
<h3 class="anchored" data-anchor-id="nf-core-run-with-the-liver-and-heart-samples">nf-core run with the liver and heart samples</h3>
<p>I used the <strong>star_salmon workflow</strong> of nf-core rnaseq pipeline (Version 3.16.1) that performs alignment using <strong>STAR</strong> and read quantification using <strong>Salmon</strong></p>
<p>The general command that could be run on a HPC or server would look like below:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile conda \</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>output<span class="sc">/</span> \</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>workdir<span class="sc">/</span> \</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>input <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>samplesheet<span class="sc">/</span>samplesheet.csv \</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gtf <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>gencode.vM35.basic.annotation.gtf \</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>fasta <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>GRCm39.primary_assembly.genome.fa.gz \</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>aligner star_salmon <span class="sc">--</span>gencode <span class="sc">-</span>resume</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Status:</strong> The above run keeps failing due to memory issues in my system. I need access to a HPC to execute this pipeline.</p>
<p>Due to memory limitations on my personal laptop, I resorted to the following command skipping all the various QC steps of the pipeline. Despite adding in memory and cpu limits, the star_salmon alignment strategy does not get executed in my PC.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#in the conda environment, run</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile conda \</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>output<span class="sc">/</span> \</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>workdir<span class="sc">/</span> \</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>input <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>samplesheet<span class="sc">/</span>samplesheet.csv \</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gtf <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>gencode.vM35.basic.annotation.gtf \</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>fasta <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>GRCm39.primary_assembly.genome.fa.gz \</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>aligner star_salmon <span class="sc">--</span>gencode <span class="sc">-</span>resume <span class="sc">--</span>max_cpus <span class="dv">8</span> <span class="sc">--</span>max_memory <span class="fl">8.</span>GB <span class="sc">--</span>trimmer <span class="st">'fastp'</span> \</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>skip_markduplicates <span class="sc">--</span>skip_rseqc <span class="sc">--</span>skip_preseq <span class="sc">--</span>skip_dupradar <span class="sc">--</span>skip_qualimap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Status:</strong> The above run keeps failing after initial QC due to memory issues. I need access to a HPC or server to execute this <strong>star_salmon</strong> workflow.</p>
<p>To circumvent this issue, I am trying with using pseudoaligner <strong>Salmon</strong> instead of memory-intensive STAR and also added in memory,cpu limits. I also set the strandedness of the libraries to “forward” so that Salmon index (memory-intensive again) is not run during the pipeline run.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile conda \</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>output<span class="sc">/</span> \</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>workdir<span class="sc">/</span> \</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>input <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>samplesheet<span class="sc">/</span>samplesheet.csv \</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gtf <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>gencode.vM35.basic.annotation.gtf \</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>fasta <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>GRCm39.primary_assembly.genome.fa.gz \</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gencode <span class="sc">-</span>resume <span class="sc">--</span>max_cpus <span class="dv">8</span> <span class="sc">--</span>max_memory <span class="fl">8.</span>GB <span class="sc">--</span>trimmer <span class="st">'fastp'</span> \</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>skip_markduplicates <span class="sc">--</span>skip_rseqc <span class="sc">--</span>skip_preseq <span class="sc">--</span>skip_dupradar <span class="sc">--</span>skip_qualimap <span class="sc">--</span>skip_alignment \</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>pseudo_aligner salmon <span class="sc">--</span>skip_gtf_filter <span class="sc">--</span>skip_gtf_transcript_filter</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Status:</strong> The run keeps failing due to memory issues. I need access to a server or cluster with resources to run the preprocessing steps.</p>
<!-- -->

</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb7" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "01_Running-RNAseq_pipeline"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">    eval: false</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>**Aim:** To execute the bioinformatics pipeline that can be used to analyse RNA sequencing data obtained from organisms with a reference genome and annotation (mouse in case). It takes a samplesheet and FASTQ files as input, performs quality control (QC), trimming, alignment, and quantification to produce a gene expression matrix and extensive QC report.</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="fu">## Github repo</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>I first created a github repo in my personal account and staged the necessary files to be tracked and pushed to Github repo. The github link for the project can be found here: <span class="ot">&lt;https://github.com/adityamiyer-personal/Bulk-RNA-seq-analysis&gt;</span>.</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="fu">## Using the nf-core pipeline to perform preprocessing of the RNA-seq FASTQs</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>nf-core RNA-seq allows us to perform the basic processing of RNA-seq reads in a reproducing fashion. I first set up a conda environment to run nf-core RNAseq pipeline. More information about this open-source pipeline can be found at <span class="ot">&lt;https://nf-co.re/rnaseq/3.14.0/&gt;</span>.</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="fu">### Creation and setting up of conda environment</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Creation of conda environment}</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Creation of conda environment</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>conda create <span class="sc">-</span>n bulkrnaseq</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="co">#Activation of conda environment</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>conda activate bulkrnaseq</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co">#install nextflow</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>c bioconda nextflow</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>c conda<span class="sc">-</span>forge openjdk<span class="ot">=</span><span class="dv">11</span>.<span class="dv">0</span>.<span class="fl">9.1</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>c conda<span class="sc">-</span>forge singularity<span class="sc">==</span><span class="dv">3</span>.<span class="fl">8.6</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Setting the channel priortiy list required for Nextflow</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>add channels defaults</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>add channels bioconda</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>add channels conda<span class="sc">-</span>forge</span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>conda config <span class="sc">--</span>set channel_priority disabled</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>conda install <span class="sc">-</span>n base conda<span class="sc">-</span>libmamba<span class="sc">-</span>solver</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a><span class="fu">### Test nfcore rnaseq run</span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>After setting up the conda environment, I performed a test run to check if the config and local machine support the test run or not. I have used **3.14.0** version of the pipeline therefore went ahead with that.</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>The script used to perform the test run in the conda environment is given below:</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Test nfcore run}</span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="co">#in the conda environment, run</span></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile test,conda \</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>test_output<span class="sc">/</span> \</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>test_workdir<span class="sc">/</span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a><span class="fu">### Samplesheet</span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>As given in the guidelines, the samplesheet follows the guidelines as given here <span class="ot">&lt;https://nf-co.re/rnaseq/3.14.0/docs/usage/&gt;</span> and <span class="ot">&lt;https://nf-co.re/rnaseq/3.14.0/parameters&gt;</span> . I set the strandedness value to **auto**, the pipeline will sub-sample the input FastQ files to 1 million reads, use Salmon Quant to automatically infer the strandedness, and then propagate this information through the rest of the pipeline.&nbsp;</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Creation of samplesheet}</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: true</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"~/Documents/BiostateAI/samplesheet/samplesheet.csv"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>) <span class="sc">%&gt;%</span> knitr<span class="sc">::</span><span class="fu">kable</span>()</span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a><span class="fu">### nf-core run with the liver and heart samples</span></span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>I used the **star_salmon workflow** of nf-core rnaseq pipeline (Version 3.16.1) that performs alignment using **STAR** and read quantification using **Salmon**</span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>The general command that could be run on a HPC or server would look like below:</span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="in">```{r A general run on a server}</span></span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile conda \</span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>output<span class="sc">/</span> \</span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>workdir<span class="sc">/</span> \</span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>input <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>samplesheet<span class="sc">/</span>samplesheet.csv \</span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gtf <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>gencode.vM35.basic.annotation.gtf \</span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>fasta <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>GRCm39.primary_assembly.genome.fa.gz \</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>aligner star_salmon <span class="sc">--</span>gencode <span class="sc">-</span>resume</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>**Status:** The above run keeps failing due to memory issues in my system. I need access to a HPC to execute this pipeline.</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>Due to memory limitations on my personal laptop, I resorted to the following command skipping all the various QC steps of the pipeline. Despite adding in memory and cpu limits, the star_salmon alignment strategy does not get executed in my PC.</span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{r nfcore run with liver and heart samples}</span></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a><span class="co">#in the conda environment, run</span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile conda \</span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>output<span class="sc">/</span> \</span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>workdir<span class="sc">/</span> \</span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>input <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>samplesheet<span class="sc">/</span>samplesheet.csv \</span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gtf <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>gencode.vM35.basic.annotation.gtf \</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>fasta <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>GRCm39.primary_assembly.genome.fa.gz \</span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>aligner star_salmon <span class="sc">--</span>gencode <span class="sc">-</span>resume <span class="sc">--</span>max_cpus <span class="dv">8</span> <span class="sc">--</span>max_memory <span class="fl">8.</span>GB <span class="sc">--</span>trimmer <span class="st">'fastp'</span> \</span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>skip_markduplicates <span class="sc">--</span>skip_rseqc <span class="sc">--</span>skip_preseq <span class="sc">--</span>skip_dupradar <span class="sc">--</span>skip_qualimap</span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a>**Status:** The above run keeps failing after initial QC due to memory issues. I need access to a HPC or server to execute this **star_salmon** workflow. </span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a>To circumvent this issue, I am trying with using pseudoaligner **Salmon** instead of memory-intensive STAR and also added in memory,cpu limits. I also set the strandedness of the libraries to "forward" so that Salmon index (memory-intensive again) is not run during the pipeline run. </span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a>nextflow run nf<span class="sc">-</span>core<span class="sc">/</span>rnaseq <span class="sc">-</span>r <span class="dv">3</span>.<span class="fl">14.0</span> \</span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>profile conda \</span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>outdir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>output<span class="sc">/</span> \</span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span>work<span class="sc">-</span>dir <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>results<span class="sc">/</span>workdir<span class="sc">/</span> \</span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>input <span class="sc">/</span>Users<span class="sc">/</span>adityamiyer<span class="sc">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>samplesheet<span class="sc">/</span>samplesheet.csv \</span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gtf <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>gencode.vM35.basic.annotation.gtf \</span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>fasta <span class="sc">~</span><span class="er">/</span>Documents<span class="sc">/</span>BiostateAI<span class="sc">/</span>data<span class="sc">/</span>fastqs<span class="sc">/</span>GRCm39.primary_assembly.genome.fa.gz \</span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>gencode <span class="sc">-</span>resume <span class="sc">--</span>max_cpus <span class="dv">8</span> <span class="sc">--</span>max_memory <span class="fl">8.</span>GB <span class="sc">--</span>trimmer <span class="st">'fastp'</span> \</span>
<span id="cb7-123"><a href="#cb7-123" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>skip_markduplicates <span class="sc">--</span>skip_rseqc <span class="sc">--</span>skip_preseq <span class="sc">--</span>skip_dupradar <span class="sc">--</span>skip_qualimap <span class="sc">--</span>skip_alignment \</span>
<span id="cb7-124"><a href="#cb7-124" aria-hidden="true" tabindex="-1"></a>  <span class="sc">--</span>pseudo_aligner salmon <span class="sc">--</span>skip_gtf_filter <span class="sc">--</span>skip_gtf_transcript_filter</span>
<span id="cb7-125"><a href="#cb7-125" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-126"><a href="#cb7-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-127"><a href="#cb7-127" aria-hidden="true" tabindex="-1"></a>**Status:** The run keeps failing due to memory issues. I need access to a server or cluster with resources to run the preprocessing steps.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>