---
title: "02_DESeq2"
format:
  html:
    code-fold: true
    code-tools: true
    toc: true
    eval: false
editor: visual
editor_options: 
  chunk_output_type: console
---

**Aim:** To perform differential gene expression analysis for different contrasts

## Loading the required libraries and importing counts

### Loading the required libraries

```{r Loading libraries}
library(tidyverse) #for data wrangling
library(DESeq2) #for DEG analysis
library(genekitr) #for pathway analysis
library(readr) #for data import
```

Refer this section <https://nf-co.re/rnaseq/3.14.0/docs/output/#pseudoalignment-and-quantification> to understand which output files and methods are suitable for further DEG analysis. I use the length-scaled counts generated by Salmon in order not to influence gene lengths in the overall quantification results and therefore DEGs on the basis of gene length. 

```{r Importing length-scaled counts from salmon output folder}

```

### Running Code

```{r}

```


