<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>SampleAnalysis_Mansingh2024_matrix</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/clipboard/clipboard.min.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/quarto.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/popper.min.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/anchor.min.js"></script>
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#loading-the-required-libraries-and-importing-required-files" id="toc-loading-the-required-libraries-and-importing-required-files" class="nav-link active" data-scroll-target="#loading-the-required-libraries-and-importing-required-files">Loading the required libraries and importing required files</a>
  <ul>
  <li><a href="#loading-the-required-libraries" id="toc-loading-the-required-libraries" class="nav-link" data-scroll-target="#loading-the-required-libraries">Loading the required libraries</a></li>
  <li><a href="#importing-the-counts-matrix" id="toc-importing-the-counts-matrix" class="nav-link" data-scroll-target="#importing-the-counts-matrix">Importing the counts matrix</a></li>
  <li><a href="#loading-the-metadata-file" id="toc-loading-the-metadata-file" class="nav-link" data-scroll-target="#loading-the-metadata-file">Loading the metadata file</a></li>
  </ul></li>
  <li><a href="#creation-and-exploration-of-dds-object" id="toc-creation-and-exploration-of-dds-object" class="nav-link" data-scroll-target="#creation-and-exploration-of-dds-object">Creation and exploration of dds object</a>
  <ul>
  <li><a href="#pca-plot" id="toc-pca-plot" class="nav-link" data-scroll-target="#pca-plot">PCA plot</a></li>
  <li><a href="#sample-correlation-heatmap-using-vst-counts" id="toc-sample-correlation-heatmap-using-vst-counts" class="nav-link" data-scroll-target="#sample-correlation-heatmap-using-vst-counts">Sample correlation heatmap using vst counts</a></li>
  </ul></li>
  <li><a href="#differential-gene-expression-analysis" id="toc-differential-gene-expression-analysis" class="nav-link" data-scroll-target="#differential-gene-expression-analysis">Differential gene expression analysis</a>
  <ul>
  <li><a href="#main-effects-due-to-genotype" id="toc-main-effects-due-to-genotype" class="nav-link" data-scroll-target="#main-effects-due-to-genotype">Main effects due to genotype</a>
  <ul class="collapse">
  <li><a href="#comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype" id="toc-comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype" class="nav-link" data-scroll-target="#comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype">Comparison of Wald versus LRT test DEGs Main effects due to genotype</a></li>
  <li><a href="#visualization-of-genotype-specific-degs-using-deseq2" id="toc-visualization-of-genotype-specific-degs-using-deseq2" class="nav-link" data-scroll-target="#visualization-of-genotype-specific-degs-using-deseq2">Visualization of genotype-specific DEGs using DESeq2</a></li>
  <li><a href="#expression-pattern-of-top-genotype-specific-degs" id="toc-expression-pattern-of-top-genotype-specific-degs" class="nav-link" data-scroll-target="#expression-pattern-of-top-genotype-specific-degs">Expression pattern of top genotype-specific DEGs</a></li>
  <li><a href="#pathway-analysis-using-the-genotype-specific-degs-main-effects-using-ora" id="toc-pathway-analysis-using-the-genotype-specific-degs-main-effects-using-ora" class="nav-link" data-scroll-target="#pathway-analysis-using-the-genotype-specific-degs-main-effects-using-ora">Pathway analysis using the genotype-specific DEGs (Main effects) using ORA</a></li>
  <li><a href="#pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea" id="toc-pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea" class="nav-link" data-scroll-target="#pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea">Pathway analysis using the genotype-specific DEGs (Main effects) using GSEA</a></li>
  </ul></li>
  <li><a href="#main-effects-due-to-timepoint" id="toc-main-effects-due-to-timepoint" class="nav-link" data-scroll-target="#main-effects-due-to-timepoint">Main effects due to timepoint</a>
  <ul class="collapse">
  <li><a href="#comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point" id="toc-comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point" class="nav-link" data-scroll-target="#comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point">Comparison of Wald versus LRT test DEGs [Main effects due to time_point]</a></li>
  <li><a href="#visualization-of-timepoint-specific-degs-using-deseq2" id="toc-visualization-of-timepoint-specific-degs-using-deseq2" class="nav-link" data-scroll-target="#visualization-of-timepoint-specific-degs-using-deseq2">Visualization of timepoint-specific DEGs using DESeq2</a></li>
  <li><a href="#expression-pattern-of-top-time-specific-degs" id="toc-expression-pattern-of-top-time-specific-degs" class="nav-link" data-scroll-target="#expression-pattern-of-top-time-specific-degs">Expression pattern of top time-specific DEGs</a></li>
  <li><a href="#pathway-analysis-using-the-timepoint-specific-degs-main-effects-using-ora" id="toc-pathway-analysis-using-the-timepoint-specific-degs-main-effects-using-ora" class="nav-link" data-scroll-target="#pathway-analysis-using-the-timepoint-specific-degs-main-effects-using-ora">Pathway analysis using the timepoint-specific DEGs (Main effects) using ORA</a></li>
  <li><a href="#pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea-1" id="toc-pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea-1" class="nav-link" data-scroll-target="#pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea-1">Pathway analysis using the genotype-specific DEGs (Main effects) using GSEA</a></li>
  </ul></li>
  <li><a href="#interaction-effects" id="toc-interaction-effects" class="nav-link" data-scroll-target="#interaction-effects">Interaction effects</a>
  <ul class="collapse">
  <li><a href="#comparison-of-wald-versus-lrt-test---interaction-effects" id="toc-comparison-of-wald-versus-lrt-test---interaction-effects" class="nav-link" data-scroll-target="#comparison-of-wald-versus-lrt-test---interaction-effects">Comparison of Wald versus LRT test - interaction effects</a></li>
  <li><a href="#visualization-of-interaction-specific-degs-using-deseq2" id="toc-visualization-of-interaction-specific-degs-using-deseq2" class="nav-link" data-scroll-target="#visualization-of-interaction-specific-degs-using-deseq2">Visualization of interaction-specific DEGs using DESeq2</a></li>
  <li><a href="#expression-pattern-of-top-interaction-specific-degs" id="toc-expression-pattern-of-top-interaction-specific-degs" class="nav-link" data-scroll-target="#expression-pattern-of-top-interaction-specific-degs">Expression pattern of top interaction-specific DEGs</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#paired-contrast-analysis---difference-between-wt-and-ko-at-each-timepoint" id="toc-paired-contrast-analysis---difference-between-wt-and-ko-at-each-timepoint" class="nav-link" data-scroll-target="#paired-contrast-analysis---difference-between-wt-and-ko-at-each-timepoint">Paired contrast analysis - Difference between WT and KO at each timepoint</a>
  <ul>
  <li><a href="#pathway-analysis-for-genotype-enriched-degs-for-timepoint-t0" id="toc-pathway-analysis-for-genotype-enriched-degs-for-timepoint-t0" class="nav-link" data-scroll-target="#pathway-analysis-for-genotype-enriched-degs-for-timepoint-t0">Pathway analysis for genotype-enriched DEGs for timepoint T0</a></li>
  <li><a href="#pathway-analysis-for-genotype-enriched-degs-for-timepoint-t12" id="toc-pathway-analysis-for-genotype-enriched-degs-for-timepoint-t12" class="nav-link" data-scroll-target="#pathway-analysis-for-genotype-enriched-degs-for-timepoint-t12">Pathway analysis for genotype-enriched DEGs for timepoint T12</a></li>
  </ul></li>
  <li><a href="#heirarchical-clustering-of-all-the-degs" id="toc-heirarchical-clustering-of-all-the-degs" class="nav-link" data-scroll-target="#heirarchical-clustering-of-all-the-degs">Heirarchical clustering of all the DEGs</a></li>
  <li><a href="#plot-the-expression-pattern-of-the-degs-obtained-from-all-the-above-comparisons" id="toc-plot-the-expression-pattern-of-the-degs-obtained-from-all-the-above-comparisons" class="nav-link" data-scroll-target="#plot-the-expression-pattern-of-the-degs-obtained-from-all-the-above-comparisons">Plot the expression pattern of the DEGs obtained from all the above comparisons</a>
  <ul>
  <li><a href="#pathway-analysis-of-various-clusters-obtained-from-the-above-analysis" id="toc-pathway-analysis-of-various-clusters-obtained-from-the-above-analysis" class="nav-link" data-scroll-target="#pathway-analysis-of-various-clusters-obtained-from-the-above-analysis">Pathway analysis of various clusters obtained from the above analysis</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">SampleAnalysis_Mansingh2024_matrix</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Aim:</strong> To perform differential gene expression analysis for different contrasts</p>
<section id="loading-the-required-libraries-and-importing-required-files" class="level2">
<h2 class="anchored" data-anchor-id="loading-the-required-libraries-and-importing-required-files">Loading the required libraries and importing required files</h2>
<section id="loading-the-required-libraries" class="level3">
<h3 class="anchored" data-anchor-id="loading-the-required-libraries">Loading the required libraries</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co">#for data wrangling</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2) <span class="co">#for DEG analysis</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genekitr) <span class="co">#for pathway analysis</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr) <span class="co">#for data import</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(eatTools) <span class="co">#for rbind functions</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(strex) <span class="co">#for string related functions</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr) <span class="co">#for beautiful tables</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt) <span class="co">#for customized table layouts</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap) <span class="co">#for beautiful heatmaps</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyheatmaps) <span class="co">#for heatmap generation from dataframe</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co">#for adding multiple figures to a plot</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer) <span class="co">#for color schemas</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) <span class="co">#for plotting</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix) <span class="co">#to explore complicated design in the DESeq2 formula</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR) <span class="co">#for upset plots</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DEGreport) <span class="co">#for clustering of genes into groups based on expression pattern</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="importing-the-counts-matrix" class="level3">
<h3 class="anchored" data-anchor-id="importing-the-counts-matrix">Importing the counts matrix</h3>
<p>The counts file were imported from the publication <a href="https://physoc.onlinelibrary.wiley.com/doi/10.1113/JP285585" class="uri">https://physoc.onlinelibrary.wiley.com/doi/10.1113/JP285585</a>. The authors investigated the outcome of perturbations of the primary and secondary feedback loops of the molecular clock in skeletal muscle by specific gene ablation of Period circadian regulator 2 (Per2) and RAR-related orphan receptor alpha (Rorα), respectively.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#loading the length-scaled counts tsv file</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"~/Documents/BiostateAI/data/Mansingh2024_expression_matrix.txt"</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span>T) <span class="co">#also attached in the Github folder</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#converting NA into 0 and round off the counts</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>cts[<span class="fu">is.na</span>(cts)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.numeric, <span class="fu">list</span>(round))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="loading-the-metadata-file" class="level3">
<h3 class="anchored" data-anchor-id="loading-the-metadata-file">Loading the metadata file</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">str_detect</span>( <span class="fu">colnames</span>(cts) , <span class="st">'CTRL'</span> )  )</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">revalue</span>( genotype , <span class="fu">c</span>( <span class="st">'TRUE'</span><span class="ot">=</span><span class="st">'WT'</span> , <span class="st">'FALSE'</span> <span class="ot">=</span> <span class="st">'KO'</span>  )  ) <span class="co">#TRUE is WT, FALSE is CTRL</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>time_point <span class="ot">=</span> <span class="fu">factor</span>( <span class="fu">substring</span>(<span class="fu">colnames</span>(cts), <span class="fu">regexpr</span>(<span class="st">"_Z"</span>,<span class="fu">colnames</span>(cts)) <span class="sc">+</span> <span class="dv">2</span>) , <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>,<span class="st">'T4'</span>,<span class="st">'T8'</span>,<span class="st">'T12'</span>,<span class="st">'T16'</span>,<span class="st">'T20'</span>))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> <span class="fu">sapply</span>( <span class="fu">strsplit</span>( <span class="fu">colnames</span>( cts ) , <span class="st">"_"</span>) , <span class="st">'['</span> , <span class="dv">5</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>replicate <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(cts) , <span class="dv">1</span>, <span class="fu">nchar</span>( <span class="st">"HL3YYBGX5"</span> ) ) </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( <span class="at">sample =</span> <span class="fu">colnames</span>(cts) ,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">genotype =</span> genotype ,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">time_point =</span> time_point,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mouse =</span> mouse,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">replicate =</span> replicate</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>                       )</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> coldata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(time_point <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>time_point <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>time_point, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">select</span>(coldata<span class="sc">$</span>sample)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Metadata file"</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Sample counts for genotype and timepoint combinations"</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="afoobevsxs" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#afoobevsxs table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#afoobevsxs thead, #afoobevsxs tbody, #afoobevsxs tfoot, #afoobevsxs tr, #afoobevsxs td, #afoobevsxs th {
  border-style: none;
}

#afoobevsxs p {
  margin: 0;
  padding: 0;
}

#afoobevsxs .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#afoobevsxs .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#afoobevsxs .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#afoobevsxs .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#afoobevsxs .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#afoobevsxs .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#afoobevsxs .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#afoobevsxs .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#afoobevsxs .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#afoobevsxs .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#afoobevsxs .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#afoobevsxs .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#afoobevsxs .gt_spanner_row {
  border-bottom-style: hidden;
}

#afoobevsxs .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#afoobevsxs .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#afoobevsxs .gt_from_md > :first-child {
  margin-top: 0;
}

#afoobevsxs .gt_from_md > :last-child {
  margin-bottom: 0;
}

#afoobevsxs .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#afoobevsxs .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#afoobevsxs .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#afoobevsxs .gt_row_group_first td {
  border-top-width: 2px;
}

#afoobevsxs .gt_row_group_first th {
  border-top-width: 2px;
}

#afoobevsxs .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#afoobevsxs .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#afoobevsxs .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#afoobevsxs .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#afoobevsxs .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#afoobevsxs .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#afoobevsxs .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#afoobevsxs .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#afoobevsxs .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#afoobevsxs .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#afoobevsxs .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#afoobevsxs .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#afoobevsxs .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#afoobevsxs .gt_left {
  text-align: left;
}

#afoobevsxs .gt_center {
  text-align: center;
}

#afoobevsxs .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#afoobevsxs .gt_font_normal {
  font-weight: normal;
}

#afoobevsxs .gt_font_bold {
  font-weight: bold;
}

#afoobevsxs .gt_font_italic {
  font-style: italic;
}

#afoobevsxs .gt_super {
  font-size: 65%;
}

#afoobevsxs .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#afoobevsxs .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#afoobevsxs .gt_indent_1 {
  text-indent: 5px;
}

#afoobevsxs .gt_indent_2 {
  text-indent: 10px;
}

#afoobevsxs .gt_indent_3 {
  text-indent: 15px;
}

#afoobevsxs .gt_indent_4 {
  text-indent: 20px;
}

#afoobevsxs .gt_indent_5 {
  text-indent: 25px;
}

#afoobevsxs .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#afoobevsxs div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="sample">sample</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="genotype">genotype</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="time_point">time_point</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="mouse">mouse</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="replicate">replicate</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_6_4__23_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">23</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_7__19_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">19</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_2__6_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">6</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_8__22_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">22</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_2__6_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">6</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_6_4__23_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">23</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_5_5__2_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">2</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_7__19_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">19</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_5_5__2_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">2</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_2__6_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">6</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_5_6__5_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">5</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_5_6__5_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">5</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_8__22_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">22</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_5_6__5_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">5</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_1__3_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">3</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_1__1_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">1</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_8__24_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">24</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_7__21_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">21</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_6_3__20_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">20</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_7__21_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">21</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_2__4_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">4</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_7__21_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">21</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_1__3_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">3</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_8__24_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">24</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_5_5__2_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">2</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_6_3__20_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">20</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_8__22_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">22</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_7__19_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">19</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_1__1_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">1</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_8__24_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">24</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_2__4_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">4</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_1__1_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">1</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_6_3__20_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">20</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_2__4_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">4</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_6_4__23_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">23</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_1__3_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">3</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(genotype,time_point) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="grfpuwetxp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#grfpuwetxp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#grfpuwetxp thead, #grfpuwetxp tbody, #grfpuwetxp tfoot, #grfpuwetxp tr, #grfpuwetxp td, #grfpuwetxp th {
  border-style: none;
}

#grfpuwetxp p {
  margin: 0;
  padding: 0;
}

#grfpuwetxp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#grfpuwetxp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#grfpuwetxp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#grfpuwetxp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#grfpuwetxp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#grfpuwetxp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#grfpuwetxp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#grfpuwetxp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#grfpuwetxp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#grfpuwetxp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#grfpuwetxp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#grfpuwetxp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#grfpuwetxp .gt_spanner_row {
  border-bottom-style: hidden;
}

#grfpuwetxp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#grfpuwetxp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#grfpuwetxp .gt_from_md > :first-child {
  margin-top: 0;
}

#grfpuwetxp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#grfpuwetxp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#grfpuwetxp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#grfpuwetxp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#grfpuwetxp .gt_row_group_first td {
  border-top-width: 2px;
}

#grfpuwetxp .gt_row_group_first th {
  border-top-width: 2px;
}

#grfpuwetxp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#grfpuwetxp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#grfpuwetxp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#grfpuwetxp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#grfpuwetxp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#grfpuwetxp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#grfpuwetxp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#grfpuwetxp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#grfpuwetxp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#grfpuwetxp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#grfpuwetxp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#grfpuwetxp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#grfpuwetxp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#grfpuwetxp .gt_left {
  text-align: left;
}

#grfpuwetxp .gt_center {
  text-align: center;
}

#grfpuwetxp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#grfpuwetxp .gt_font_normal {
  font-weight: normal;
}

#grfpuwetxp .gt_font_bold {
  font-weight: bold;
}

#grfpuwetxp .gt_font_italic {
  font-style: italic;
}

#grfpuwetxp .gt_super {
  font-size: 65%;
}

#grfpuwetxp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#grfpuwetxp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#grfpuwetxp .gt_indent_1 {
  text-indent: 5px;
}

#grfpuwetxp .gt_indent_2 {
  text-indent: 10px;
}

#grfpuwetxp .gt_indent_3 {
  text-indent: 15px;
}

#grfpuwetxp .gt_indent_4 {
  text-indent: 20px;
}

#grfpuwetxp .gt_indent_5 {
  text-indent: 25px;
}

#grfpuwetxp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#grfpuwetxp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="genotype">genotype</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="time_point">time_point</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
    <tr><td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
    <tr><td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
    <tr><td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
</section>
</section>
<section id="creation-and-exploration-of-dds-object" class="level2">
<h2 class="anchored" data-anchor-id="creation-and-exploration-of-dds-object">Creation and exploration of dds object</h2>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#creation of dds object without prefiltering</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#paired design tissue + time if need to adjust for confounders (time in this case)</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(<span class="at">sampleData =</span> coldata,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">designFormula =</span> <span class="sc">~</span> genotype)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#print(vd$plotlist) #can visualize the model matrix and the various coefficients</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">relevel</span>(dds<span class="sc">$</span>genotype, <span class="at">ref =</span> <span class="st">"WT"</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds) <span class="co">#Main effects for genotype #Expression affected by genotype not correcting for  the timepoint effects</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>)) <span class="co">#FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram without prefiltering of counts"</span>)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Conversion%20to%20a%20DESeq%20object-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">p-value histogram without prefiltering of counts</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Conversion%20to%20a%20DESeq%20object-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Genes with high p-values do seem to have lower gene expression</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>p-value histogram distribution looks bimodal. High expressing count genes fall into low p-value bin which is great. Let’s look at the pattern after prefiltering of the counts.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;</span> <span class="dv">10</span>,)] <span class="co">#opted out after the visualization</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#resultsNames(dds)</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>)) <span class="co">#FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "p-value histogram after prefiltering of counts"</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genes with high p-values do seem to have lower gene expression"</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram after prefiltering of counts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Prefiltering%20of%20counts%20using%20the%20dds%20object-1.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Prefiltering%20of%20counts%20using%20the%20dds%20object-2.png" class="img-fluid"></p>
</div>
</div>
<p>Various contrasts can be identified using <strong>resultNames(dds)</strong> function</p>
<p>The p-value histogram after prefiltering of counts is not looking good. Therefore, I haven’t done prefiltering. Supposedly, DESeq2 takes care of the low counts while identifying DEGs. Refer this link <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/" class="uri">http://varianceexplained.org/statistics/interpreting-pvalue-histogram/</a> for more info on p-value histograms and their interpretations.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#creation of dds object without prefiltering</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#paired design tissue + time if need to adjust for confounders (time in this case)</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="pca-plot" class="level3">
<h3 class="anchored" data-anchor-id="pca-plot">PCA plot</h3>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>vstcounts <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> T)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>normalized_cts <span class="ot">&lt;-</span> <span class="fu">assay</span>(vstcounts) <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>pca_prcomp <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(normalized_cts), <span class="at">center =</span> T, <span class="at">scale =</span> F)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#names(pca_prcomp)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>PC1_and_PC2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">1</span>],</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">PC2 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">type =</span> <span class="fu">rownames</span>(pca_prcomp<span class="sc">$</span>x))</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#using the complete normalized matrix</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot(PC1_and_PC2, aes(x = PC1, y = PC2, col=type)) +</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a> <span class="co"># geom_point() +</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(aes(label = type), hjust = 0, vjust = 0) +</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#coord_fixed()</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co">#using variable features</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vstcounts, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/PCA%20plot-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">PCA when colored by genotype and timepoint</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="sample-correlation-heatmap-using-vst-counts" class="level3">
<h3 class="anchored" data-anchor-id="sample-correlation-heatmap-using-vst-counts">Sample correlation heatmap using vst counts</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sampleCor <span class="ot">&lt;-</span> <span class="fu">cor</span>(normalized_cts)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(normalized_cts))</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDist)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>blueColours <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Blues"</span>)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(blueColours))(<span class="dv">255</span>)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Group =</span> <span class="fu">str_after_nth</span>(<span class="fu">colnames</span>(sampleDistMatrix), <span class="at">n =</span> <span class="dv">5</span>, <span class="st">"_"</span>)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation_row) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sampleDistMatrix)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clustering_distance_cols =</span> sampleDist, </span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> colors,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fontsize_row =</span> <span class="dv">6</span>,</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fontsize_col =</span> <span class="dv">8</span>,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">annotation_row =</span> annotation_row,</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cutree_rows =</span> <span class="dv">4</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>                   )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Sample%20correlation%20heatmap%20using%20vst%20counts-1.png" class="img-fluid"></p>
</div>
</div>
<p>All the timepoint and genotype replicates cluster in 4 groups corresponding to WT, KO and T0,12 combinations.</p>
</section>
</section>
<section id="differential-gene-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="differential-gene-expression-analysis">Differential gene expression analysis</h2>
<section id="main-effects-due-to-genotype" class="level3">
<h3 class="anchored" data-anchor-id="main-effects-due-to-genotype">Main effects due to genotype</h3>
<p>Here, I am identifying genotype-specific (KO versus WT) DEGs. These DEGs would be KO- or WT-enriched.</p>
<section id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype" class="level4">
<h4 class="anchored" data-anchor-id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype">Comparison of Wald versus LRT test DEGs <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></h4>
<p><strong>DEGs can be defined genes passing the threshold: abs(logFC) &gt; 1 &amp; padj &lt; 0.01</strong></p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#not accounting for timepoint</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Wald"</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>res1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> time_point)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> res2 <span class="sc">%&gt;%</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-LRT"</span>,</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>res2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res1,res2)</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="bakryhzcim" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#bakryhzcim table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#bakryhzcim thead, #bakryhzcim tbody, #bakryhzcim tfoot, #bakryhzcim tr, #bakryhzcim td, #bakryhzcim th {
  border-style: none;
}

#bakryhzcim p {
  margin: 0;
  padding: 0;
}

#bakryhzcim .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#bakryhzcim .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#bakryhzcim .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#bakryhzcim .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#bakryhzcim .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bakryhzcim .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bakryhzcim .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bakryhzcim .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#bakryhzcim .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#bakryhzcim .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#bakryhzcim .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#bakryhzcim .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#bakryhzcim .gt_spanner_row {
  border-bottom-style: hidden;
}

#bakryhzcim .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#bakryhzcim .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#bakryhzcim .gt_from_md > :first-child {
  margin-top: 0;
}

#bakryhzcim .gt_from_md > :last-child {
  margin-bottom: 0;
}

#bakryhzcim .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#bakryhzcim .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#bakryhzcim .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#bakryhzcim .gt_row_group_first td {
  border-top-width: 2px;
}

#bakryhzcim .gt_row_group_first th {
  border-top-width: 2px;
}

#bakryhzcim .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bakryhzcim .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#bakryhzcim .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#bakryhzcim .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bakryhzcim .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bakryhzcim .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#bakryhzcim .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#bakryhzcim .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#bakryhzcim .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bakryhzcim .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bakryhzcim .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bakryhzcim .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bakryhzcim .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bakryhzcim .gt_left {
  text-align: left;
}

#bakryhzcim .gt_center {
  text-align: center;
}

#bakryhzcim .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#bakryhzcim .gt_font_normal {
  font-weight: normal;
}

#bakryhzcim .gt_font_bold {
  font-weight: bold;
}

#bakryhzcim .gt_font_italic {
  font-style: italic;
}

#bakryhzcim .gt_super {
  font-size: 65%;
}

#bakryhzcim .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#bakryhzcim .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#bakryhzcim .gt_indent_1 {
  text-indent: 5px;
}

#bakryhzcim .gt_indent_2 {
  text-indent: 10px;
}

#bakryhzcim .gt_indent_3 {
  text-indent: 15px;
}

#bakryhzcim .gt_indent_4 {
  text-indent: 20px;
}

#bakryhzcim .gt_indent_5 {
  text-indent: 25px;
}

#bakryhzcim .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#bakryhzcim div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Wald</td>
<td headers="enrichment" class="gt_row gt_left">KO</td>
<td headers="n" class="gt_row gt_right">163</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Wald</td>
<td headers="enrichment" class="gt_row gt_left">WT</td>
<td headers="n" class="gt_row gt_right">169</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Wald</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18252</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="bnxuzldqqc" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#bnxuzldqqc table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#bnxuzldqqc thead, #bnxuzldqqc tbody, #bnxuzldqqc tfoot, #bnxuzldqqc tr, #bnxuzldqqc td, #bnxuzldqqc th {
  border-style: none;
}

#bnxuzldqqc p {
  margin: 0;
  padding: 0;
}

#bnxuzldqqc .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#bnxuzldqqc .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#bnxuzldqqc .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#bnxuzldqqc .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#bnxuzldqqc .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bnxuzldqqc .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bnxuzldqqc .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bnxuzldqqc .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#bnxuzldqqc .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#bnxuzldqqc .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#bnxuzldqqc .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#bnxuzldqqc .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#bnxuzldqqc .gt_spanner_row {
  border-bottom-style: hidden;
}

#bnxuzldqqc .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#bnxuzldqqc .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#bnxuzldqqc .gt_from_md > :first-child {
  margin-top: 0;
}

#bnxuzldqqc .gt_from_md > :last-child {
  margin-bottom: 0;
}

#bnxuzldqqc .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#bnxuzldqqc .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#bnxuzldqqc .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#bnxuzldqqc .gt_row_group_first td {
  border-top-width: 2px;
}

#bnxuzldqqc .gt_row_group_first th {
  border-top-width: 2px;
}

#bnxuzldqqc .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bnxuzldqqc .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#bnxuzldqqc .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#bnxuzldqqc .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bnxuzldqqc .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bnxuzldqqc .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#bnxuzldqqc .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#bnxuzldqqc .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#bnxuzldqqc .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bnxuzldqqc .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bnxuzldqqc .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bnxuzldqqc .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bnxuzldqqc .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bnxuzldqqc .gt_left {
  text-align: left;
}

#bnxuzldqqc .gt_center {
  text-align: center;
}

#bnxuzldqqc .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#bnxuzldqqc .gt_font_normal {
  font-weight: normal;
}

#bnxuzldqqc .gt_font_bold {
  font-weight: bold;
}

#bnxuzldqqc .gt_font_italic {
  font-style: italic;
}

#bnxuzldqqc .gt_super {
  font-size: 65%;
}

#bnxuzldqqc .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#bnxuzldqqc .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#bnxuzldqqc .gt_indent_1 {
  text-indent: 5px;
}

#bnxuzldqqc .gt_indent_2 {
  text-indent: 10px;
}

#bnxuzldqqc .gt_indent_3 {
  text-indent: 15px;
}

#bnxuzldqqc .gt_indent_4 {
  text-indent: 20px;
}

#bnxuzldqqc .gt_indent_5 {
  text-indent: 25px;
}

#bnxuzldqqc .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#bnxuzldqqc div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-LRT</td>
<td headers="enrichment" class="gt_row gt_left">KO</td>
<td headers="n" class="gt_row gt_right">177</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-LRT</td>
<td headers="enrichment" class="gt_row gt_left">WT</td>
<td headers="n" class="gt_row gt_right">182</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18225</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Comparison%20of%20Wald%20test%20and%20LRT%20-%20genotype-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">DEGs obtained with Wald test <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>DEGs with and without accounting for timepoint yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to genotype.</p>
</section>
<section id="visualization-of-genotype-specific-degs-using-deseq2" class="level4">
<h4 class="anchored" data-anchor-id="visualization-of-genotype-specific-degs-using-deseq2">Visualization of genotype-specific DEGs using DESeq2</h4>
<p>I am going with res1 object that uses Wald test to identify main effects due to genotype.</p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res1,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res1<span class="sc">$</span>gene,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"KO vs WT comparison [Single-factor design]"</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res1<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20genotype-specific%20DEGs-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20genotype-specific%20DEGs-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Tissue-specific DEGs agnostic of the sampling timepoint</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>The genotype based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</p>
</section>
<section id="expression-pattern-of-top-genotype-specific-degs" class="level4">
<h4 class="anchored" data-anchor-id="expression-pattern-of-top-genotype-specific-degs">Expression pattern of top genotype-specific DEGs</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">genotype =</span> <span class="fu">c</span>(<span class="st">"WT"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"KO"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> T,</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> F,</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>),</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>        )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20genotype-specific%20DEGs-1.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span> </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>), <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20genotype-specific%20DEGs-2.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="pathway-analysis-using-the-genotype-specific-degs-main-effects-using-ora" class="level4">
<h4 class="anchored" data-anchor-id="pathway-analysis-using-the-genotype-specific-degs-main-effects-using-ora">Pathway analysis using the genotype-specific DEGs (Main effects) using ORA</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>msigdb_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getMsigdb</span>(<span class="at">org =</span> <span class="st">"mouse"</span>, <span class="at">category =</span> <span class="st">"H"</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>go_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getGO</span>(<span class="at">org =</span> <span class="st">"mouse"</span>, <span class="at">ont =</span> <span class="st">"bp"</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>kegg_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getKEGG</span>(<span class="at">org =</span> <span class="st">"mus"</span>, <span class="at">category =</span> <span class="st">"pathway"</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>react_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getReactome</span>(<span class="at">org =</span> <span class="st">"mouse"</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>gene_set <span class="ot">=</span> <span class="fu">list</span>(msigdb_gs, go_gs, kegg_gs, react_gs)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"KO"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res1))<span class="sc">$</span>result</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"KO"</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res1))<span class="sc">$</span>result</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"WT"</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway profile across multiple databases and genotype conditions"</span></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="tkqjnxvyxk" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#tkqjnxvyxk table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#tkqjnxvyxk thead, #tkqjnxvyxk tbody, #tkqjnxvyxk tfoot, #tkqjnxvyxk tr, #tkqjnxvyxk td, #tkqjnxvyxk th {
  border-style: none;
}

#tkqjnxvyxk p {
  margin: 0;
  padding: 0;
}

#tkqjnxvyxk .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#tkqjnxvyxk .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#tkqjnxvyxk .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#tkqjnxvyxk .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#tkqjnxvyxk .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#tkqjnxvyxk .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tkqjnxvyxk .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#tkqjnxvyxk .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#tkqjnxvyxk .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#tkqjnxvyxk .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#tkqjnxvyxk .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#tkqjnxvyxk .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#tkqjnxvyxk .gt_spanner_row {
  border-bottom-style: hidden;
}

#tkqjnxvyxk .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#tkqjnxvyxk .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#tkqjnxvyxk .gt_from_md > :first-child {
  margin-top: 0;
}

#tkqjnxvyxk .gt_from_md > :last-child {
  margin-bottom: 0;
}

#tkqjnxvyxk .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#tkqjnxvyxk .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#tkqjnxvyxk .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#tkqjnxvyxk .gt_row_group_first td {
  border-top-width: 2px;
}

#tkqjnxvyxk .gt_row_group_first th {
  border-top-width: 2px;
}

#tkqjnxvyxk .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#tkqjnxvyxk .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#tkqjnxvyxk .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#tkqjnxvyxk .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tkqjnxvyxk .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#tkqjnxvyxk .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#tkqjnxvyxk .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#tkqjnxvyxk .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#tkqjnxvyxk .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tkqjnxvyxk .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#tkqjnxvyxk .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#tkqjnxvyxk .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#tkqjnxvyxk .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#tkqjnxvyxk .gt_left {
  text-align: left;
}

#tkqjnxvyxk .gt_center {
  text-align: center;
}

#tkqjnxvyxk .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#tkqjnxvyxk .gt_font_normal {
  font-weight: normal;
}

#tkqjnxvyxk .gt_font_bold {
  font-weight: bold;
}

#tkqjnxvyxk .gt_font_italic {
  font-style: italic;
}

#tkqjnxvyxk .gt_super {
  font-size: 65%;
}

#tkqjnxvyxk .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#tkqjnxvyxk .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#tkqjnxvyxk .gt_indent_1 {
  text-indent: 5px;
}

#tkqjnxvyxk .gt_indent_2 {
  text-indent: 10px;
}

#tkqjnxvyxk .gt_indent_3 {
  text-indent: 15px;
}

#tkqjnxvyxk .gt_indent_4 {
  text-indent: 20px;
}

#tkqjnxvyxk .gt_indent_5 {
  text-indent: 25px;
}

#tkqjnxvyxk .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#tkqjnxvyxk div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Description">Description</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="GeneRatio">GeneRatio</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="BgRatio">BgRatio</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="pvalue">pvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="p.adjust">p.adjust</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="qvalue">qvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID">geneID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID_symbol">geneID_symbol</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Count">Count</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="FoldEnrich">FoldEnrich</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="RichFactor">RichFactor</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneset">geneset</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><th id="stub_1_1" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">1</th>
<td headers="stub_1_1 Description" class="gt_row gt_left">HALLMARK_APICAL_JUNCTION</td>
<td headers="stub_1_1 GeneRatio" class="gt_row gt_right">0.10344828</td>
<td headers="stub_1_1 BgRatio" class="gt_row gt_right">83/1553</td>
<td headers="stub_1_1 pvalue" class="gt_row gt_right">0.1992608</td>
<td headers="stub_1_1 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_1 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_1 geneID" class="gt_row gt_left">ENSMUSG00000029307/ENSMUSG00000038319/ENSMUSG00000044322</td>
<td headers="stub_1_1 geneID_symbol" class="gt_row gt_left">Dmp1/Kcnh2/Dsc1</td>
<td headers="stub_1_1 Count" class="gt_row gt_right">3</td>
<td headers="stub_1_1 FoldEnrich" class="gt_row gt_right">1.935604</td>
<td headers="stub_1_1 RichFactor" class="gt_row gt_right">0.03614458</td>
<td headers="stub_1_1 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_1 enrichment" class="gt_row gt_left">KO</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">2</th>
<td headers="stub_1_2 Description" class="gt_row gt_left">HALLMARK_ESTROGEN_RESPONSE_EARLY</td>
<td headers="stub_1_2 GeneRatio" class="gt_row gt_right">0.06896552</td>
<td headers="stub_1_2 BgRatio" class="gt_row gt_right">62/1553</td>
<td headers="stub_1_2 pvalue" class="gt_row gt_right">0.3235980</td>
<td headers="stub_1_2 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_2 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_2 geneID" class="gt_row gt_left">ENSMUSG00000003051/ENSMUSG00000023043</td>
<td headers="stub_1_2 geneID_symbol" class="gt_row gt_left">Elf3/Krt18</td>
<td headers="stub_1_2 Count" class="gt_row gt_right">2</td>
<td headers="stub_1_2 FoldEnrich" class="gt_row gt_right">1.727475</td>
<td headers="stub_1_2 RichFactor" class="gt_row gt_right">0.03225806</td>
<td headers="stub_1_2 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_2 enrichment" class="gt_row gt_left">KO</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">3</th>
<td headers="stub_1_3 Description" class="gt_row gt_left">HALLMARK_MYOGENESIS</td>
<td headers="stub_1_3 GeneRatio" class="gt_row gt_right">0.10344828</td>
<td headers="stub_1_3 BgRatio" class="gt_row gt_right">113/1553</td>
<td headers="stub_1_3 pvalue" class="gt_row gt_right">0.3542122</td>
<td headers="stub_1_3 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_3 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_3 geneID" class="gt_row gt_left">ENSMUSG00000026253/ENSMUSG00000038319/ENSMUSG00000022037</td>
<td headers="stub_1_3 geneID_symbol" class="gt_row gt_left">Chrng/Kcnh2/Clu</td>
<td headers="stub_1_3 Count" class="gt_row gt_right">3</td>
<td headers="stub_1_3 FoldEnrich" class="gt_row gt_right">1.421727</td>
<td headers="stub_1_3 RichFactor" class="gt_row gt_right">0.02654867</td>
<td headers="stub_1_3 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_3 enrichment" class="gt_row gt_left">KO</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">4</th>
<td headers="stub_1_4 Description" class="gt_row gt_left">HALLMARK_GLYCOLYSIS</td>
<td headers="stub_1_4 GeneRatio" class="gt_row gt_right">0.06896552</td>
<td headers="stub_1_4 BgRatio" class="gt_row gt_right">68/1553</td>
<td headers="stub_1_4 pvalue" class="gt_row gt_right">0.3653448</td>
<td headers="stub_1_4 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_4 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_4 geneID" class="gt_row gt_left">ENSMUSG00000003051/ENSMUSG00000028527</td>
<td headers="stub_1_4 geneID_symbol" class="gt_row gt_left">Elf3/Ak4</td>
<td headers="stub_1_4 Count" class="gt_row gt_right">2</td>
<td headers="stub_1_4 FoldEnrich" class="gt_row gt_right">1.575051</td>
<td headers="stub_1_4 RichFactor" class="gt_row gt_right">0.02941176</td>
<td headers="stub_1_4 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_4 enrichment" class="gt_row gt_left">KO</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">5</th>
<td headers="stub_1_5 Description" class="gt_row gt_left">HALLMARK_INTERFERON_ALPHA_RESPONSE</td>
<td headers="stub_1_5 GeneRatio" class="gt_row gt_right">0.03448276</td>
<td headers="stub_1_5 BgRatio" class="gt_row gt_right">26/1553</td>
<td headers="stub_1_5 pvalue" class="gt_row gt_right">0.3898979</td>
<td headers="stub_1_5 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_5 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_5 geneID" class="gt_row gt_left">ENSMUSG00000031712</td>
<td headers="stub_1_5 geneID_symbol" class="gt_row gt_left">Il15</td>
<td headers="stub_1_5 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_5 FoldEnrich" class="gt_row gt_right">2.059682</td>
<td headers="stub_1_5 RichFactor" class="gt_row gt_right">0.03846154</td>
<td headers="stub_1_5 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_5 enrichment" class="gt_row gt_left">KO</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">6</th>
<td headers="stub_1_6 Description" class="gt_row gt_left">HALLMARK_CHOLESTEROL_HOMEOSTASIS</td>
<td headers="stub_1_6 GeneRatio" class="gt_row gt_right">0.03448276</td>
<td headers="stub_1_6 BgRatio" class="gt_row gt_right">28/1553</td>
<td headers="stub_1_6 pvalue" class="gt_row gt_right">0.4128588</td>
<td headers="stub_1_6 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_6 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_6 geneID" class="gt_row gt_left">ENSMUSG00000022037</td>
<td headers="stub_1_6 geneID_symbol" class="gt_row gt_left">Clu</td>
<td headers="stub_1_6 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_6 FoldEnrich" class="gt_row gt_right">1.912562</td>
<td headers="stub_1_6 RichFactor" class="gt_row gt_right">0.03571429</td>
<td headers="stub_1_6 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_6 enrichment" class="gt_row gt_left">KO</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>None of the genotype-enriched DEGs fall into significant pathways across multiple pathway databases (KEGG, GO-BP, MSigDB hallmark and Reactome) when using ORA approach along with the universe argument</strong></p>
<p>Let’s see if change of approach from ORA to GSEA lead to better results.</p>
<p>ORA method is easy to conduct, but it will lose information when genes’ differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by fold change cutoff. Unlike ORA, FCS tools do not set a threshold for differentially expressed genes. Instead, it gives each detected gene a differential expression score and then evaluates whether the scores are more positive or negative than expected by chance for each gene set.</p>
</section>
<section id="pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea" class="level4">
<h4 class="anchored" data-anchor-id="pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea">Pathway analysis using the genotype-specific DEGs (Main effects) using GSEA</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>rankedGenes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(gene, padj, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue) <span class="sc">*</span> <span class="fu">sign</span>(log2FoldChange),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">entrez =</span> <span class="fu">transId</span>(gene, <span class="at">transTo =</span> <span class="st">"entrez"</span>, <span class="at">keepNA =</span> T, <span class="at">unique =</span> T, <span class="at">org =</span> <span class="st">"mouse"</span>)<span class="sc">$</span>entrezid) <span class="sc">%&gt;%</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(rank) <span class="sc">&amp;</span> <span class="fu">is.finite</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(rank, gene)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">#head(rankedGenes)</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  gsea_results1 <span class="ot">=</span> <span class="fu">safely</span>(genGSEA)(<span class="at">genelist =</span> rankedGenes,</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">q_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]])<span class="sc">$</span>result</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(gsea_results1<span class="sc">$</span>gsea_df)) {</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>    barplots_1[[i]] <span class="ot">&lt;-</span> <span class="fu">plotGSEA</span>(gsea_results1, <span class="at">plot_type =</span> <span class="st">"bar"</span>, <span class="at">colour =</span> <span class="fu">c</span>(<span class="st">"navyblue"</span>, <span class="st">"orange"</span>))</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>    df1 <span class="ot">=</span> <span class="fu">rbind_common</span>(df1, gsea_results1<span class="sc">$</span>gsea_df <span class="sc">%&gt;%</span> </span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">mutate</span>(<span class="at">enrichment =</span> <span class="fu">ifelse</span>(NES <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>),</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]]<span class="sc">$</span>type))</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res1%20output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res1%20output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res1%20output-3.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway table across multiple databases and genotype conditions"</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Signficant pathways (p.adjust &lt; 0.05) identified using GSEA using multiple databases"</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="qstfyfcfip" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#qstfyfcfip table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#qstfyfcfip thead, #qstfyfcfip tbody, #qstfyfcfip tfoot, #qstfyfcfip tr, #qstfyfcfip td, #qstfyfcfip th {
  border-style: none;
}

#qstfyfcfip p {
  margin: 0;
  padding: 0;
}

#qstfyfcfip .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#qstfyfcfip .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#qstfyfcfip .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#qstfyfcfip .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#qstfyfcfip .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qstfyfcfip .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qstfyfcfip .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qstfyfcfip .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#qstfyfcfip .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#qstfyfcfip .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#qstfyfcfip .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#qstfyfcfip .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#qstfyfcfip .gt_spanner_row {
  border-bottom-style: hidden;
}

#qstfyfcfip .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#qstfyfcfip .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#qstfyfcfip .gt_from_md > :first-child {
  margin-top: 0;
}

#qstfyfcfip .gt_from_md > :last-child {
  margin-bottom: 0;
}

#qstfyfcfip .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#qstfyfcfip .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#qstfyfcfip .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#qstfyfcfip .gt_row_group_first td {
  border-top-width: 2px;
}

#qstfyfcfip .gt_row_group_first th {
  border-top-width: 2px;
}

#qstfyfcfip .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qstfyfcfip .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#qstfyfcfip .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#qstfyfcfip .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qstfyfcfip .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qstfyfcfip .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#qstfyfcfip .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#qstfyfcfip .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#qstfyfcfip .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qstfyfcfip .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qstfyfcfip .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qstfyfcfip .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qstfyfcfip .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qstfyfcfip .gt_left {
  text-align: left;
}

#qstfyfcfip .gt_center {
  text-align: center;
}

#qstfyfcfip .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#qstfyfcfip .gt_font_normal {
  font-weight: normal;
}

#qstfyfcfip .gt_font_bold {
  font-weight: bold;
}

#qstfyfcfip .gt_font_italic {
  font-style: italic;
}

#qstfyfcfip .gt_super {
  font-size: 65%;
}

#qstfyfcfip .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#qstfyfcfip .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#qstfyfcfip .gt_indent_1 {
  text-indent: 5px;
}

#qstfyfcfip .gt_indent_2 {
  text-indent: 10px;
}

#qstfyfcfip .gt_indent_3 {
  text-indent: 15px;
}

#qstfyfcfip .gt_indent_4 {
  text-indent: 20px;
}

#qstfyfcfip .gt_indent_5 {
  text-indent: 25px;
}

#qstfyfcfip .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#qstfyfcfip div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Description">Description</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="setSize">setSize</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="enrichmentScore">enrichmentScore</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="NES">NES</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="pvalue">pvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="p.adjust">p.adjust</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="qvalue">qvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="rank">rank</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="leading_edge">leading_edge</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID">geneID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Count">Count</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneset">geneset</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><th id="stub_1_1" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">1</th>
<td headers="stub_1_1 Description" class="gt_row gt_left">HALLMARK_OXIDATIVE_PHOSPHORYLATION</td>
<td headers="stub_1_1 setSize" class="gt_row gt_right">199</td>
<td headers="stub_1_1 enrichmentScore" class="gt_row gt_right">0.8186806</td>
<td headers="stub_1_1 NES" class="gt_row gt_right">2.081145</td>
<td headers="stub_1_1 pvalue" class="gt_row gt_right">5.477149e-12</td>
<td headers="stub_1_1 p.adjust" class="gt_row gt_right">2.738574e-10</td>
<td headers="stub_1_1 qvalue" class="gt_row gt_right">2.479130e-10</td>
<td headers="stub_1_1 rank" class="gt_row gt_right">1349</td>
<td headers="stub_1_1 leading_edge" class="gt_row gt_left">tags=52%, list=8%, signal=49%</td>
<td headers="stub_1_1 geneID" class="gt_row gt_left">67834/67863/66046/76614/68375/12857/21854/11947/11429/14194/56451/66925/20916/28080/11950/17995/11951/22273/12974/170718/226646/66108/17448/66445/68263/78920/66694/71679/110323/12867/11957/11946/235339/214952/228033/230075/15929/11739/78830/66945/15526/12858/69597/66052/18293/27402/11655/14719/67003/66594/18674/22333/227197/18597/625249/22272/72416/66091/57423/12034/66043/14661/67264/66916/75406/225887/407785/13177/14297/66142/12864/70383/11370/66152/113868/109754/110842/67126/72900/231086/12369/17449/16922/68349/66865/226139/13382/68197/67130/66576/67530/110826/66223/52538/12859/17713/64704/12039/68202/22334/66290/66495/11958/97212</td>
<td headers="stub_1_1 Count" class="gt_row gt_right">104</td>
<td headers="stub_1_1 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_1 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">2</th>
<td headers="stub_1_2 Description" class="gt_row gt_left">cellular respiration</td>
<td headers="stub_1_2 setSize" class="gt_row gt_right">214</td>
<td headers="stub_1_2 enrichmentScore" class="gt_row gt_right">0.8409519</td>
<td headers="stub_1_2 NES" class="gt_row gt_right">2.162813</td>
<td headers="stub_1_2 pvalue" class="gt_row gt_right">1.202172e-14</td>
<td headers="stub_1_2 p.adjust" class="gt_row gt_right">7.179370e-11</td>
<td headers="stub_1_2 qvalue" class="gt_row gt_right">7.179370e-11</td>
<td headers="stub_1_2 rank" class="gt_row gt_right">1117</td>
<td headers="stub_1_2 leading_edge" class="gt_row gt_left">tags=41%, list=6%, signal=39%</td>
<td headers="stub_1_2 geneID" class="gt_row gt_left">Idh3a/Ndufb5/Ndufa8/Coq10a/Cox4i1/Aco2/Fh1/Ppif/Suclg1/Coq9/Sdhd/Trap1/Sucla2/Atp5pb/Ndufv1/Uqcrc1/Cs/Idh3b/Ndufs2/Nipsnap2/Ndufa9/Mdh2/Cyc1/Pdhb/Dlst/Uqcrfs1/Cox7c/Ndufa11/Dlat/Abcd1/Sod2/Ndufa10/Ndufb6/Idh3g/Slc25a12/Sdha/Cox5a/Ndufb9/Sdhc/Ogdh/Sirt3/Flcn/Uqcrc2/Uqcr11/Ndufb10/Adsl/Ak4/Akt1/Ndufs1/Pdha1/Cox7a1/Prkaca/Uqcrq/Sdhaf2/Ndufa3/Ogdhl/Atp5mf/Suclg2/Mybbp1a/Ndufb8/Ndufb7/Ndufs7/Ndufs8/Ndufs6/Pik3ca/Mtfr1/Shmt2/Fxn/Slc25a13/Slc25a25/Cox10/Uqcr10/Dnajc15/Stoml2/Chchd10/Etfa/Atp5f1e/Ndufv2/Mdh1/Ndufs3/Dld/Ndufc2/Ndufa6/Uqcrh/Uqcrb/Etfb/Antkmt/Cox5b</td>
<td headers="stub_1_2 Count" class="gt_row gt_right">88</td>
<td headers="stub_1_2 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_2 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">3</th>
<td headers="stub_1_3 Description" class="gt_row gt_left">aerobic respiration</td>
<td headers="stub_1_3 setSize" class="gt_row gt_right">165</td>
<td headers="stub_1_3 enrichmentScore" class="gt_row gt_right">0.8488595</td>
<td headers="stub_1_3 NES" class="gt_row gt_right">2.133432</td>
<td headers="stub_1_3 pvalue" class="gt_row gt_right">3.547798e-12</td>
<td headers="stub_1_3 p.adjust" class="gt_row gt_right">1.059372e-08</td>
<td headers="stub_1_3 qvalue" class="gt_row gt_right">1.059372e-08</td>
<td headers="stub_1_3 rank" class="gt_row gt_right">1117</td>
<td headers="stub_1_3 leading_edge" class="gt_row gt_left">tags=46%, list=6%, signal=44%</td>
<td headers="stub_1_3 geneID" class="gt_row gt_left">Idh3a/Ndufb5/Ndufa8/Cox4i1/Aco2/Fh1/Ppif/Suclg1/Coq9/Sdhd/Sucla2/Atp5pb/Ndufv1/Uqcrc1/Cs/Idh3b/Ndufs2/Nipsnap2/Ndufa9/Mdh2/Cyc1/Pdhb/Dlst/Uqcrfs1/Cox7c/Ndufa11/Dlat/Abcd1/Ndufa10/Ndufb6/Idh3g/Sdha/Cox5a/Ndufb9/Sdhc/Ogdh/Sirt3/Uqcrc2/Uqcr11/Ndufb10/Adsl/Ak4/Akt1/Ndufs1/Pdha1/Cox7a1/Uqcrq/Sdhaf2/Ndufa3/Ogdhl/Atp5mf/Suclg2/Ndufb8/Ndufb7/Ndufs7/Ndufs8/Ndufs6/Mtfr1/Shmt2/Fxn/Cox10/Uqcr10/Dnajc15/Stoml2/Chchd10/Atp5f1e/Ndufv2/Mdh1/Ndufs3/Dld/Ndufc2/Ndufa6/Uqcrh/Uqcrb/Antkmt/Cox5b</td>
<td headers="stub_1_3 Count" class="gt_row gt_right">76</td>
<td headers="stub_1_3 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_3 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">4</th>
<td headers="stub_1_4 Description" class="gt_row gt_left">energy derivation by oxidation of organic compounds</td>
<td headers="stub_1_4 setSize" class="gt_row gt_right">298</td>
<td headers="stub_1_4 enrichmentScore" class="gt_row gt_right">0.7521968</td>
<td headers="stub_1_4 NES" class="gt_row gt_right">1.995603</td>
<td headers="stub_1_4 pvalue" class="gt_row gt_right">3.761816e-10</td>
<td headers="stub_1_4 p.adjust" class="gt_row gt_right">7.488521e-07</td>
<td headers="stub_1_4 qvalue" class="gt_row gt_right">7.488521e-07</td>
<td headers="stub_1_4 rank" class="gt_row gt_right">1043</td>
<td headers="stub_1_4 leading_edge" class="gt_row gt_left">tags=31%, list=6%, signal=30%</td>
<td headers="stub_1_4 geneID" class="gt_row gt_left">Idh3a/Ppp1r1a/Ndufb5/Ndufa8/Stbd1/Coq10a/Cox4i1/Aco2/Fh1/Ppif/Suclg1/Coq9/Sdhd/Trap1/Sucla2/Atp5pb/Ndufv1/Uqcrc1/Cs/Idh3b/Ndufs2/Nipsnap2/Ndufa9/Ugp2/Mdh2/Cyc1/Pdhb/Dlst/Uqcrfs1/Cox7c/Ndufa11/Dlat/Abcd1/Sod2/Ndufa10/Ndufb6/Idh3g/Slc25a12/Sdha/Cox5a/Ndufb9/Sdhc/Ogdh/Sirt3/Flcn/Uqcrc2/Sorbs1/Uqcr11/Ndufb10/Adsl/Ak4/Akt1/Ndufs1/Pdha1/Cox7a1/Prkaca/Uqcrq/Sdhaf2/Ndufa3/Ogdhl/Atp5mf/Suclg2/Mybbp1a/Ndufb8/Ndufb7/Akt2/Ppp1r3b/Ndufs7/Ndufs8/Ndufs6/Pik3ca/Mtfr1/Shmt2/Fxn/Slc25a13/Gys1/Slc25a25/Cox10/Uqcr10/Dnajc15/Stoml2/Chchd10/Etfa/Atp5f1e/Ndufv2/Mdh1/Ndufs3/Dld/Ndufc2/Ndufa6/Uqcrh/Uqcrb/Etfb</td>
<td headers="stub_1_4 Count" class="gt_row gt_right">93</td>
<td headers="stub_1_4 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_4 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">5</th>
<td headers="stub_1_5 Description" class="gt_row gt_left">nucleotide biosynthetic process</td>
<td headers="stub_1_5 setSize" class="gt_row gt_right">231</td>
<td headers="stub_1_5 enrichmentScore" class="gt_row gt_right">0.7360242</td>
<td headers="stub_1_5 NES" class="gt_row gt_right">1.895365</td>
<td headers="stub_1_5 pvalue" class="gt_row gt_right">1.892726e-07</td>
<td headers="stub_1_5 p.adjust" class="gt_row gt_right">2.260672e-04</td>
<td headers="stub_1_5 qvalue" class="gt_row gt_right">2.260672e-04</td>
<td headers="stub_1_5 rank" class="gt_row gt_right">1082</td>
<td headers="stub_1_5 leading_edge" class="gt_row gt_left">tags=27%, list=6%, signal=26%</td>
<td headers="stub_1_5 geneID" class="gt_row gt_left">Ndufb5/Adcy2/Ndufa8/Nos1/Sdhd/Pdk2/Atp5po/Atp5pb/Ndufv1/Atp5mc1/Ndufs2/Ndufa9/Aldoa/Pdhb/Ampd3/Atp5pd/Atp5f1a/Ldhd/Ndufa11/Dlat/Atp5mc3/Ndufa10/Ndufb6/Sdha/Adss1/Ndufb9/Gmpr/Sdhc/Pdhx/Flcn/Stat3/Ak1/Ndufb10/Adsl/Shmt1/Ak4/Ndufs1/Nmnat1/Pdha1/Ndufa3/Atp5mf/Mpc2/Atp5f1d/Ndufb8/Ndufb7/Flad1/Ndufs7/Ndufs8/Ndufs6/Slc25a13/Stoml2/Atp5f1e/Ndufv2/Npr2/Nmrk2/Ndufs3/Acss2/Dld/Ndufc2/Ndufa6/Pdk1/Antkmt</td>
<td headers="stub_1_5 Count" class="gt_row gt_right">62</td>
<td headers="stub_1_5 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_5 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">6</th>
<td headers="stub_1_6 Description" class="gt_row gt_left">nucleoside phosphate biosynthetic process</td>
<td headers="stub_1_6 setSize" class="gt_row gt_right">234</td>
<td headers="stub_1_6 enrichmentScore" class="gt_row gt_right">0.7344934</td>
<td headers="stub_1_6 NES" class="gt_row gt_right">1.891555</td>
<td headers="stub_1_6 pvalue" class="gt_row gt_right">1.763002e-07</td>
<td headers="stub_1_6 p.adjust" class="gt_row gt_right">2.260672e-04</td>
<td headers="stub_1_6 qvalue" class="gt_row gt_right">2.260672e-04</td>
<td headers="stub_1_6 rank" class="gt_row gt_right">1265</td>
<td headers="stub_1_6 leading_edge" class="gt_row gt_left">tags=28%, list=7%, signal=27%</td>
<td headers="stub_1_6 geneID" class="gt_row gt_left">Ndufb5/Adcy2/Ndufa8/Nos1/Sdhd/Pdk2/Atp5po/Atp5pb/Ndufv1/Atp5mc1/Ndufs2/Ndufa9/Aldoa/Pdhb/Ampd3/Atp5pd/Atp5f1a/Ldhd/Ndufa11/Dlat/Atp5mc3/Ndufa10/Ndufb6/Sdha/Adss1/Ndufb9/Gmpr/Sdhc/Pdhx/Flcn/Stat3/Ak1/Ndufb10/Adsl/Shmt1/Ak4/Ndufs1/Nmnat1/Pdha1/Ndufa3/Atp5mf/Mpc2/Atp5f1d/Ndufb8/Ndufb7/Flad1/Ndufs7/Ndufs8/Ndufs6/Slc25a13/Stoml2/Atp5f1e/Ndufv2/Npr2/Nmrk2/Ndufs3/Acss2/Dld/Ndufc2/Ndufa6/Pdk1/Antkmt/Ampd1/Ndufa5/Ak7/Mlycd</td>
<td headers="stub_1_6 Count" class="gt_row gt_right">66</td>
<td headers="stub_1_6 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_6 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_7" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">7</th>
<td headers="stub_1_7 Description" class="gt_row gt_left">oxidative phosphorylation</td>
<td headers="stub_1_7 setSize" class="gt_row gt_right">123</td>
<td headers="stub_1_7 enrichmentScore" class="gt_row gt_right">0.8214854</td>
<td headers="stub_1_7 NES" class="gt_row gt_right">1.982079</td>
<td headers="stub_1_7 pvalue" class="gt_row gt_right">4.729736e-07</td>
<td headers="stub_1_7 p.adjust" class="gt_row gt_right">3.530748e-04</td>
<td headers="stub_1_7 qvalue" class="gt_row gt_right">3.530748e-04</td>
<td headers="stub_1_7 rank" class="gt_row gt_right">1117</td>
<td headers="stub_1_7 leading_edge" class="gt_row gt_left">tags=44%, list=6%, signal=41%</td>
<td headers="stub_1_7 geneID" class="gt_row gt_left">Ndufb5/Ndufa8/Cox4i1/Ppif/Coq9/Sdhd/Atp5pb/Ndufv1/Uqcrc1/Ndufs2/Nipsnap2/Ndufa9/Cyc1/Uqcrfs1/Cox7c/Ndufa11/Abcd1/Ndufa10/Ndufb6/Sdha/Cox5a/Ndufb9/Sdhc/Uqcrc2/Uqcr11/Ndufb10/Ak4/Ndufs1/Cox7a1/Uqcrq/Sdhaf2/Ndufa3/Atp5mf/Ndufb8/Ndufb7/Ndufs7/Ndufs8/Ndufs6/Shmt2/Fxn/Uqcr10/Dnajc15/Stoml2/Chchd10/Atp5f1e/Ndufv2/Ndufs3/Dld/Ndufc2/Ndufa6/Uqcrh/Uqcrb/Antkmt/Cox5b</td>
<td headers="stub_1_7 Count" class="gt_row gt_right">54</td>
<td headers="stub_1_7 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_7 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_8" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">8</th>
<td headers="stub_1_8 Description" class="gt_row gt_left">nucleoside triphosphate biosynthetic process</td>
<td headers="stub_1_8 setSize" class="gt_row gt_right">116</td>
<td headers="stub_1_8 enrichmentScore" class="gt_row gt_right">0.8190367</td>
<td headers="stub_1_8 NES" class="gt_row gt_right">1.963647</td>
<td headers="stub_1_8 pvalue" class="gt_row gt_right">4.124633e-07</td>
<td headers="stub_1_8 p.adjust" class="gt_row gt_right">3.530748e-04</td>
<td headers="stub_1_8 qvalue" class="gt_row gt_right">3.530748e-04</td>
<td headers="stub_1_8 rank" class="gt_row gt_right">1082</td>
<td headers="stub_1_8 leading_edge" class="gt_row gt_left">tags=36%, list=6%, signal=34%</td>
<td headers="stub_1_8 geneID" class="gt_row gt_left">Ndufb5/Ndufa8/Sdhd/Atp5po/Atp5pb/Ndufv1/Atp5mc1/Ndufs2/Ndufa9/Aldoa/Atp5pd/Atp5f1a/Ldhd/Ndufa11/Atp5mc3/Ndufa10/Ndufb6/Sdha/Ndufb9/Sdhc/Flcn/Stat3/Ak1/Ndufb10/Ak4/Ndufs1/Ndufa3/Atp5mf/Atp5f1d/Ndufb8/Ndufb7/Ndufs7/Ndufs8/Ndufs6/Slc25a13/Stoml2/Atp5f1e/Ndufv2/Ndufs3/Ndufc2/Ndufa6/Antkmt</td>
<td headers="stub_1_8 Count" class="gt_row gt_right">42</td>
<td headers="stub_1_8 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_8 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_9" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">9</th>
<td headers="stub_1_9 Description" class="gt_row gt_left">purine-containing compound biosynthetic process</td>
<td headers="stub_1_9 setSize" class="gt_row gt_right">207</td>
<td headers="stub_1_9 enrichmentScore" class="gt_row gt_right">0.7499895</td>
<td headers="stub_1_9 NES" class="gt_row gt_right">1.927976</td>
<td headers="stub_1_9 pvalue" class="gt_row gt_right">4.702901e-07</td>
<td headers="stub_1_9 p.adjust" class="gt_row gt_right">3.530748e-04</td>
<td headers="stub_1_9 qvalue" class="gt_row gt_right">3.530748e-04</td>
<td headers="stub_1_9 rank" class="gt_row gt_right">1082</td>
<td headers="stub_1_9 leading_edge" class="gt_row gt_left">tags=29%, list=6%, signal=28%</td>
<td headers="stub_1_9 geneID" class="gt_row gt_left">Ndufb5/Adcy2/Ndufa8/Nos1/Sdhd/Pdk2/Atp5po/Atp5pb/Ndufv1/Atp5mc1/Ndufs2/Ndufa9/Aldoa/Pdhb/Ampd3/Atp5pd/Atp5f1a/Ldhd/Ndufa11/Dlat/Atp5mc3/Ndufa10/Ndufb6/Sdha/Adss1/Ndufb9/Gmpr/Sdhc/Pdhx/Flcn/Stat3/Ak1/Ndufb10/Adsl/Shmt1/Ak4/Ndufs1/Nmnat1/Pdha1/Ndufa3/Atp5mf/Mpc2/Atp5f1d/Ndufb8/Ndufb7/Ndufs7/Ndufs8/Ndufs6/Shmt2/Slc25a13/Stoml2/Atp5f1e/Ndufv2/Npr2/Ndufs3/Acss2/Dld/Ndufc2/Ndufa6/Pdk1/Antkmt</td>
<td headers="stub_1_9 Count" class="gt_row gt_right">61</td>
<td headers="stub_1_9 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_9 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_10" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">10</th>
<td headers="stub_1_10 Description" class="gt_row gt_left">purine nucleoside triphosphate biosynthetic process</td>
<td headers="stub_1_10 setSize" class="gt_row gt_right">103</td>
<td headers="stub_1_10 enrichmentScore" class="gt_row gt_right">0.8358823</td>
<td headers="stub_1_10 NES" class="gt_row gt_right">1.988286</td>
<td headers="stub_1_10 pvalue" class="gt_row gt_right">8.727454e-07</td>
<td headers="stub_1_10 p.adjust" class="gt_row gt_right">4.863205e-04</td>
<td headers="stub_1_10 qvalue" class="gt_row gt_right">4.863205e-04</td>
<td headers="stub_1_10 rank" class="gt_row gt_right">1082</td>
<td headers="stub_1_10 leading_edge" class="gt_row gt_left">tags=40%, list=6%, signal=38%</td>
<td headers="stub_1_10 geneID" class="gt_row gt_left">Ndufb5/Ndufa8/Sdhd/Atp5po/Atp5pb/Ndufv1/Atp5mc1/Ndufs2/Ndufa9/Aldoa/Atp5pd/Atp5f1a/Ldhd/Ndufa11/Atp5mc3/Ndufa10/Ndufb6/Sdha/Ndufb9/Sdhc/Flcn/Stat3/Ndufb10/Ak4/Ndufs1/Ndufa3/Atp5mf/Atp5f1d/Ndufb8/Ndufb7/Ndufs7/Ndufs8/Ndufs6/Slc25a13/Stoml2/Atp5f1e/Ndufv2/Ndufs3/Ndufc2/Ndufa6/Antkmt</td>
<td headers="stub_1_10 Count" class="gt_row gt_right">41</td>
<td headers="stub_1_10 enrichment" class="gt_row gt_left">KO</td>
<td headers="stub_1_10 geneset" class="gt_row gt_left">bp</td></tr>
    <tr><th id="stub_1_11" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier; font-size: x-small; background-color: #E4E4E4;">11..40</th>
<td headers="stub_1_11 Description" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 setSize" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 enrichmentScore" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 NES" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 pvalue" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 p.adjust" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 qvalue" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 rank" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 leading_edge" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 geneID" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 Count" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 enrichment" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 geneset" class="gt_row gt_left" style="background-color: #E4E4E4;"></td></tr>
    <tr><th id="stub_1_12" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">41</th>
<td headers="stub_1_12 Description" class="gt_row gt_left">Hyaluronan uptake and degradation</td>
<td headers="stub_1_12 setSize" class="gt_row gt_right">12</td>
<td headers="stub_1_12 enrichmentScore" class="gt_row gt_right">-0.9584554</td>
<td headers="stub_1_12 NES" class="gt_row gt_right">-1.669763</td>
<td headers="stub_1_12 pvalue" class="gt_row gt_right">3.560509e-04</td>
<td headers="stub_1_12 p.adjust" class="gt_row gt_right">4.579705e-02</td>
<td headers="stub_1_12 qvalue" class="gt_row gt_right">4.450637e-02</td>
<td headers="stub_1_12 rank" class="gt_row gt_right">6</td>
<td headers="stub_1_12 leading_edge" class="gt_row gt_left">tags=17%, list=0%, signal=17%</td>
<td headers="stub_1_12 geneID" class="gt_row gt_left">15211/192188</td>
<td headers="stub_1_12 Count" class="gt_row gt_right">2</td>
<td headers="stub_1_12 enrichment" class="gt_row gt_left">WT</td>
<td headers="stub_1_12 geneset" class="gt_row gt_left">reactome</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment, <span class="at">legend_labels =</span> <span class="st">"-log10(p.adjust)"</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res1%20output-4.png" class="img-fluid"></p>
</div>
</div>
<p>All the significant pathways (p.adjust &lt; 0.05) are labelled in blue color.</p>
<p><strong>Conclusion: Genotype-specific DEGs fell into multiple respiratory and mitochondrial related pathways across multiple pathway databases (KEGG, GO-BP, MSigDB hallmark and Reactome) when using GSEA approach. The statistically robust approach GSEA is able to identify pathways but not ORA which is surprising.</strong></p>
</section>
</section>
<section id="main-effects-due-to-timepoint" class="level3">
<h3 class="anchored" data-anchor-id="main-effects-due-to-timepoint">Main effects due to timepoint</h3>
<section id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point" class="level4">
<h4 class="anchored" data-anchor-id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point">Comparison of Wald versus LRT test DEGs [Main effects due to time_point]</h4>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> time_point) <span class="co">#not accounting for timepoint</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-Wald"</span>,</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>res4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> genotype)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> res5 <span class="sc">%&gt;%</span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-LRT"</span>,</span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>res5 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res4,res5)</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="dhlnvqrvkt" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#dhlnvqrvkt table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#dhlnvqrvkt thead, #dhlnvqrvkt tbody, #dhlnvqrvkt tfoot, #dhlnvqrvkt tr, #dhlnvqrvkt td, #dhlnvqrvkt th {
  border-style: none;
}

#dhlnvqrvkt p {
  margin: 0;
  padding: 0;
}

#dhlnvqrvkt .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#dhlnvqrvkt .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#dhlnvqrvkt .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#dhlnvqrvkt .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#dhlnvqrvkt .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dhlnvqrvkt .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dhlnvqrvkt .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dhlnvqrvkt .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#dhlnvqrvkt .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#dhlnvqrvkt .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#dhlnvqrvkt .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#dhlnvqrvkt .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#dhlnvqrvkt .gt_spanner_row {
  border-bottom-style: hidden;
}

#dhlnvqrvkt .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#dhlnvqrvkt .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#dhlnvqrvkt .gt_from_md > :first-child {
  margin-top: 0;
}

#dhlnvqrvkt .gt_from_md > :last-child {
  margin-bottom: 0;
}

#dhlnvqrvkt .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#dhlnvqrvkt .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#dhlnvqrvkt .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#dhlnvqrvkt .gt_row_group_first td {
  border-top-width: 2px;
}

#dhlnvqrvkt .gt_row_group_first th {
  border-top-width: 2px;
}

#dhlnvqrvkt .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhlnvqrvkt .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#dhlnvqrvkt .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#dhlnvqrvkt .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dhlnvqrvkt .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhlnvqrvkt .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#dhlnvqrvkt .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#dhlnvqrvkt .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#dhlnvqrvkt .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dhlnvqrvkt .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dhlnvqrvkt .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhlnvqrvkt .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dhlnvqrvkt .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhlnvqrvkt .gt_left {
  text-align: left;
}

#dhlnvqrvkt .gt_center {
  text-align: center;
}

#dhlnvqrvkt .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#dhlnvqrvkt .gt_font_normal {
  font-weight: normal;
}

#dhlnvqrvkt .gt_font_bold {
  font-weight: bold;
}

#dhlnvqrvkt .gt_font_italic {
  font-style: italic;
}

#dhlnvqrvkt .gt_super {
  font-size: 65%;
}

#dhlnvqrvkt .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#dhlnvqrvkt .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#dhlnvqrvkt .gt_indent_1 {
  text-indent: 5px;
}

#dhlnvqrvkt .gt_indent_2 {
  text-indent: 10px;
}

#dhlnvqrvkt .gt_indent_3 {
  text-indent: 15px;
}

#dhlnvqrvkt .gt_indent_4 {
  text-indent: 20px;
}

#dhlnvqrvkt .gt_indent_5 {
  text-indent: 25px;
}

#dhlnvqrvkt .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#dhlnvqrvkt div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-Wald</td>
<td headers="enrichment" class="gt_row gt_left">T0</td>
<td headers="n" class="gt_row gt_right">231</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-Wald</td>
<td headers="enrichment" class="gt_row gt_left">T12</td>
<td headers="n" class="gt_row gt_right">113</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-Wald</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18240</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test [Main effects due to time_point]</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="ppaxmwtmuh" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ppaxmwtmuh table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ppaxmwtmuh thead, #ppaxmwtmuh tbody, #ppaxmwtmuh tfoot, #ppaxmwtmuh tr, #ppaxmwtmuh td, #ppaxmwtmuh th {
  border-style: none;
}

#ppaxmwtmuh p {
  margin: 0;
  padding: 0;
}

#ppaxmwtmuh .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ppaxmwtmuh .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ppaxmwtmuh .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ppaxmwtmuh .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ppaxmwtmuh .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ppaxmwtmuh .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ppaxmwtmuh .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ppaxmwtmuh .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ppaxmwtmuh .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ppaxmwtmuh .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ppaxmwtmuh .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ppaxmwtmuh .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ppaxmwtmuh .gt_spanner_row {
  border-bottom-style: hidden;
}

#ppaxmwtmuh .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ppaxmwtmuh .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ppaxmwtmuh .gt_from_md > :first-child {
  margin-top: 0;
}

#ppaxmwtmuh .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ppaxmwtmuh .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ppaxmwtmuh .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ppaxmwtmuh .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ppaxmwtmuh .gt_row_group_first td {
  border-top-width: 2px;
}

#ppaxmwtmuh .gt_row_group_first th {
  border-top-width: 2px;
}

#ppaxmwtmuh .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ppaxmwtmuh .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ppaxmwtmuh .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ppaxmwtmuh .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ppaxmwtmuh .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ppaxmwtmuh .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ppaxmwtmuh .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ppaxmwtmuh .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ppaxmwtmuh .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ppaxmwtmuh .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ppaxmwtmuh .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ppaxmwtmuh .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ppaxmwtmuh .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ppaxmwtmuh .gt_left {
  text-align: left;
}

#ppaxmwtmuh .gt_center {
  text-align: center;
}

#ppaxmwtmuh .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ppaxmwtmuh .gt_font_normal {
  font-weight: normal;
}

#ppaxmwtmuh .gt_font_bold {
  font-weight: bold;
}

#ppaxmwtmuh .gt_font_italic {
  font-style: italic;
}

#ppaxmwtmuh .gt_super {
  font-size: 65%;
}

#ppaxmwtmuh .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ppaxmwtmuh .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ppaxmwtmuh .gt_indent_1 {
  text-indent: 5px;
}

#ppaxmwtmuh .gt_indent_2 {
  text-indent: 10px;
}

#ppaxmwtmuh .gt_indent_3 {
  text-indent: 15px;
}

#ppaxmwtmuh .gt_indent_4 {
  text-indent: 20px;
}

#ppaxmwtmuh .gt_indent_5 {
  text-indent: 25px;
}

#ppaxmwtmuh .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ppaxmwtmuh div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">T0</td>
<td headers="n" class="gt_row gt_right">242</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">T12</td>
<td headers="n" class="gt_row gt_right">124</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18763</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test [Main effects due to time_point]</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Comparison%20of%20Wald%20test%20and%20LRT%20-%20Timepoint-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">DEGs obtained with Wald test [Main effects due to time_point]</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>DEGs with and without accounting for genotype yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to timepoints.</p>
</section>
<section id="visualization-of-timepoint-specific-degs-using-deseq2" class="level4">
<h4 class="anchored" data-anchor-id="visualization-of-timepoint-specific-degs-using-deseq2">Visualization of timepoint-specific DEGs using DESeq2</h4>
<p>I am going with res4 object that uses Wald test to identify main effects due to genotype. Wald test DEGs are all represented in the LRT DEGs as determined by the Upset plot shown above.</p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res4,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res4<span class="sc">$</span>gene,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"T12 vs T0 comparison [Single-factor design]"</span>,</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res4<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20timepoint-specific%20DEGs-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20timepoint-specific%20DEGs-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Genotype-specific DEGs agnostic of the sampling timepoint</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>The timepoint based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</p>
</section>
<section id="expression-pattern-of-top-time-specific-degs" class="level4">
<h4 class="anchored" data-anchor-id="expression-pattern-of-top-time-specific-degs">Expression pattern of top time-specific DEGs</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">time_point =</span> <span class="fu">c</span>(<span class="st">"T0"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"T12"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span>,</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>        )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20time-specific%20DEGs-1.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span> </span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20time-specific%20DEGs-2.png" class="img-fluid"></p>
</div>
</div>
<p>The timepoint based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</p>
</section>
<section id="pathway-analysis-using-the-timepoint-specific-degs-main-effects-using-ora" class="level4">
<h4 class="anchored" data-anchor-id="pathway-analysis-using-the-timepoint-specific-degs-main-effects-using-ora">Pathway analysis using the timepoint-specific DEGs (Main effects) using ORA</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res4 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"T12"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res4))<span class="sc">$</span>result</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"T12"</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res4 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"T0"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res4))<span class="sc">$</span>result</span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"T0"</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1, <span class="at">plot_type =</span> <span class="st">"dot"</span>)</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway enrichment stats across multiple databases and genotype conditions"</span></span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Significant pathways (labelled in blue) across timepoints when using ORA and multiple pathway databases"</span></span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="xvdhtzbpyt" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#xvdhtzbpyt table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#xvdhtzbpyt thead, #xvdhtzbpyt tbody, #xvdhtzbpyt tfoot, #xvdhtzbpyt tr, #xvdhtzbpyt td, #xvdhtzbpyt th {
  border-style: none;
}

#xvdhtzbpyt p {
  margin: 0;
  padding: 0;
}

#xvdhtzbpyt .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#xvdhtzbpyt .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#xvdhtzbpyt .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#xvdhtzbpyt .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#xvdhtzbpyt .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xvdhtzbpyt .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xvdhtzbpyt .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xvdhtzbpyt .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#xvdhtzbpyt .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#xvdhtzbpyt .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#xvdhtzbpyt .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#xvdhtzbpyt .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#xvdhtzbpyt .gt_spanner_row {
  border-bottom-style: hidden;
}

#xvdhtzbpyt .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#xvdhtzbpyt .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#xvdhtzbpyt .gt_from_md > :first-child {
  margin-top: 0;
}

#xvdhtzbpyt .gt_from_md > :last-child {
  margin-bottom: 0;
}

#xvdhtzbpyt .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#xvdhtzbpyt .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#xvdhtzbpyt .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#xvdhtzbpyt .gt_row_group_first td {
  border-top-width: 2px;
}

#xvdhtzbpyt .gt_row_group_first th {
  border-top-width: 2px;
}

#xvdhtzbpyt .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xvdhtzbpyt .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#xvdhtzbpyt .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#xvdhtzbpyt .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xvdhtzbpyt .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xvdhtzbpyt .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#xvdhtzbpyt .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#xvdhtzbpyt .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#xvdhtzbpyt .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xvdhtzbpyt .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xvdhtzbpyt .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#xvdhtzbpyt .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xvdhtzbpyt .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#xvdhtzbpyt .gt_left {
  text-align: left;
}

#xvdhtzbpyt .gt_center {
  text-align: center;
}

#xvdhtzbpyt .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#xvdhtzbpyt .gt_font_normal {
  font-weight: normal;
}

#xvdhtzbpyt .gt_font_bold {
  font-weight: bold;
}

#xvdhtzbpyt .gt_font_italic {
  font-style: italic;
}

#xvdhtzbpyt .gt_super {
  font-size: 65%;
}

#xvdhtzbpyt .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#xvdhtzbpyt .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#xvdhtzbpyt .gt_indent_1 {
  text-indent: 5px;
}

#xvdhtzbpyt .gt_indent_2 {
  text-indent: 10px;
}

#xvdhtzbpyt .gt_indent_3 {
  text-indent: 15px;
}

#xvdhtzbpyt .gt_indent_4 {
  text-indent: 20px;
}

#xvdhtzbpyt .gt_indent_5 {
  text-indent: 25px;
}

#xvdhtzbpyt .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#xvdhtzbpyt div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Description">Description</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="GeneRatio">GeneRatio</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="BgRatio">BgRatio</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="pvalue">pvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="p.adjust">p.adjust</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="qvalue">qvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID">geneID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneID_symbol">geneID_symbol</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Count">Count</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="FoldEnrich">FoldEnrich</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="RichFactor">RichFactor</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneset">geneset</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><th id="stub_1_1" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">1</th>
<td headers="stub_1_1 Description" class="gt_row gt_left">HALLMARK_MITOTIC_SPINDLE</td>
<td headers="stub_1_1 GeneRatio" class="gt_row gt_right">0.05263158</td>
<td headers="stub_1_1 BgRatio" class="gt_row gt_right">56/1553</td>
<td headers="stub_1_1 pvalue" class="gt_row gt_right">0.5043751</td>
<td headers="stub_1_1 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_1 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_1 geneID" class="gt_row gt_left">ENSMUSG00000026622</td>
<td headers="stub_1_1 geneID_symbol" class="gt_row gt_left">Nek2</td>
<td headers="stub_1_1 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_1 FoldEnrich" class="gt_row gt_right">1.459586</td>
<td headers="stub_1_1 RichFactor" class="gt_row gt_right">0.01785714</td>
<td headers="stub_1_1 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_1 enrichment" class="gt_row gt_left">T12</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">2</th>
<td headers="stub_1_2 Description" class="gt_row gt_left">HALLMARK_SPERMATOGENESIS</td>
<td headers="stub_1_2 GeneRatio" class="gt_row gt_right">0.05263158</td>
<td headers="stub_1_2 BgRatio" class="gt_row gt_right">59/1553</td>
<td headers="stub_1_2 pvalue" class="gt_row gt_right">0.5230204</td>
<td headers="stub_1_2 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_2 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_2 geneID" class="gt_row gt_left">ENSMUSG00000026622</td>
<td headers="stub_1_2 geneID_symbol" class="gt_row gt_left">Nek2</td>
<td headers="stub_1_2 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_2 FoldEnrich" class="gt_row gt_right">1.385370</td>
<td headers="stub_1_2 RichFactor" class="gt_row gt_right">0.01694915</td>
<td headers="stub_1_2 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_2 enrichment" class="gt_row gt_left">T12</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">3</th>
<td headers="stub_1_3 Description" class="gt_row gt_left">HALLMARK_UV_RESPONSE_DN</td>
<td headers="stub_1_3 GeneRatio" class="gt_row gt_right">0.05263158</td>
<td headers="stub_1_3 BgRatio" class="gt_row gt_right">60/1553</td>
<td headers="stub_1_3 pvalue" class="gt_row gt_right">0.5290864</td>
<td headers="stub_1_3 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_3 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_3 geneID" class="gt_row gt_left">ENSMUSG00000059824</td>
<td headers="stub_1_3 geneID_symbol" class="gt_row gt_left">Dbp</td>
<td headers="stub_1_3 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_3 FoldEnrich" class="gt_row gt_right">1.362281</td>
<td headers="stub_1_3 RichFactor" class="gt_row gt_right">0.01666667</td>
<td headers="stub_1_3 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_3 enrichment" class="gt_row gt_left">T12</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">4</th>
<td headers="stub_1_4 Description" class="gt_row gt_left">HALLMARK_KRAS_SIGNALING_DN</td>
<td headers="stub_1_4 GeneRatio" class="gt_row gt_right">0.05263158</td>
<td headers="stub_1_4 BgRatio" class="gt_row gt_right">70/1553</td>
<td headers="stub_1_4 pvalue" class="gt_row gt_right">0.5858587</td>
<td headers="stub_1_4 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_4 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_4 geneID" class="gt_row gt_left">ENSMUSG00000071424</td>
<td headers="stub_1_4 geneID_symbol" class="gt_row gt_left">Grid2</td>
<td headers="stub_1_4 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_4 FoldEnrich" class="gt_row gt_right">1.167669</td>
<td headers="stub_1_4 RichFactor" class="gt_row gt_right">0.01428571</td>
<td headers="stub_1_4 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_4 enrichment" class="gt_row gt_left">T12</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">5</th>
<td headers="stub_1_5 Description" class="gt_row gt_left">HALLMARK_FATTY_ACID_METABOLISM</td>
<td headers="stub_1_5 GeneRatio" class="gt_row gt_right">0.05263158</td>
<td headers="stub_1_5 BgRatio" class="gt_row gt_right">79/1553</td>
<td headers="stub_1_5 pvalue" class="gt_row gt_right">0.6313532</td>
<td headers="stub_1_5 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_5 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_5 geneID" class="gt_row gt_left">ENSMUSG00000074207</td>
<td headers="stub_1_5 geneID_symbol" class="gt_row gt_left">Adh1</td>
<td headers="stub_1_5 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_5 FoldEnrich" class="gt_row gt_right">1.034644</td>
<td headers="stub_1_5 RichFactor" class="gt_row gt_right">0.01265823</td>
<td headers="stub_1_5 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_5 enrichment" class="gt_row gt_left">T12</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">6</th>
<td headers="stub_1_6 Description" class="gt_row gt_left">HALLMARK_G2M_CHECKPOINT</td>
<td headers="stub_1_6 GeneRatio" class="gt_row gt_right">0.05263158</td>
<td headers="stub_1_6 BgRatio" class="gt_row gt_right">80/1553</td>
<td headers="stub_1_6 pvalue" class="gt_row gt_right">0.6361051</td>
<td headers="stub_1_6 p.adjust" class="gt_row gt_right">1</td>
<td headers="stub_1_6 qvalue" class="gt_row gt_right">1</td>
<td headers="stub_1_6 geneID" class="gt_row gt_left">ENSMUSG00000026622</td>
<td headers="stub_1_6 geneID_symbol" class="gt_row gt_left">Nek2</td>
<td headers="stub_1_6 Count" class="gt_row gt_right">1</td>
<td headers="stub_1_6 FoldEnrich" class="gt_row gt_right">1.021711</td>
<td headers="stub_1_6 RichFactor" class="gt_row gt_right">0.01250000</td>
<td headers="stub_1_6 geneset" class="gt_row gt_left">msigdb</td>
<td headers="stub_1_6 enrichment" class="gt_row gt_left">T12</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid"></p>
</div>
</div>
<p><strong>Only the T0-enriched DEGs are associated with Reactome pathways significantly when using ORA approach along with the universe argument. T0-enriched pathways are mainly collagen and ECM related.</strong></p>
<p>Let’s see if change of approach from ORA to GSEA lead to different results.</p>
<p>ORA method is easy to conduct, but it will lose information when genes’ differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by fold change cutoff. Unlike ORA, FCS tools do not set a threshold for differentially expressed genes. Instead, it gives each detected gene a differential expression score and then evaluates whether the scores are more positive or negative than expected by chance for each gene set.</p>
</section>
<section id="pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea-1" class="level4">
<h4 class="anchored" data-anchor-id="pathway-analysis-using-the-genotype-specific-degs-main-effects-using-gsea-1">Pathway analysis using the genotype-specific DEGs (Main effects) using GSEA</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>rankedGenes <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(gene, padj, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue) <span class="sc">*</span> <span class="fu">sign</span>(log2FoldChange),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">entrez =</span> <span class="fu">transId</span>(gene, <span class="at">transTo =</span> <span class="st">"entrez"</span>, <span class="at">keepNA =</span> T, <span class="at">unique =</span> T, <span class="at">org =</span> <span class="st">"mouse"</span>)<span class="sc">$</span>entrezid) <span class="sc">%&gt;%</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(rank) <span class="sc">&amp;</span> <span class="fu">is.finite</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(rank, gene)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co">#head(rankedGenes)</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  gsea_results1 <span class="ot">=</span> <span class="fu">safely</span>(genGSEA)(<span class="at">genelist =</span> rankedGenes,</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">q_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]])<span class="sc">$</span>result</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(gsea_results1<span class="sc">$</span>gsea_df)) {</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>    barplots_1[[i]] <span class="ot">&lt;-</span> <span class="fu">plotGSEA</span>(gsea_results1, <span class="at">plot_type =</span> <span class="st">"bar"</span>, <span class="at">colour =</span> <span class="fu">c</span>(<span class="st">"navyblue"</span>, <span class="st">"orange"</span>))</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>    df1 <span class="ot">=</span> <span class="fu">rbind_common</span>(df1, gsea_results1<span class="sc">$</span>gsea_df <span class="sc">%&gt;%</span> </span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">mutate</span>(<span class="at">enrichment =</span> <span class="fu">ifelse</span>(NES <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>),</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]]<span class="sc">$</span>type))</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res4%20output-1.png" class="img-fluid"></p>
</div>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res4%20output-2.png" class="img-fluid"></p>
</div>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res4%20output-3.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway table across multiple databases and genotype conditions"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Signficant pathways (p.adjust &lt; 0.01) identified using GSEA using multiple databases"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="ocnpdmefdd" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ocnpdmefdd table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ocnpdmefdd thead, #ocnpdmefdd tbody, #ocnpdmefdd tfoot, #ocnpdmefdd tr, #ocnpdmefdd td, #ocnpdmefdd th {
  border-style: none;
}

#ocnpdmefdd p {
  margin: 0;
  padding: 0;
}

#ocnpdmefdd .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ocnpdmefdd .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ocnpdmefdd .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ocnpdmefdd .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ocnpdmefdd .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ocnpdmefdd .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ocnpdmefdd .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ocnpdmefdd .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ocnpdmefdd .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ocnpdmefdd .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ocnpdmefdd .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ocnpdmefdd .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ocnpdmefdd .gt_spanner_row {
  border-bottom-style: hidden;
}

#ocnpdmefdd .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ocnpdmefdd .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ocnpdmefdd .gt_from_md > :first-child {
  margin-top: 0;
}

#ocnpdmefdd .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ocnpdmefdd .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ocnpdmefdd .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ocnpdmefdd .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ocnpdmefdd .gt_row_group_first td {
  border-top-width: 2px;
}

#ocnpdmefdd .gt_row_group_first th {
  border-top-width: 2px;
}

#ocnpdmefdd .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ocnpdmefdd .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ocnpdmefdd .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ocnpdmefdd .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ocnpdmefdd .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ocnpdmefdd .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ocnpdmefdd .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ocnpdmefdd .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ocnpdmefdd .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ocnpdmefdd .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ocnpdmefdd .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ocnpdmefdd .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ocnpdmefdd .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ocnpdmefdd .gt_left {
  text-align: left;
}

#ocnpdmefdd .gt_center {
  text-align: center;
}

#ocnpdmefdd .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ocnpdmefdd .gt_font_normal {
  font-weight: normal;
}

#ocnpdmefdd .gt_font_bold {
  font-weight: bold;
}

#ocnpdmefdd .gt_font_italic {
  font-style: italic;
}

#ocnpdmefdd .gt_super {
  font-size: 65%;
}

#ocnpdmefdd .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ocnpdmefdd .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ocnpdmefdd .gt_indent_1 {
  text-indent: 5px;
}

#ocnpdmefdd .gt_indent_2 {
  text-indent: 10px;
}

#ocnpdmefdd .gt_indent_3 {
  text-indent: 15px;
}

#ocnpdmefdd .gt_indent_4 {
  text-indent: 20px;
}

#ocnpdmefdd .gt_indent_5 {
  text-indent: 25px;
}

#ocnpdmefdd .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ocnpdmefdd div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Description">Description</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="setSize">setSize</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="enrichmentScore">enrichmentScore</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="NES">NES</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="pvalue">pvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="p.adjust">p.adjust</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="qvalue">qvalue</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="rank">rank</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="leading_edge">leading_edge</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="geneID">geneID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Count">Count</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="geneset">geneset</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><th id="stub_1_1" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">1</th>
<td headers="stub_1_1 Description" class="gt_row gt_left">HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION</td>
<td headers="stub_1_1 setSize" class="gt_row gt_right">187</td>
<td headers="stub_1_1 enrichmentScore" class="gt_row gt_right">-0.8342020</td>
<td headers="stub_1_1 NES" class="gt_row gt_right">-2.163463</td>
<td headers="stub_1_1 pvalue" class="gt_row gt_right">2.455245e-14</td>
<td headers="stub_1_1 p.adjust" class="gt_row gt_right">1.227622e-12</td>
<td headers="stub_1_1 qvalue" class="gt_row gt_right">8.270298e-13</td>
<td headers="stub_1_1 rank" class="gt_row gt_right">1150</td>
<td headers="stub_1_1 leading_edge" class="gt_row gt_left">tags=41%, list=7%, signal=38%</td>
<td headers="stub_1_1 geneID" class="gt_row gt_right">16416/20583/12552/17181/56213/21825/18858/21817/13132/17392/12931/326618/16790/50706/16949/329941/20969/16971/24053/14115/16402/13601/21929/50530/107581/16774/22339/107589/12505/14451/65112/30794/12111/27279/16948/16010/18822/16398/23876/18596/21923/21838/22418/14268/102857/12153/98932/52118/17294/21345/192176/226519/17390/20971/11475/14314/13003/18542/22240/12831/109624/94352/12843/20692/12842/17387/13717/12834/20315/12825/12827/12835/12826/12832/14118/12406</td>
<td headers="stub_1_1 Count" class="gt_row gt_right">76</td>
<td headers="stub_1_1 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_1 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">2</th>
<td headers="stub_1_2 Description" class="gt_row gt_left">HALLMARK_APICAL_JUNCTION</td>
<td headers="stub_1_2 setSize" class="gt_row gt_right">183</td>
<td headers="stub_1_2 enrichmentScore" class="gt_row gt_right">-0.7115389</td>
<td headers="stub_1_2 NES" class="gt_row gt_right">-1.837410</td>
<td headers="stub_1_2 pvalue" class="gt_row gt_right">1.105202e-05</td>
<td headers="stub_1_2 p.adjust" class="gt_row gt_right">2.763005e-04</td>
<td headers="stub_1_2 qvalue" class="gt_row gt_right">1.861393e-04</td>
<td headers="stub_1_2 rank" class="gt_row gt_right">1461</td>
<td headers="stub_1_2 leading_edge" class="gt_row gt_left">tags=28%, list=8%, signal=26%</td>
<td headers="stub_1_2 geneID" class="gt_row gt_right">14026/21810/12388/20312/16412/65964/14086/192156/12552/65970/109711/17886/16480/57342/12331/19264/16400/17698/103733/13822/68481/14678/107581/75599/16780/16774/21827/15896/73341/58235/16398/22330/18613/21838/229715/12153/12798/231070/77579/98932/16367/22793/22323/17390/12741/13003/17395/11490/11465/60595/11461/14118</td>
<td headers="stub_1_2 Count" class="gt_row gt_right">52</td>
<td headers="stub_1_2 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_2 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">3</th>
<td headers="stub_1_3 Description" class="gt_row gt_left">HALLMARK_MYOGENESIS</td>
<td headers="stub_1_3 setSize" class="gt_row gt_right">198</td>
<td headers="stub_1_3 enrichmentScore" class="gt_row gt_right">-0.6753379</td>
<td headers="stub_1_3 NES" class="gt_row gt_right">-1.753949</td>
<td headers="stub_1_3 pvalue" class="gt_row gt_right">8.009097e-05</td>
<td headers="stub_1_3 p.adjust" class="gt_row gt_right">1.334849e-03</td>
<td headers="stub_1_3 qvalue" class="gt_row gt_right">8.992670e-04</td>
<td headers="stub_1_3 rank" class="gt_row gt_right">1759</td>
<td headers="stub_1_3 leading_edge" class="gt_row gt_left">tags=26%, list=10%, signal=24%</td>
<td headers="stub_1_3 geneID" class="gt_row gt_right">17913/18128/11790/140570/30953/16985/20399/29856/11820/20249/17928/13845/16412/11537/24131/17880/71994/17886/20740/227753/17907/13644/58226/24053/20657/78286/17532/107589/29817/14245/53412/14613/16404/53311/76954/102857/12391/21345/14313/12575/12709/97064/23796/20250/20692/12842/21679/12834/12825/12827/12819/12835</td>
<td headers="stub_1_3 Count" class="gt_row gt_right">52</td>
<td headers="stub_1_3 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_3 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">4</th>
<td headers="stub_1_4 Description" class="gt_row gt_left">HALLMARK_COAGULATION</td>
<td headers="stub_1_4 setSize" class="gt_row gt_right">99</td>
<td headers="stub_1_4 enrichmentScore" class="gt_row gt_right">-0.7592688</td>
<td headers="stub_1_4 NES" class="gt_row gt_right">-1.815097</td>
<td headers="stub_1_4 pvalue" class="gt_row gt_right">1.534910e-04</td>
<td headers="stub_1_4 p.adjust" class="gt_row gt_right">1.918638e-03</td>
<td headers="stub_1_4 qvalue" class="gt_row gt_right">1.292556e-03</td>
<td headers="stub_1_4 rank" class="gt_row gt_right">1331</td>
<td headers="stub_1_4 leading_edge" class="gt_row gt_left">tags=32%, list=8%, signal=30%</td>
<td headers="stub_1_4 geneID" class="gt_row gt_right">68337/22041/11537/16416/12259/18550/56213/13007/21825/17392/227753/16971/13036/11843/14360/93691/18791/14066/12258/16398/18613/22388/14268/12153/17388/17390/17395/18591/21824/20692/17387/14118</td>
<td headers="stub_1_4 Count" class="gt_row gt_right">32</td>
<td headers="stub_1_4 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_4 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">5</th>
<td headers="stub_1_5 Description" class="gt_row gt_left">HALLMARK_MYC_TARGETS_V2</td>
<td headers="stub_1_5 setSize" class="gt_row gt_right">58</td>
<td headers="stub_1_5 enrichmentScore" class="gt_row gt_right">0.7499887</td>
<td headers="stub_1_5 NES" class="gt_row gt_right">1.803142</td>
<td headers="stub_1_5 pvalue" class="gt_row gt_right">1.226233e-03</td>
<td headers="stub_1_5 p.adjust" class="gt_row gt_right">8.219668e-03</td>
<td headers="stub_1_5 qvalue" class="gt_row gt_right">5.537460e-03</td>
<td headers="stub_1_5 rank" class="gt_row gt_right">1314</td>
<td headers="stub_1_5 leading_edge" class="gt_row gt_left">tags=29%, list=8%, signal=27%</td>
<td headers="stub_1_5 geneID" class="gt_row gt_right">53608/15277/72515/70683/67134/17217/17869/338359/59028/27966/30877/15278/18148/66942/64934/20810/76130</td>
<td headers="stub_1_5 Count" class="gt_row gt_right">17</td>
<td headers="stub_1_5 enrichment" class="gt_row gt_left">T12</td>
<td headers="stub_1_5 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">6</th>
<td headers="stub_1_6 Description" class="gt_row gt_left">HALLMARK_TGF_BETA_SIGNALING</td>
<td headers="stub_1_6 setSize" class="gt_row gt_right">53</td>
<td headers="stub_1_6 enrichmentScore" class="gt_row gt_right">-0.7598829</td>
<td headers="stub_1_6 NES" class="gt_row gt_right">-1.688685</td>
<td headers="stub_1_6 pvalue" class="gt_row gt_right">1.315147e-03</td>
<td headers="stub_1_6 p.adjust" class="gt_row gt_right">8.219668e-03</td>
<td headers="stub_1_6 qvalue" class="gt_row gt_right">5.537460e-03</td>
<td headers="stub_1_6 rank" class="gt_row gt_right">1088</td>
<td headers="stub_1_6 leading_edge" class="gt_row gt_left">tags=25%, list=6%, signal=23%</td>
<td headers="stub_1_6 geneID" class="gt_row gt_right">18550/17872/21825/20481/21815/20602/106572/66313/65112/11477/97064/13805/17127</td>
<td headers="stub_1_6 Count" class="gt_row gt_right">13</td>
<td headers="stub_1_6 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_6 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_7" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">7</th>
<td headers="stub_1_7 Description" class="gt_row gt_left">HALLMARK_ANGIOGENESIS</td>
<td headers="stub_1_7 setSize" class="gt_row gt_right">32</td>
<td headers="stub_1_7 enrichmentScore" class="gt_row gt_right">-0.8217764</td>
<td headers="stub_1_7 NES" class="gt_row gt_right">-1.684566</td>
<td headers="stub_1_7 pvalue" class="gt_row gt_right">1.024757e-03</td>
<td headers="stub_1_7 p.adjust" class="gt_row gt_right">8.219668e-03</td>
<td headers="stub_1_7 qvalue" class="gt_row gt_right">5.537460e-03</td>
<td headers="stub_1_7 rank" class="gt_row gt_right">843</td>
<td headers="stub_1_7 leading_edge" class="gt_row gt_left">tags=50%, list=5%, signal=48%</td>
<td headers="stub_1_7 geneID" class="gt_row gt_right">21857/14083/17701/16450/18590/16449/11820/50706/22339/12444/14314/13003/21824/22370/12825/12832</td>
<td headers="stub_1_7 Count" class="gt_row gt_right">16</td>
<td headers="stub_1_7 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_7 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_8" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">8</th>
<td headers="stub_1_8 Description" class="gt_row gt_left">HALLMARK_OXIDATIVE_PHOSPHORYLATION</td>
<td headers="stub_1_8 setSize" class="gt_row gt_right">199</td>
<td headers="stub_1_8 enrichmentScore" class="gt_row gt_right">0.5868094</td>
<td headers="stub_1_8 NES" class="gt_row gt_right">1.673654</td>
<td headers="stub_1_8 pvalue" class="gt_row gt_right">1.278134e-03</td>
<td headers="stub_1_8 p.adjust" class="gt_row gt_right">8.219668e-03</td>
<td headers="stub_1_8 qvalue" class="gt_row gt_right">5.537460e-03</td>
<td headers="stub_1_8 rank" class="gt_row gt_right">3047</td>
<td headers="stub_1_8 leading_edge" class="gt_row gt_left">tags=23%, list=17%, signal=19%</td>
<td headers="stub_1_8 geneID" class="gt_row gt_right">381511/27273/12894/104776/338359/66335/74143/59054/27425/53375/12039/170731/11973/11931/18105/57279/11306/68349/11949/72416/18984/66383/20463/108664/73834/28185/15926/66841/93747/18115/18597/66290/12856/55951/211253/67442/14661/15526/69089/22334/59040/11739/22333/110446/11984/67680</td>
<td headers="stub_1_8 Count" class="gt_row gt_right">46</td>
<td headers="stub_1_8 enrichment" class="gt_row gt_left">T12</td>
<td headers="stub_1_8 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_9" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">9</th>
<td headers="stub_1_9 Description" class="gt_row gt_left">HALLMARK_NOTCH_SIGNALING</td>
<td headers="stub_1_9 setSize" class="gt_row gt_right">32</td>
<td headers="stub_1_9 enrichmentScore" class="gt_row gt_right">-0.8099967</td>
<td headers="stub_1_9 NES" class="gt_row gt_right">-1.660419</td>
<td headers="stub_1_9 pvalue" class="gt_row gt_right">1.509749e-03</td>
<td headers="stub_1_9 p.adjust" class="gt_row gt_right">8.387493e-03</td>
<td headers="stub_1_9 qvalue" class="gt_row gt_right">5.650521e-03</td>
<td headers="stub_1_9 rank" class="gt_row gt_right">1716</td>
<td headers="stub_1_9 leading_edge" class="gt_row gt_left">tags=44%, list=10%, signal=40%</td>
<td headers="stub_1_9 geneID" class="gt_row gt_right">18128/16449/14534/12443/18129/19015/18750/56198/18131/22418/13388/60406/103583/207521</td>
<td headers="stub_1_9 Count" class="gt_row gt_right">14</td>
<td headers="stub_1_9 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_9 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_10" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">10</th>
<td headers="stub_1_10 Description" class="gt_row gt_left">HALLMARK_CHOLESTEROL_HOMEOSTASIS</td>
<td headers="stub_1_10 setSize" class="gt_row gt_right">70</td>
<td headers="stub_1_10 enrichmentScore" class="gt_row gt_right">-0.7487038</td>
<td headers="stub_1_10 NES" class="gt_row gt_right">-1.720135</td>
<td headers="stub_1_10 pvalue" class="gt_row gt_right">1.816091e-03</td>
<td headers="stub_1_10 p.adjust" class="gt_row gt_right">9.080456e-03</td>
<td headers="stub_1_10 qvalue" class="gt_row gt_right">6.117360e-03</td>
<td headers="stub_1_10 rank" class="gt_row gt_right">1501</td>
<td headers="stub_1_10 leading_edge" class="gt_row gt_left">tags=34%, list=9%, signal=31%</td>
<td headers="stub_1_10 geneID" class="gt_row gt_right">66102/16449/20195/73166/63913/20249/192156/208715/16592/13121/71914/15357/27279/170459/11910/16835/20788/18030/56473/20775/11465/20250/14104/11747</td>
<td headers="stub_1_10 Count" class="gt_row gt_right">24</td>
<td headers="stub_1_10 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_10 geneset" class="gt_row gt_left">msigdb</td></tr>
    <tr><th id="stub_1_11" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier; font-size: x-small; background-color: #E4E4E4;">11..213</th>
<td headers="stub_1_11 Description" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 setSize" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 enrichmentScore" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 NES" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 pvalue" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 p.adjust" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 qvalue" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 rank" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 leading_edge" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 geneID" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 Count" class="gt_row gt_right" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 enrichment" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 geneset" class="gt_row gt_left" style="background-color: #E4E4E4;"></td></tr>
    <tr><th id="stub_1_12" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">214</th>
<td headers="stub_1_12 Description" class="gt_row gt_left">RAF/MAP kinase cascade</td>
<td headers="stub_1_12 setSize" class="gt_row gt_right">234</td>
<td headers="stub_1_12 enrichmentScore" class="gt_row gt_right">-0.5936877</td>
<td headers="stub_1_12 NES" class="gt_row gt_right">-1.555362</td>
<td headers="stub_1_12 pvalue" class="gt_row gt_right">2.156019e-03</td>
<td headers="stub_1_12 p.adjust" class="gt_row gt_right">4.822921e-02</td>
<td headers="stub_1_12 qvalue" class="gt_row gt_right">4.223232e-02</td>
<td headers="stub_1_12 rank" class="gt_row gt_right">1313</td>
<td headers="stub_1_12 leading_edge" class="gt_row gt_left">tags=16%, list=8%, signal=15%</td>
<td headers="stub_1_12 geneID" class="gt_row gt_right">18706/13537/14186/19053/73699/240057/16416/17169/16653/12988/240672/12314/14177/83379/17532/14360/19731/16706/110157/22330/18596/14586/18595/14268/16367/54153/19730/69601/22190/226849/18591/11465/19713/21687/18611/18218/11461</td>
<td headers="stub_1_12 Count" class="gt_row gt_right">37</td>
<td headers="stub_1_12 enrichment" class="gt_row gt_left">T0</td>
<td headers="stub_1_12 geneset" class="gt_row gt_left">reactome</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.01</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment, <span class="at">legend_labels =</span> <span class="st">"-log10(p.adjust)"</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/GSEA%20approach%20using%20the%20res4%20output-4.png" class="img-fluid"></p>
</div>
</div>
<p>All the significant pathways (p.adjust &lt; 0.01) are labelled in blue color. To beautify the output and the plots, I resorted to a stricter cutoff for pathways i.e.&nbsp;p.adjust &lt; 0.01 instead of p.adjust &lt; 0.05</p>
<p><strong>Conclusion: T0-enriched DEGs fell into ECM, collagen and EMT related pathways when using multiple pathway databases (KEGG, GO-BP, MSigDB hallmark and Reactome) when using GSEA approach. GSEA approach also confirmed the findings from the ORA approach. In addition, T12 associated pathways belong to mitochondria, respiration and Oxidative phosphorylation pathway terms. The statistically robust approach GSEA is able to identify more pathways compared to ORA which is surprising.</strong></p>
</section>
</section>
<section id="interaction-effects" class="level3">
<h3 class="anchored" data-anchor-id="interaction-effects">Interaction effects</h3>
<section id="comparison-of-wald-versus-lrt-test---interaction-effects" class="level4">
<h4 class="anchored" data-anchor-id="comparison-of-wald-versus-lrt-test---interaction-effects">Comparison of Wald versus LRT test - interaction effects</h4>
<p><strong>DEGs can be defined genes passing the threshold: abs(logFC) &gt; 1 &amp; padj &lt; 0.01</strong></p>
<p>This is useful to detect interaction between genotype and timepoint.</p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">*</span> time_point) <span class="co">#interaction term</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co">#vd &lt;- VisualizeDesign(sampleData = coldata,</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a> <span class="co">#                     designFormula = ~ genotype * time_point)</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">#print(vd$plotlist) #can visualize the model matrix and the various coefficients</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>res6 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">name =</span> <span class="st">"genotypeWT.time_pointT12"</span>)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>res6 <span class="ot">&lt;-</span> res6 <span class="sc">%&gt;%</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"Interaction-Wald"</span>,</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>res6 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">*</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">*</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point)</span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>res7 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">name =</span> <span class="st">"genotypeWT.time_pointT12"</span>)</span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>res7 <span class="ot">&lt;-</span> res7 <span class="sc">%&gt;%</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"Interaction-LRT"</span>,</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a>res7 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>res8 <span class="ot">=</span> <span class="fu">rbind_common</span>(res6,res7)</span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res8, <span class="at">f =</span> <span class="fu">list</span>(res8<span class="sc">$</span>comparison, res8<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="dsmmyemkgh" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#dsmmyemkgh table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#dsmmyemkgh thead, #dsmmyemkgh tbody, #dsmmyemkgh tfoot, #dsmmyemkgh tr, #dsmmyemkgh td, #dsmmyemkgh th {
  border-style: none;
}

#dsmmyemkgh p {
  margin: 0;
  padding: 0;
}

#dsmmyemkgh .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#dsmmyemkgh .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#dsmmyemkgh .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#dsmmyemkgh .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#dsmmyemkgh .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dsmmyemkgh .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dsmmyemkgh .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dsmmyemkgh .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#dsmmyemkgh .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#dsmmyemkgh .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#dsmmyemkgh .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#dsmmyemkgh .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#dsmmyemkgh .gt_spanner_row {
  border-bottom-style: hidden;
}

#dsmmyemkgh .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#dsmmyemkgh .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#dsmmyemkgh .gt_from_md > :first-child {
  margin-top: 0;
}

#dsmmyemkgh .gt_from_md > :last-child {
  margin-bottom: 0;
}

#dsmmyemkgh .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#dsmmyemkgh .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#dsmmyemkgh .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#dsmmyemkgh .gt_row_group_first td {
  border-top-width: 2px;
}

#dsmmyemkgh .gt_row_group_first th {
  border-top-width: 2px;
}

#dsmmyemkgh .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dsmmyemkgh .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#dsmmyemkgh .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#dsmmyemkgh .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dsmmyemkgh .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dsmmyemkgh .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#dsmmyemkgh .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#dsmmyemkgh .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#dsmmyemkgh .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dsmmyemkgh .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dsmmyemkgh .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dsmmyemkgh .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dsmmyemkgh .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dsmmyemkgh .gt_left {
  text-align: left;
}

#dsmmyemkgh .gt_center {
  text-align: center;
}

#dsmmyemkgh .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#dsmmyemkgh .gt_font_normal {
  font-weight: normal;
}

#dsmmyemkgh .gt_font_bold {
  font-weight: bold;
}

#dsmmyemkgh .gt_font_italic {
  font-style: italic;
}

#dsmmyemkgh .gt_super {
  font-size: 65%;
}

#dsmmyemkgh .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#dsmmyemkgh .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#dsmmyemkgh .gt_indent_1 {
  text-indent: 5px;
}

#dsmmyemkgh .gt_indent_2 {
  text-indent: 10px;
}

#dsmmyemkgh .gt_indent_3 {
  text-indent: 15px;
}

#dsmmyemkgh .gt_indent_4 {
  text-indent: 20px;
}

#dsmmyemkgh .gt_indent_5 {
  text-indent: 25px;
}

#dsmmyemkgh .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#dsmmyemkgh div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">Interaction-Wald</td>
<td headers="enrichment" class="gt_row gt_left">KO</td>
<td headers="n" class="gt_row gt_right">12</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">Interaction-Wald</td>
<td headers="enrichment" class="gt_row gt_left">WT</td>
<td headers="n" class="gt_row gt_right">2</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">Interaction-Wald</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">13117</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="jxprhawyhp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jxprhawyhp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jxprhawyhp thead, #jxprhawyhp tbody, #jxprhawyhp tfoot, #jxprhawyhp tr, #jxprhawyhp td, #jxprhawyhp th {
  border-style: none;
}

#jxprhawyhp p {
  margin: 0;
  padding: 0;
}

#jxprhawyhp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jxprhawyhp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jxprhawyhp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jxprhawyhp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jxprhawyhp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jxprhawyhp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jxprhawyhp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jxprhawyhp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jxprhawyhp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jxprhawyhp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jxprhawyhp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jxprhawyhp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jxprhawyhp .gt_spanner_row {
  border-bottom-style: hidden;
}

#jxprhawyhp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jxprhawyhp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jxprhawyhp .gt_from_md > :first-child {
  margin-top: 0;
}

#jxprhawyhp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jxprhawyhp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jxprhawyhp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jxprhawyhp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jxprhawyhp .gt_row_group_first td {
  border-top-width: 2px;
}

#jxprhawyhp .gt_row_group_first th {
  border-top-width: 2px;
}

#jxprhawyhp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxprhawyhp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jxprhawyhp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jxprhawyhp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jxprhawyhp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxprhawyhp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jxprhawyhp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jxprhawyhp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jxprhawyhp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jxprhawyhp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jxprhawyhp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxprhawyhp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jxprhawyhp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxprhawyhp .gt_left {
  text-align: left;
}

#jxprhawyhp .gt_center {
  text-align: center;
}

#jxprhawyhp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jxprhawyhp .gt_font_normal {
  font-weight: normal;
}

#jxprhawyhp .gt_font_bold {
  font-weight: bold;
}

#jxprhawyhp .gt_font_italic {
  font-style: italic;
}

#jxprhawyhp .gt_super {
  font-size: 65%;
}

#jxprhawyhp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jxprhawyhp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jxprhawyhp .gt_indent_1 {
  text-indent: 5px;
}

#jxprhawyhp .gt_indent_2 {
  text-indent: 10px;
}

#jxprhawyhp .gt_indent_3 {
  text-indent: 15px;
}

#jxprhawyhp .gt_indent_4 {
  text-indent: 20px;
}

#jxprhawyhp .gt_indent_5 {
  text-indent: 25px;
}

#jxprhawyhp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#jxprhawyhp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">Interaction-LRT</td>
<td headers="enrichment" class="gt_row gt_left">KO</td>
<td headers="n" class="gt_row gt_right">11</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">Interaction-LRT</td>
<td headers="enrichment" class="gt_row gt_left">WT</td>
<td headers="n" class="gt_row gt_right">2</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">Interaction-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">13663</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Comparison%20of%20Wald%20test%20and%20LRT%20-%20interaction%20effects-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">DEGs obtained with Wald test <a href="#main-effects-due-to-genotype">Main effects due to genotype</a></figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>DEGs are common irrespective of Wald or LRT test used to obtain interaction effects.</p>
</section>
<section id="visualization-of-interaction-specific-degs-using-deseq2" class="level4">
<h4 class="anchored" data-anchor-id="visualization-of-interaction-specific-degs-using-deseq2">Visualization of interaction-specific DEGs using DESeq2</h4>
<p>I am going with res6 object that uses Wald test to identify main effects due to genotype. Wald test DEGs are all represented in the LRT DEGs as determined by the previous Upset plot</p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res6,</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res6<span class="sc">$</span>gene,</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"comparison [Interaction effect]"</span>,</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res6<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20interaction-specific%20DEGs-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20interaction-specific%20DEGs-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Genotype-specific DEGs agnostic of the sampling timepoint</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>The timepoint based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</p>
</section>
<section id="expression-pattern-of-top-interaction-specific-degs" class="level4">
<h4 class="anchored" data-anchor-id="expression-pattern-of-top-interaction-specific-degs">Expression pattern of top interaction-specific DEGs</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res6 <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">time_point =</span> <span class="fu">c</span>(<span class="st">"T0"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"T12"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span>,</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>)</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>        )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20interaction-specific%20DEGs-1.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res6 <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span> </span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20interaction-specific%20DEGs-2.png" class="img-fluid"></p>
</div>
</div>
</section>
</section>
</section>
<section id="paired-contrast-analysis---difference-between-wt-and-ko-at-each-timepoint" class="level2">
<h2 class="anchored" data-anchor-id="paired-contrast-analysis---difference-between-wt-and-ko-at-each-timepoint">Paired contrast analysis - Difference between WT and KO at each timepoint</h2>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">paste0</span>(coldata<span class="sc">$</span>genotype, <span class="st">"_"</span>, coldata<span class="sc">$</span>time_point)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> group)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Comparison between KO and WT at time T0</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>res9 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"KO_T0"</span>, <span class="st">"WT_T0"</span>))</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>res9 <span class="ot">&lt;-</span> res9 <span class="sc">%&gt;%</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Time0-LRT"</span>,</span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO_T0"</span>,</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT_T0"</span>,</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>res9 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Comparison between KO and WT at time T12</span></span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>res10 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"KO_T12"</span>, <span class="st">"WT_T12"</span>))</span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>res10 <span class="ot">&lt;-</span> res10 <span class="sc">%&gt;%</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Time12-LRT"</span>,</span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO_T12"</span>,</span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT_T12"</span>,</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>res10 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res9,res10)</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="datriybddk" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#datriybddk table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#datriybddk thead, #datriybddk tbody, #datriybddk tfoot, #datriybddk tr, #datriybddk td, #datriybddk th {
  border-style: none;
}

#datriybddk p {
  margin: 0;
  padding: 0;
}

#datriybddk .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#datriybddk .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#datriybddk .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#datriybddk .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#datriybddk .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#datriybddk .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#datriybddk .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#datriybddk .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#datriybddk .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#datriybddk .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#datriybddk .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#datriybddk .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#datriybddk .gt_spanner_row {
  border-bottom-style: hidden;
}

#datriybddk .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#datriybddk .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#datriybddk .gt_from_md > :first-child {
  margin-top: 0;
}

#datriybddk .gt_from_md > :last-child {
  margin-bottom: 0;
}

#datriybddk .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#datriybddk .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#datriybddk .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#datriybddk .gt_row_group_first td {
  border-top-width: 2px;
}

#datriybddk .gt_row_group_first th {
  border-top-width: 2px;
}

#datriybddk .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#datriybddk .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#datriybddk .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#datriybddk .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#datriybddk .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#datriybddk .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#datriybddk .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#datriybddk .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#datriybddk .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#datriybddk .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#datriybddk .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#datriybddk .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#datriybddk .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#datriybddk .gt_left {
  text-align: left;
}

#datriybddk .gt_center {
  text-align: center;
}

#datriybddk .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#datriybddk .gt_font_normal {
  font-weight: normal;
}

#datriybddk .gt_font_bold {
  font-weight: bold;
}

#datriybddk .gt_font_italic {
  font-style: italic;
}

#datriybddk .gt_super {
  font-size: 65%;
}

#datriybddk .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#datriybddk .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#datriybddk .gt_indent_1 {
  text-indent: 5px;
}

#datriybddk .gt_indent_2 {
  text-indent: 10px;
}

#datriybddk .gt_indent_3 {
  text-indent: 15px;
}

#datriybddk .gt_indent_4 {
  text-indent: 20px;
}

#datriybddk .gt_indent_5 {
  text-indent: 25px;
}

#datriybddk .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#datriybddk div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Time0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">KO_T0</td>
<td headers="n" class="gt_row gt_right">165</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Time0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">WT_T0</td>
<td headers="n" class="gt_row gt_right">165</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Time0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">16618</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01) by comparing KO versus WT at time T0</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="fpjqgydkgp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#fpjqgydkgp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#fpjqgydkgp thead, #fpjqgydkgp tbody, #fpjqgydkgp tfoot, #fpjqgydkgp tr, #fpjqgydkgp td, #fpjqgydkgp th {
  border-style: none;
}

#fpjqgydkgp p {
  margin: 0;
  padding: 0;
}

#fpjqgydkgp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#fpjqgydkgp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#fpjqgydkgp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#fpjqgydkgp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#fpjqgydkgp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fpjqgydkgp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fpjqgydkgp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fpjqgydkgp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#fpjqgydkgp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#fpjqgydkgp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#fpjqgydkgp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#fpjqgydkgp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#fpjqgydkgp .gt_spanner_row {
  border-bottom-style: hidden;
}

#fpjqgydkgp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#fpjqgydkgp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#fpjqgydkgp .gt_from_md > :first-child {
  margin-top: 0;
}

#fpjqgydkgp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#fpjqgydkgp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#fpjqgydkgp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#fpjqgydkgp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#fpjqgydkgp .gt_row_group_first td {
  border-top-width: 2px;
}

#fpjqgydkgp .gt_row_group_first th {
  border-top-width: 2px;
}

#fpjqgydkgp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fpjqgydkgp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#fpjqgydkgp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#fpjqgydkgp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fpjqgydkgp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fpjqgydkgp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#fpjqgydkgp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#fpjqgydkgp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#fpjqgydkgp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fpjqgydkgp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fpjqgydkgp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fpjqgydkgp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fpjqgydkgp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fpjqgydkgp .gt_left {
  text-align: left;
}

#fpjqgydkgp .gt_center {
  text-align: center;
}

#fpjqgydkgp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#fpjqgydkgp .gt_font_normal {
  font-weight: normal;
}

#fpjqgydkgp .gt_font_bold {
  font-weight: bold;
}

#fpjqgydkgp .gt_font_italic {
  font-style: italic;
}

#fpjqgydkgp .gt_super {
  font-size: 65%;
}

#fpjqgydkgp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#fpjqgydkgp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#fpjqgydkgp .gt_indent_1 {
  text-indent: 5px;
}

#fpjqgydkgp .gt_indent_2 {
  text-indent: 10px;
}

#fpjqgydkgp .gt_indent_3 {
  text-indent: 15px;
}

#fpjqgydkgp .gt_indent_4 {
  text-indent: 20px;
}

#fpjqgydkgp .gt_indent_5 {
  text-indent: 25px;
}

#fpjqgydkgp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#fpjqgydkgp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Time12-LRT</td>
<td headers="enrichment" class="gt_row gt_left">KO_T12</td>
<td headers="n" class="gt_row gt_right">124</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Time12-LRT</td>
<td headers="enrichment" class="gt_row gt_left">WT_T12</td>
<td headers="n" class="gt_row gt_right">147</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Time12-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">16677</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01) by comparing KO versus WT at time T0</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20DEGs%20between%20WT%20and%20KO%20for%20every%20timepoint-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01) by comparing KO versus WT at time T0</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<section id="pathway-analysis-for-genotype-enriched-degs-for-timepoint-t0" class="level4">
<h4 class="anchored" data-anchor-id="pathway-analysis-for-genotype-enriched-degs-for-timepoint-t0">Pathway analysis for genotype-enriched DEGs for timepoint T0</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res9 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"KO"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res9))<span class="sc">$</span>result</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"KO"</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res9 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res9))<span class="sc">$</span>result</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"WT"</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway enrichment stats across multiple databases and genotype conditions"</span></span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Significant pathways (labelled in blue) across timepoints when using ORA and multiple pathway databases"</span></span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(df1)) { df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>) }</span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(df1)) {</span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="pathway-analysis-for-genotype-enriched-degs-for-timepoint-t12" class="level4">
<h4 class="anchored" data-anchor-id="pathway-analysis-for-genotype-enriched-degs-for-timepoint-t12">Pathway analysis for genotype-enriched DEGs for timepoint T12</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res10 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"KO"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res10))<span class="sc">$</span>result</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"KO"</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res10 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res10))<span class="sc">$</span>result</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"WT"</span></span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb35-43"><a href="#cb35-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-44"><a href="#cb35-44" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb35-45"><a href="#cb35-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-46"><a href="#cb35-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-47"><a href="#cb35-47" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb35-48"><a href="#cb35-48" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb35-49"><a href="#cb35-49" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway enrichment stats across multiple databases and genotype conditions"</span></span>
<span id="cb35-50"><a href="#cb35-50" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Significant pathways (labelled in blue) across timepoints when using ORA and multiple pathway databases"</span></span>
<span id="cb35-51"><a href="#cb35-51" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(df1)) { df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>) }</span>
<span id="cb35-52"><a href="#cb35-52" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(df1)) {</span>
<span id="cb35-53"><a href="#cb35-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb35-54"><a href="#cb35-54" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb35-55"><a href="#cb35-55" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="heirarchical-clustering-of-all-the-degs" class="level2">
<h2 class="anchored" data-anchor-id="heirarchical-clustering-of-all-the-degs">Heirarchical clustering of all the DEGs</h2>
<p>Main effect DEGs are in res1 (Genotype), res4 (Timepoint) objects whereas interaction specific DEGs are in res6 objects. res9 and res10 objects contain DEGs between KO and WT comparison for timepoints T0 and T12 respectively.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(res1 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res4 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res6 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res9 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res10 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)))</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">genotype =</span> <span class="fu">c</span>(<span class="st">"WT"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"KO"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> F,</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> T,</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> T,</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_km =</span> <span class="dv">2</span>,</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_km =</span> <span class="dv">4</span>,</span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>),</span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>        )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid"></p>
</div>
</div>
<p>From the heatmap, we see that clustering of samples based on the genotype then the timepoint. Genotype effects drive major source of variation compared to the timepoint. I roughly divided the DEGs obtained from various comparisons into 4 clusters - Cluster 2 has genes active in T0 in both genotypes Cluster 1 has genes active in KO genotype in both timepoints Cluster 4 has genes active in WT genotype in both timepoints Cluster 3 has genes active in T12 in both genotypes</p>
</section>
<section id="plot-the-expression-pattern-of-the-degs-obtained-from-all-the-above-comparisons" class="level2">
<h2 class="anchored" data-anchor-id="plot-the-expression-pattern-of-the-degs-obtained-from-all-the-above-comparisons">Plot the expression pattern of the DEGs obtained from all the above comparisons</h2>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>ma <span class="ot">=</span> <span class="fu">assay</span>(<span class="fu">vst</span>(dds))[significant_genes,]</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>coldata1 <span class="ot">=</span> coldata <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">paste0</span>(genotype,<span class="st">"_"</span>, time_point),</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">group =</span> <span class="fu">factor</span>(group))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">degPatterns</span>(ma, </span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">metadata =</span> coldata1, </span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">time =</span> <span class="st">"group"</span>, <span class="at">plot =</span> T)</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>clusters<span class="sc">$</span>df <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Grouping all the DEGs into clusters based on expression pattern trends</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="fwghqtcfnn" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#fwghqtcfnn table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#fwghqtcfnn thead, #fwghqtcfnn tbody, #fwghqtcfnn tfoot, #fwghqtcfnn tr, #fwghqtcfnn td, #fwghqtcfnn th {
  border-style: none;
}

#fwghqtcfnn p {
  margin: 0;
  padding: 0;
}

#fwghqtcfnn .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#fwghqtcfnn .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#fwghqtcfnn .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#fwghqtcfnn .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#fwghqtcfnn .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fwghqtcfnn .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fwghqtcfnn .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fwghqtcfnn .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#fwghqtcfnn .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#fwghqtcfnn .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#fwghqtcfnn .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#fwghqtcfnn .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#fwghqtcfnn .gt_spanner_row {
  border-bottom-style: hidden;
}

#fwghqtcfnn .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#fwghqtcfnn .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#fwghqtcfnn .gt_from_md > :first-child {
  margin-top: 0;
}

#fwghqtcfnn .gt_from_md > :last-child {
  margin-bottom: 0;
}

#fwghqtcfnn .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#fwghqtcfnn .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#fwghqtcfnn .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#fwghqtcfnn .gt_row_group_first td {
  border-top-width: 2px;
}

#fwghqtcfnn .gt_row_group_first th {
  border-top-width: 2px;
}

#fwghqtcfnn .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwghqtcfnn .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#fwghqtcfnn .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#fwghqtcfnn .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fwghqtcfnn .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwghqtcfnn .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#fwghqtcfnn .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#fwghqtcfnn .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#fwghqtcfnn .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fwghqtcfnn .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fwghqtcfnn .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwghqtcfnn .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fwghqtcfnn .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#fwghqtcfnn .gt_left {
  text-align: left;
}

#fwghqtcfnn .gt_center {
  text-align: center;
}

#fwghqtcfnn .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#fwghqtcfnn .gt_font_normal {
  font-weight: normal;
}

#fwghqtcfnn .gt_font_bold {
  font-weight: bold;
}

#fwghqtcfnn .gt_font_italic {
  font-style: italic;
}

#fwghqtcfnn .gt_super {
  font-size: 65%;
}

#fwghqtcfnn .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#fwghqtcfnn .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#fwghqtcfnn .gt_indent_1 {
  text-indent: 5px;
}

#fwghqtcfnn .gt_indent_2 {
  text-indent: 10px;
}

#fwghqtcfnn .gt_indent_3 {
  text-indent: 15px;
}

#fwghqtcfnn .gt_indent_4 {
  text-indent: 20px;
}

#fwghqtcfnn .gt_indent_5 {
  text-indent: 25px;
}

#fwghqtcfnn .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#fwghqtcfnn div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="a::stub"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="genes">genes</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="cluster">cluster</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><th id="stub_1_1" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">1</th>
<td headers="stub_1_1 genes" class="gt_row gt_left">ENSMUSG00000055214</td>
<td headers="stub_1_1 cluster" class="gt_row gt_right">1</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">2</th>
<td headers="stub_1_2 genes" class="gt_row gt_left">ENSMUSG00000047343</td>
<td headers="stub_1_2 cluster" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">3</th>
<td headers="stub_1_3 genes" class="gt_row gt_left">ENSMUSG00000101746</td>
<td headers="stub_1_3 cluster" class="gt_row gt_right">3</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">4</th>
<td headers="stub_1_4 genes" class="gt_row gt_left">ENSMUSG00000032883</td>
<td headers="stub_1_4 cluster" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">5</th>
<td headers="stub_1_5 genes" class="gt_row gt_left">ENSMUSG00000026494</td>
<td headers="stub_1_5 cluster" class="gt_row gt_right">4</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">6</th>
<td headers="stub_1_6 genes" class="gt_row gt_left">ENSMUSG00000100502</td>
<td headers="stub_1_6 cluster" class="gt_row gt_right">3</td></tr>
    <tr><th id="stub_1_7" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">7</th>
<td headers="stub_1_7 genes" class="gt_row gt_left">ENSMUSG00000032769</td>
<td headers="stub_1_7 cluster" class="gt_row gt_right">3</td></tr>
    <tr><th id="stub_1_8" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">8</th>
<td headers="stub_1_8 genes" class="gt_row gt_left">ENSMUSG00000026387</td>
<td headers="stub_1_8 cluster" class="gt_row gt_right">4</td></tr>
    <tr><th id="stub_1_9" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">9</th>
<td headers="stub_1_9 genes" class="gt_row gt_left">ENSMUSG00000026224</td>
<td headers="stub_1_9 cluster" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_10" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">10</th>
<td headers="stub_1_10 genes" class="gt_row gt_left">ENSMUSG00000026253</td>
<td headers="stub_1_10 cluster" class="gt_row gt_right">3</td></tr>
    <tr><th id="stub_1_11" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier; font-size: x-small; background-color: #E4E4E4;">11..767</th>
<td headers="stub_1_11 genes" class="gt_row gt_left" style="background-color: #E4E4E4;"></td>
<td headers="stub_1_11 cluster" class="gt_row gt_right" style="background-color: #E4E4E4;"></td></tr>
    <tr><th id="stub_1_12" scope="row" class="gt_row gt_right gt_stub" style="font-family: Courier;">768</th>
<td headers="stub_1_12 genes" class="gt_row gt_left">ENSMUSG00000045689</td>
<td headers="stub_1_12 cluster" class="gt_row gt_right">2</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>Grouping all the DEGs into clusters based on expression pattern trends</p>
</div>
</div>
</div>
</div>
<p>The output is the list of genes along with cluster IDs that facilitate filtering of genes with a particular expression pattern trend.</p>
<section id="pathway-analysis-of-various-clusters-obtained-from-the-above-analysis" class="level3">
<h3 class="anchored" data-anchor-id="pathway-analysis-of-various-clusters-obtained-from-the-above-analysis">Pathway analysis of various clusters obtained from the above analysis</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>hg_gs <span class="ot">=</span> geneset<span class="sc">::</span><span class="fu">getMsigdb</span>(<span class="at">org =</span> <span class="st">"human"</span>, <span class="at">category =</span> <span class="st">"H"</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">unique</span>(clusters<span class="sc">$</span>df<span class="sc">$</span>cluster), <span class="cf">function</span>(x) {</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(clusters<span class="sc">$</span>df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cluster <span class="sc">==</span> x) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes),</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> hg_gs,</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#p_cutoff = 1, #to retrieve the whole output</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#q_cutoff = 1, #to retrieve the whole output</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> clusters<span class="sc">$</span>df <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes))<span class="sc">$</span>result</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a><span class="fu">list_rbind</span>(df1, <span class="at">names_to =</span> <span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>data frame with 0 columns and 0 rows</code></pre>
</div>
</div>
<p>No pathways associated with each of the cluster genes.</p>
<!-- -->

</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb40" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "SampleAnalysis_Mansingh2024_matrix"</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 4</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co">    eval: true</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co">    warning: false</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="co">    message: false</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-height: 7</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-width: 7</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: png</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-dpi: 300</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a><span class="co">  pdf: </span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 4</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="co">    eval: true</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="co">    warning: false</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="co">    message: false</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-height: 7</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-width: 7</span></span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: png</span></span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-dpi: 300</span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>**Aim:** To perform differential gene expression analysis for different contrasts</span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a><span class="fu">## Loading the required libraries and importing required files</span></span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a><span class="fu">### Loading the required libraries</span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Loading libraries}</span></span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co">#for data wrangling</span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2) <span class="co">#for DEG analysis</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genekitr) <span class="co">#for pathway analysis</span></span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr) <span class="co">#for data import</span></span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(eatTools) <span class="co">#for rbind functions</span></span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(strex) <span class="co">#for string related functions</span></span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr) <span class="co">#for beautiful tables</span></span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt) <span class="co">#for customized table layouts</span></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap) <span class="co">#for beautiful heatmaps</span></span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyheatmaps) <span class="co">#for heatmap generation from dataframe</span></span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co">#for adding multiple figures to a plot</span></span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer) <span class="co">#for color schemas</span></span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) <span class="co">#for plotting</span></span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix) <span class="co">#to explore complicated design in the DESeq2 formula</span></span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR) <span class="co">#for upset plots</span></span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DEGreport) <span class="co">#for clustering of genes into groups based on expression pattern</span></span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a><span class="fu">### Importing the counts matrix</span></span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a>The counts file were imported from the publication <span class="ot">&lt;https://physoc.onlinelibrary.wiley.com/doi/10.1113/JP285585&gt;</span>. The authors investigated the outcome of perturbations of the primary and secondary feedback loops of the molecular clock in skeletal muscle by specific gene ablation of Period circadian regulator 2 (Per2) and RAR-related orphan receptor alpha (Rorα), respectively. </span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Importing the counts and conversion to a DESeq object}</span></span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a><span class="co">#loading the length-scaled counts tsv file</span></span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"~/Documents/BiostateAI/data/Mansingh2024_expression_matrix.txt"</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span>T) <span class="co">#also attached in the Github folder</span></span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a><span class="co">#converting NA into 0 and round off the counts</span></span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a>cts[<span class="fu">is.na</span>(cts)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb40-70"><a href="#cb40-70" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.numeric, <span class="fu">list</span>(round))</span>
<span id="cb40-71"><a href="#cb40-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a><span class="fu">### Loading the metadata file</span></span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Import metadata file}</span></span>
<span id="cb40-76"><a href="#cb40-76" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">str_detect</span>( <span class="fu">colnames</span>(cts) , <span class="st">'CTRL'</span> )  )</span>
<span id="cb40-77"><a href="#cb40-77" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">revalue</span>( genotype , <span class="fu">c</span>( <span class="st">'TRUE'</span><span class="ot">=</span><span class="st">'WT'</span> , <span class="st">'FALSE'</span> <span class="ot">=</span> <span class="st">'KO'</span>  )  ) <span class="co">#TRUE is WT, FALSE is CTRL</span></span>
<span id="cb40-78"><a href="#cb40-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-79"><a href="#cb40-79" aria-hidden="true" tabindex="-1"></a>time_point <span class="ot">=</span> <span class="fu">factor</span>( <span class="fu">substring</span>(<span class="fu">colnames</span>(cts), <span class="fu">regexpr</span>(<span class="st">"_Z"</span>,<span class="fu">colnames</span>(cts)) <span class="sc">+</span> <span class="dv">2</span>) , <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>,<span class="st">'T4'</span>,<span class="st">'T8'</span>,<span class="st">'T12'</span>,<span class="st">'T16'</span>,<span class="st">'T20'</span>))</span>
<span id="cb40-80"><a href="#cb40-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-81"><a href="#cb40-81" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> <span class="fu">sapply</span>( <span class="fu">strsplit</span>( <span class="fu">colnames</span>( cts ) , <span class="st">"_"</span>) , <span class="st">'['</span> , <span class="dv">5</span>)</span>
<span id="cb40-82"><a href="#cb40-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-83"><a href="#cb40-83" aria-hidden="true" tabindex="-1"></a>replicate <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(cts) , <span class="dv">1</span>, <span class="fu">nchar</span>( <span class="st">"HL3YYBGX5"</span> ) ) </span>
<span id="cb40-84"><a href="#cb40-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-85"><a href="#cb40-85" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( <span class="at">sample =</span> <span class="fu">colnames</span>(cts) ,</span>
<span id="cb40-86"><a href="#cb40-86" aria-hidden="true" tabindex="-1"></a>                       <span class="at">genotype =</span> genotype ,</span>
<span id="cb40-87"><a href="#cb40-87" aria-hidden="true" tabindex="-1"></a>                       <span class="at">time_point =</span> time_point,</span>
<span id="cb40-88"><a href="#cb40-88" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mouse =</span> mouse,</span>
<span id="cb40-89"><a href="#cb40-89" aria-hidden="true" tabindex="-1"></a>                       <span class="at">replicate =</span> replicate</span>
<span id="cb40-90"><a href="#cb40-90" aria-hidden="true" tabindex="-1"></a>                       )</span>
<span id="cb40-91"><a href="#cb40-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-92"><a href="#cb40-92" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> coldata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(time_point <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb40-93"><a href="#cb40-93" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>time_point <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>time_point, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb40-94"><a href="#cb40-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-95"><a href="#cb40-95" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">select</span>(coldata<span class="sc">$</span>sample)</span>
<span id="cb40-96"><a href="#cb40-96" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb40-97"><a href="#cb40-97" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-98"><a href="#cb40-98" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Metadata file"</span></span>
<span id="cb40-99"><a href="#cb40-99" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Sample counts for genotype and timepoint combinations"</span></span>
<span id="cb40-100"><a href="#cb40-100" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-101"><a href="#cb40-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-102"><a href="#cb40-102" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(genotype,time_point) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-103"><a href="#cb40-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-104"><a href="#cb40-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-105"><a href="#cb40-105" aria-hidden="true" tabindex="-1"></a><span class="fu">## Creation and exploration of dds object</span></span>
<span id="cb40-106"><a href="#cb40-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-107"><a href="#cb40-107" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Conversion to a DESeq object}</span></span>
<span id="cb40-108"><a href="#cb40-108" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-109"><a href="#cb40-109" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-110"><a href="#cb40-110" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "p-value histogram without prefiltering of counts"</span></span>
<span id="cb40-111"><a href="#cb40-111" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genes with high p-values do seem to have lower gene expression"</span></span>
<span id="cb40-112"><a href="#cb40-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-113"><a href="#cb40-113" aria-hidden="true" tabindex="-1"></a><span class="co">#creation of dds object without prefiltering</span></span>
<span id="cb40-114"><a href="#cb40-114" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-115"><a href="#cb40-115" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-116"><a href="#cb40-116" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#paired design tissue + time if need to adjust for confounders (time in this case)</span></span>
<span id="cb40-117"><a href="#cb40-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-118"><a href="#cb40-118" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(<span class="at">sampleData =</span> coldata,</span>
<span id="cb40-119"><a href="#cb40-119" aria-hidden="true" tabindex="-1"></a>                      <span class="at">designFormula =</span> <span class="sc">~</span> genotype)</span>
<span id="cb40-120"><a href="#cb40-120" aria-hidden="true" tabindex="-1"></a><span class="co">#print(vd$plotlist) #can visualize the model matrix and the various coefficients</span></span>
<span id="cb40-121"><a href="#cb40-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-122"><a href="#cb40-122" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">relevel</span>(dds<span class="sc">$</span>genotype, <span class="at">ref =</span> <span class="st">"WT"</span>)</span>
<span id="cb40-123"><a href="#cb40-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-124"><a href="#cb40-124" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds) <span class="co">#Main effects for genotype #Expression affected by genotype not correcting for  the timepoint effects</span></span>
<span id="cb40-125"><a href="#cb40-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-126"><a href="#cb40-126" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>)) <span class="co">#FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb40-127"><a href="#cb40-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-128"><a href="#cb40-128" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-129"><a href="#cb40-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-130"><a href="#cb40-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb40-131"><a href="#cb40-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb40-132"><a href="#cb40-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram without prefiltering of counts"</span>)</span>
<span id="cb40-133"><a href="#cb40-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-134"><a href="#cb40-134" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-135"><a href="#cb40-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-136"><a href="#cb40-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb40-137"><a href="#cb40-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-138"><a href="#cb40-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb40-139"><a href="#cb40-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb40-140"><a href="#cb40-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-141"><a href="#cb40-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-142"><a href="#cb40-142" aria-hidden="true" tabindex="-1"></a>p-value histogram distribution looks bimodal. High expressing count genes fall into low p-value bin which is great. Let's look at the pattern after prefiltering of the counts.</span>
<span id="cb40-143"><a href="#cb40-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-144"><a href="#cb40-144" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Prefiltering of counts using the dds object}</span></span>
<span id="cb40-145"><a href="#cb40-145" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;</span> <span class="dv">10</span>,)] <span class="co">#opted out after the visualization</span></span>
<span id="cb40-146"><a href="#cb40-146" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb40-147"><a href="#cb40-147" aria-hidden="true" tabindex="-1"></a><span class="co">#resultsNames(dds)</span></span>
<span id="cb40-148"><a href="#cb40-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-149"><a href="#cb40-149" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>)) <span class="co">#FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb40-150"><a href="#cb40-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-151"><a href="#cb40-151" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-152"><a href="#cb40-152" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-153"><a href="#cb40-153" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "p-value histogram after prefiltering of counts"</span></span>
<span id="cb40-154"><a href="#cb40-154" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genes with high p-values do seem to have lower gene expression"</span></span>
<span id="cb40-155"><a href="#cb40-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-156"><a href="#cb40-156" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-157"><a href="#cb40-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-158"><a href="#cb40-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb40-159"><a href="#cb40-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb40-160"><a href="#cb40-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram after prefiltering of counts"</span>)</span>
<span id="cb40-161"><a href="#cb40-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-162"><a href="#cb40-162" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-163"><a href="#cb40-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-164"><a href="#cb40-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb40-165"><a href="#cb40-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-166"><a href="#cb40-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb40-167"><a href="#cb40-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb40-168"><a href="#cb40-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-169"><a href="#cb40-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-170"><a href="#cb40-170" aria-hidden="true" tabindex="-1"></a>Various contrasts can be identified using **resultNames(dds)** function</span>
<span id="cb40-171"><a href="#cb40-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-172"><a href="#cb40-172" aria-hidden="true" tabindex="-1"></a>The p-value histogram after prefiltering of counts is not looking good. Therefore, I haven't done prefiltering. Supposedly, DESeq2 takes care of the low counts while identifying DEGs. Refer this link <span class="ot">&lt;http://varianceexplained.org/statistics/interpreting-pvalue-histogram/&gt;</span> for more info on p-value histograms and their interpretations.</span>
<span id="cb40-173"><a href="#cb40-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-174"><a href="#cb40-174" aria-hidden="true" tabindex="-1"></a><span class="in">```{r creating a dds object without prefiltering again}</span></span>
<span id="cb40-175"><a href="#cb40-175" aria-hidden="true" tabindex="-1"></a><span class="co">#creation of dds object without prefiltering</span></span>
<span id="cb40-176"><a href="#cb40-176" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-177"><a href="#cb40-177" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-178"><a href="#cb40-178" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#paired design tissue + time if need to adjust for confounders (time in this case)</span></span>
<span id="cb40-179"><a href="#cb40-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-180"><a href="#cb40-180" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb40-181"><a href="#cb40-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-182"><a href="#cb40-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-183"><a href="#cb40-183" aria-hidden="true" tabindex="-1"></a><span class="fu">### PCA plot</span></span>
<span id="cb40-184"><a href="#cb40-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-185"><a href="#cb40-185" aria-hidden="true" tabindex="-1"></a><span class="in">```{r PCA plot}</span></span>
<span id="cb40-186"><a href="#cb40-186" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-187"><a href="#cb40-187" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-188"><a href="#cb40-188" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "PCA when colored by genotype and timepoint"</span></span>
<span id="cb40-189"><a href="#cb40-189" aria-hidden="true" tabindex="-1"></a>vstcounts <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> T)</span>
<span id="cb40-190"><a href="#cb40-190" aria-hidden="true" tabindex="-1"></a>normalized_cts <span class="ot">&lt;-</span> <span class="fu">assay</span>(vstcounts) <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb40-191"><a href="#cb40-191" aria-hidden="true" tabindex="-1"></a>pca_prcomp <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(normalized_cts), <span class="at">center =</span> T, <span class="at">scale =</span> F)</span>
<span id="cb40-192"><a href="#cb40-192" aria-hidden="true" tabindex="-1"></a><span class="co">#names(pca_prcomp)</span></span>
<span id="cb40-193"><a href="#cb40-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-194"><a href="#cb40-194" aria-hidden="true" tabindex="-1"></a>PC1_and_PC2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">1</span>],</span>
<span id="cb40-195"><a href="#cb40-195" aria-hidden="true" tabindex="-1"></a>                          <span class="at">PC2 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb40-196"><a href="#cb40-196" aria-hidden="true" tabindex="-1"></a>                          <span class="at">type =</span> <span class="fu">rownames</span>(pca_prcomp<span class="sc">$</span>x))</span>
<span id="cb40-197"><a href="#cb40-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-198"><a href="#cb40-198" aria-hidden="true" tabindex="-1"></a><span class="co">#using the complete normalized matrix</span></span>
<span id="cb40-199"><a href="#cb40-199" aria-hidden="true" tabindex="-1"></a><span class="co">#ggplot(PC1_and_PC2, aes(x = PC1, y = PC2, col=type)) +</span></span>
<span id="cb40-200"><a href="#cb40-200" aria-hidden="true" tabindex="-1"></a> <span class="co"># geom_point() +</span></span>
<span id="cb40-201"><a href="#cb40-201" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(aes(label = type), hjust = 0, vjust = 0) +</span></span>
<span id="cb40-202"><a href="#cb40-202" aria-hidden="true" tabindex="-1"></a>  <span class="co">#coord_fixed()</span></span>
<span id="cb40-203"><a href="#cb40-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-204"><a href="#cb40-204" aria-hidden="true" tabindex="-1"></a><span class="co">#using variable features</span></span>
<span id="cb40-205"><a href="#cb40-205" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vstcounts, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>))</span>
<span id="cb40-206"><a href="#cb40-206" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-207"><a href="#cb40-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-208"><a href="#cb40-208" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample correlation heatmap using vst counts</span></span>
<span id="cb40-209"><a href="#cb40-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-210"><a href="#cb40-210" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Sample correlation heatmap using vst counts}</span></span>
<span id="cb40-211"><a href="#cb40-211" aria-hidden="true" tabindex="-1"></a>sampleCor <span class="ot">&lt;-</span> <span class="fu">cor</span>(normalized_cts)</span>
<span id="cb40-212"><a href="#cb40-212" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(normalized_cts))</span>
<span id="cb40-213"><a href="#cb40-213" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDist)</span>
<span id="cb40-214"><a href="#cb40-214" aria-hidden="true" tabindex="-1"></a>blueColours <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Blues"</span>)</span>
<span id="cb40-215"><a href="#cb40-215" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(blueColours))(<span class="dv">255</span>)</span>
<span id="cb40-216"><a href="#cb40-216" aria-hidden="true" tabindex="-1"></a>annotation_row <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb40-217"><a href="#cb40-217" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Group =</span> <span class="fu">str_after_nth</span>(<span class="fu">colnames</span>(sampleDistMatrix), <span class="at">n =</span> <span class="dv">5</span>, <span class="st">"_"</span>)</span>
<span id="cb40-218"><a href="#cb40-218" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb40-219"><a href="#cb40-219" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation_row) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sampleDistMatrix)</span>
<span id="cb40-220"><a href="#cb40-220" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix,</span>
<span id="cb40-221"><a href="#cb40-221" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clustering_distance_cols =</span> sampleDist, </span>
<span id="cb40-222"><a href="#cb40-222" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> colors,</span>
<span id="cb40-223"><a href="#cb40-223" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fontsize_row =</span> <span class="dv">6</span>,</span>
<span id="cb40-224"><a href="#cb40-224" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fontsize_col =</span> <span class="dv">8</span>,</span>
<span id="cb40-225"><a href="#cb40-225" aria-hidden="true" tabindex="-1"></a>                   <span class="at">annotation_row =</span> annotation_row,</span>
<span id="cb40-226"><a href="#cb40-226" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cutree_rows =</span> <span class="dv">4</span></span>
<span id="cb40-227"><a href="#cb40-227" aria-hidden="true" tabindex="-1"></a>                   )</span>
<span id="cb40-228"><a href="#cb40-228" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-229"><a href="#cb40-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-230"><a href="#cb40-230" aria-hidden="true" tabindex="-1"></a>All the timepoint and genotype replicates cluster in 4 groups corresponding to WT, KO and T0,12 combinations.</span>
<span id="cb40-231"><a href="#cb40-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-232"><a href="#cb40-232" aria-hidden="true" tabindex="-1"></a><span class="fu">## Differential gene expression analysis</span></span>
<span id="cb40-233"><a href="#cb40-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-234"><a href="#cb40-234" aria-hidden="true" tabindex="-1"></a><span class="fu">### Main effects due to genotype</span></span>
<span id="cb40-235"><a href="#cb40-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-236"><a href="#cb40-236" aria-hidden="true" tabindex="-1"></a>Here, I am identifying genotype-specific (KO versus WT) DEGs. These DEGs would be KO- or WT-enriched.</span>
<span id="cb40-237"><a href="#cb40-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-238"><a href="#cb40-238" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Comparison of Wald versus LRT test DEGs [Main effects due to genotype]</span></span>
<span id="cb40-239"><a href="#cb40-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-240"><a href="#cb40-240" aria-hidden="true" tabindex="-1"></a>**DEGs can be defined genes passing the threshold: abs(logFC) \&gt; 1 &amp; padj \&lt; 0.01**</span>
<span id="cb40-241"><a href="#cb40-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-242"><a href="#cb40-242" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Comparison of Wald test and LRT - genotype}</span></span>
<span id="cb40-243"><a href="#cb40-243" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-244"><a href="#cb40-244" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-245"><a href="#cb40-245" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with Wald test [Main effects due to genotype]"</span></span>
<span id="cb40-246"><a href="#cb40-246" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with paired design and with using LRT [Main effects due to genotype]"</span></span>
<span id="cb40-247"><a href="#cb40-247" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to genotype]"</span></span>
<span id="cb40-248"><a href="#cb40-248" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-249"><a href="#cb40-249" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-250"><a href="#cb40-250" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#not accounting for timepoint</span></span>
<span id="cb40-251"><a href="#cb40-251" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb40-252"><a href="#cb40-252" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb40-253"><a href="#cb40-253" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb40-254"><a href="#cb40-254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-255"><a href="#cb40-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-256"><a href="#cb40-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-257"><a href="#cb40-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-258"><a href="#cb40-258" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-259"><a href="#cb40-259" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-260"><a href="#cb40-260" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Wald"</span>,</span>
<span id="cb40-261"><a href="#cb40-261" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb40-262"><a href="#cb40-262" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb40-263"><a href="#cb40-263" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb40-264"><a href="#cb40-264" aria-hidden="true" tabindex="-1"></a>res1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-265"><a href="#cb40-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-266"><a href="#cb40-266" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb40-267"><a href="#cb40-267" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-268"><a href="#cb40-268" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-269"><a href="#cb40-269" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb40-270"><a href="#cb40-270" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> time_point)</span>
<span id="cb40-271"><a href="#cb40-271" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb40-272"><a href="#cb40-272" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> res2 <span class="sc">%&gt;%</span></span>
<span id="cb40-273"><a href="#cb40-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-274"><a href="#cb40-274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-275"><a href="#cb40-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-276"><a href="#cb40-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-277"><a href="#cb40-277" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-278"><a href="#cb40-278" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-279"><a href="#cb40-279" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-LRT"</span>,</span>
<span id="cb40-280"><a href="#cb40-280" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb40-281"><a href="#cb40-281" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb40-282"><a href="#cb40-282" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb40-283"><a href="#cb40-283" aria-hidden="true" tabindex="-1"></a>res2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-284"><a href="#cb40-284" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res1,res2)</span>
<span id="cb40-285"><a href="#cb40-285" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb40-286"><a href="#cb40-286" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb40-287"><a href="#cb40-287" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb40-288"><a href="#cb40-288" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb40-289"><a href="#cb40-289" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb40-290"><a href="#cb40-290" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb40-291"><a href="#cb40-291" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span>
<span id="cb40-292"><a href="#cb40-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-293"><a href="#cb40-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-294"><a href="#cb40-294" aria-hidden="true" tabindex="-1"></a>DEGs with and without accounting for timepoint yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to genotype.</span>
<span id="cb40-295"><a href="#cb40-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-296"><a href="#cb40-296" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Visualization of genotype-specific DEGs using DESeq2</span></span>
<span id="cb40-297"><a href="#cb40-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-298"><a href="#cb40-298" aria-hidden="true" tabindex="-1"></a>I am going with res1 object that uses Wald test to identify main effects due to genotype.</span>
<span id="cb40-299"><a href="#cb40-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-300"><a href="#cb40-300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Identification of genotype-specific DEGs}</span></span>
<span id="cb40-301"><a href="#cb40-301" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-302"><a href="#cb40-302" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-303"><a href="#cb40-303" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-304"><a href="#cb40-304" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Tissue-specific DEGs agnostic of the sampling timepoint"</span></span>
<span id="cb40-305"><a href="#cb40-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-306"><a href="#cb40-306" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb40-307"><a href="#cb40-307" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res1,</span>
<span id="cb40-308"><a href="#cb40-308" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb40-309"><a href="#cb40-309" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb40-310"><a href="#cb40-310" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res1<span class="sc">$</span>gene,</span>
<span id="cb40-311"><a href="#cb40-311" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb40-312"><a href="#cb40-312" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-313"><a href="#cb40-313" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"KO vs WT comparison [Single-factor design]"</span>,</span>
<span id="cb40-314"><a href="#cb40-314" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb40-315"><a href="#cb40-315" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb40-316"><a href="#cb40-316" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb40-317"><a href="#cb40-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-318"><a href="#cb40-318" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb40-319"><a href="#cb40-319" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res1<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb40-320"><a href="#cb40-320" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-321"><a href="#cb40-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb40-322"><a href="#cb40-322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-323"><a href="#cb40-323" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb40-324"><a href="#cb40-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb40-325"><a href="#cb40-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb40-326"><a href="#cb40-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb40-327"><a href="#cb40-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb40-328"><a href="#cb40-328" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-329"><a href="#cb40-329" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb40-330"><a href="#cb40-330" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb40-331"><a href="#cb40-331" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span>
<span id="cb40-332"><a href="#cb40-332" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-333"><a href="#cb40-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-334"><a href="#cb40-334" aria-hidden="true" tabindex="-1"></a>The genotype based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</span>
<span id="cb40-335"><a href="#cb40-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-336"><a href="#cb40-336" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Expression pattern of top genotype-specific DEGs</span></span>
<span id="cb40-337"><a href="#cb40-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-338"><a href="#cb40-338" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Expression pattern of top genotype-specific DEGs}</span></span>
<span id="cb40-339"><a href="#cb40-339" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb40-340"><a href="#cb40-340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-341"><a href="#cb40-341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-342"><a href="#cb40-342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-343"><a href="#cb40-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb40-344"><a href="#cb40-344" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb40-345"><a href="#cb40-345" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb40-346"><a href="#cb40-346" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">genotype =</span> <span class="fu">c</span>(<span class="st">"WT"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"KO"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb40-347"><a href="#cb40-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-348"><a href="#cb40-348" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-349"><a href="#cb40-349" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-350"><a href="#cb40-350" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-351"><a href="#cb40-351" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb40-352"><a href="#cb40-352" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb40-353"><a href="#cb40-353" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb40-354"><a href="#cb40-354" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb40-355"><a href="#cb40-355" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> T,</span>
<span id="cb40-356"><a href="#cb40-356" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> F,</span>
<span id="cb40-357"><a href="#cb40-357" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb40-358"><a href="#cb40-358" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>),</span>
<span id="cb40-359"><a href="#cb40-359" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span></span>
<span id="cb40-360"><a href="#cb40-360" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb40-361"><a href="#cb40-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-362"><a href="#cb40-362" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb40-363"><a href="#cb40-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-364"><a href="#cb40-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-365"><a href="#cb40-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-366"><a href="#cb40-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb40-367"><a href="#cb40-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-368"><a href="#cb40-368" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb40-369"><a href="#cb40-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb40-370"><a href="#cb40-370" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb40-371"><a href="#cb40-371" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb40-372"><a href="#cb40-372" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb40-373"><a href="#cb40-373" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb40-374"><a href="#cb40-374" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span> </span>
<span id="cb40-375"><a href="#cb40-375" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>), <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb40-376"><a href="#cb40-376" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb40-377"><a href="#cb40-377" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb40-378"><a href="#cb40-378" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span>
<span id="cb40-379"><a href="#cb40-379" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-380"><a href="#cb40-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-381"><a href="#cb40-381" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathway analysis using the genotype-specific DEGs (Main effects) using ORA</span></span>
<span id="cb40-382"><a href="#cb40-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-385"><a href="#cb40-385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-386"><a href="#cb40-386" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb40-387"><a href="#cb40-387" aria-hidden="true" tabindex="-1"></a>msigdb_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getMsigdb</span>(<span class="at">org =</span> <span class="st">"mouse"</span>, <span class="at">category =</span> <span class="st">"H"</span>)</span>
<span id="cb40-388"><a href="#cb40-388" aria-hidden="true" tabindex="-1"></a>go_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getGO</span>(<span class="at">org =</span> <span class="st">"mouse"</span>, <span class="at">ont =</span> <span class="st">"bp"</span>)</span>
<span id="cb40-389"><a href="#cb40-389" aria-hidden="true" tabindex="-1"></a>kegg_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getKEGG</span>(<span class="at">org =</span> <span class="st">"mus"</span>, <span class="at">category =</span> <span class="st">"pathway"</span>)</span>
<span id="cb40-390"><a href="#cb40-390" aria-hidden="true" tabindex="-1"></a>react_gs <span class="ot">&lt;-</span> geneset<span class="sc">::</span><span class="fu">getReactome</span>(<span class="at">org =</span> <span class="st">"mouse"</span>)</span>
<span id="cb40-391"><a href="#cb40-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-392"><a href="#cb40-392" aria-hidden="true" tabindex="-1"></a>gene_set <span class="ot">=</span> <span class="fu">list</span>(msigdb_gs, go_gs, kegg_gs, react_gs)</span>
<span id="cb40-393"><a href="#cb40-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-394"><a href="#cb40-394" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb40-395"><a href="#cb40-395" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb40-396"><a href="#cb40-396" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb40-397"><a href="#cb40-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-398"><a href="#cb40-398" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb40-399"><a href="#cb40-399" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"KO"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-400"><a href="#cb40-400" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-401"><a href="#cb40-401" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-402"><a href="#cb40-402" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-403"><a href="#cb40-403" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res1))<span class="sc">$</span>result</span>
<span id="cb40-404"><a href="#cb40-404" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-405"><a href="#cb40-405" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb40-406"><a href="#cb40-406" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-407"><a href="#cb40-407" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"KO"</span></span>
<span id="cb40-408"><a href="#cb40-408" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-409"><a href="#cb40-409" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-410"><a href="#cb40-410" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-411"><a href="#cb40-411" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-412"><a href="#cb40-412" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-413"><a href="#cb40-413" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-414"><a href="#cb40-414" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res1))<span class="sc">$</span>result</span>
<span id="cb40-415"><a href="#cb40-415" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-416"><a href="#cb40-416" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-417"><a href="#cb40-417" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-418"><a href="#cb40-418" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"WT"</span></span>
<span id="cb40-419"><a href="#cb40-419" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-420"><a href="#cb40-420" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-421"><a href="#cb40-421" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb40-422"><a href="#cb40-422" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-423"><a href="#cb40-423" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-424"><a href="#cb40-424" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb40-425"><a href="#cb40-425" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-426"><a href="#cb40-426" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-427"><a href="#cb40-427" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-428"><a href="#cb40-428" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-429"><a href="#cb40-429" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb40-430"><a href="#cb40-430" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-431"><a href="#cb40-431" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-432"><a href="#cb40-432" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-433"><a href="#cb40-433" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-434"><a href="#cb40-434" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb40-435"><a href="#cb40-435" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-436"><a href="#cb40-436" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-437"><a href="#cb40-437" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-438"><a href="#cb40-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-439"><a href="#cb40-439" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-440"><a href="#cb40-440" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-441"><a href="#cb40-441" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway profile across multiple databases and genotype conditions"</span></span>
<span id="cb40-442"><a href="#cb40-442" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span>
<span id="cb40-443"><a href="#cb40-443" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb40-444"><a href="#cb40-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb40-445"><a href="#cb40-445" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb40-446"><a href="#cb40-446" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-447"><a href="#cb40-447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-448"><a href="#cb40-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-449"><a href="#cb40-449" aria-hidden="true" tabindex="-1"></a>**None of the genotype-enriched DEGs fall into significant pathways across multiple pathway databases (KEGG, GO-BP, MSigDB hallmark and Reactome) when using ORA approach along with the universe argument**</span>
<span id="cb40-450"><a href="#cb40-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-451"><a href="#cb40-451" aria-hidden="true" tabindex="-1"></a>Let's see if change of approach from ORA to GSEA lead to better results.</span>
<span id="cb40-452"><a href="#cb40-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-453"><a href="#cb40-453" aria-hidden="true" tabindex="-1"></a>ORA method is easy to conduct, but it will lose information when genes’ differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by fold change cutoff. Unlike ORA, FCS tools do not set a threshold for differentially expressed genes. Instead, it gives each detected gene a differential expression score and then evaluates whether the scores are more positive or negative than expected by chance for each gene set.</span>
<span id="cb40-454"><a href="#cb40-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-455"><a href="#cb40-455" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathway analysis using the genotype-specific DEGs (Main effects) using GSEA</span></span>
<span id="cb40-456"><a href="#cb40-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-457"><a href="#cb40-457" aria-hidden="true" tabindex="-1"></a><span class="in">```{r GSEA approach using the res1 output}</span></span>
<span id="cb40-458"><a href="#cb40-458" aria-hidden="true" tabindex="-1"></a>rankedGenes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb40-459"><a href="#cb40-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(gene, padj, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb40-460"><a href="#cb40-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue) <span class="sc">*</span> <span class="fu">sign</span>(log2FoldChange),</span>
<span id="cb40-461"><a href="#cb40-461" aria-hidden="true" tabindex="-1"></a>         <span class="at">entrez =</span> <span class="fu">transId</span>(gene, <span class="at">transTo =</span> <span class="st">"entrez"</span>, <span class="at">keepNA =</span> T, <span class="at">unique =</span> T, <span class="at">org =</span> <span class="st">"mouse"</span>)<span class="sc">$</span>entrezid) <span class="sc">%&gt;%</span></span>
<span id="cb40-462"><a href="#cb40-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(rank) <span class="sc">&amp;</span> <span class="fu">is.finite</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb40-463"><a href="#cb40-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb40-464"><a href="#cb40-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(rank, gene)</span>
<span id="cb40-465"><a href="#cb40-465" aria-hidden="true" tabindex="-1"></a><span class="co">#head(rankedGenes)</span></span>
<span id="cb40-466"><a href="#cb40-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-467"><a href="#cb40-467" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb40-468"><a href="#cb40-468" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb40-469"><a href="#cb40-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-470"><a href="#cb40-470" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb40-471"><a href="#cb40-471" aria-hidden="true" tabindex="-1"></a>  gsea_results1 <span class="ot">=</span> <span class="fu">safely</span>(genGSEA)(<span class="at">genelist =</span> rankedGenes,</span>
<span id="cb40-472"><a href="#cb40-472" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-473"><a href="#cb40-473" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">q_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-474"><a href="#cb40-474" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]])<span class="sc">$</span>result</span>
<span id="cb40-475"><a href="#cb40-475" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-476"><a href="#cb40-476" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(gsea_results1<span class="sc">$</span>gsea_df)) {</span>
<span id="cb40-477"><a href="#cb40-477" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-478"><a href="#cb40-478" aria-hidden="true" tabindex="-1"></a>    barplots_1[[i]] <span class="ot">&lt;-</span> <span class="fu">plotGSEA</span>(gsea_results1, <span class="at">plot_type =</span> <span class="st">"bar"</span>, <span class="at">colour =</span> <span class="fu">c</span>(<span class="st">"navyblue"</span>, <span class="st">"orange"</span>))</span>
<span id="cb40-479"><a href="#cb40-479" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-480"><a href="#cb40-480" aria-hidden="true" tabindex="-1"></a>    df1 <span class="ot">=</span> <span class="fu">rbind_common</span>(df1, gsea_results1<span class="sc">$</span>gsea_df <span class="sc">%&gt;%</span> </span>
<span id="cb40-481"><a href="#cb40-481" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">mutate</span>(<span class="at">enrichment =</span> <span class="fu">ifelse</span>(NES <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>),</span>
<span id="cb40-482"><a href="#cb40-482" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]]<span class="sc">$</span>type))</span>
<span id="cb40-483"><a href="#cb40-483" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-484"><a href="#cb40-484" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-485"><a href="#cb40-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-486"><a href="#cb40-486" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-487"><a href="#cb40-487" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-488"><a href="#cb40-488" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway table across multiple databases and genotype conditions"</span></span>
<span id="cb40-489"><a href="#cb40-489" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Signficant pathways (p.adjust &lt; 0.05) identified using GSEA using multiple databases"</span></span>
<span id="cb40-490"><a href="#cb40-490" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span>
<span id="cb40-491"><a href="#cb40-491" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb40-492"><a href="#cb40-492" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment, <span class="at">legend_labels =</span> <span class="st">"-log10(p.adjust)"</span></span>
<span id="cb40-493"><a href="#cb40-493" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb40-494"><a href="#cb40-494" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-495"><a href="#cb40-495" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-496"><a href="#cb40-496" aria-hidden="true" tabindex="-1"></a>All the significant pathways (p.adjust &lt; 0.05) are labelled in blue color. </span>
<span id="cb40-497"><a href="#cb40-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-498"><a href="#cb40-498" aria-hidden="true" tabindex="-1"></a>**Conclusion: Genotype-specific DEGs fell into multiple respiratory and mitochondrial related pathways across multiple pathway databases (KEGG, GO-BP, MSigDB hallmark and Reactome) when using GSEA approach. The statistically robust approach GSEA is able to identify pathways but not ORA which is surprising.**</span>
<span id="cb40-499"><a href="#cb40-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-500"><a href="#cb40-500" aria-hidden="true" tabindex="-1"></a><span class="fu">### Main effects due to timepoint</span></span>
<span id="cb40-501"><a href="#cb40-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-502"><a href="#cb40-502" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Comparison of Wald versus LRT test DEGs \[Main effects due to time_point\]</span></span>
<span id="cb40-503"><a href="#cb40-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-504"><a href="#cb40-504" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Comparison of Wald test and LRT - Timepoint}</span></span>
<span id="cb40-505"><a href="#cb40-505" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-506"><a href="#cb40-506" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-507"><a href="#cb40-507" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with Wald test [Main effects due to time_point]"</span></span>
<span id="cb40-508"><a href="#cb40-508" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with paired design and with using LRT [Main effects due to time_point]"</span></span>
<span id="cb40-509"><a href="#cb40-509" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to time_point]"</span></span>
<span id="cb40-510"><a href="#cb40-510" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-511"><a href="#cb40-511" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-512"><a href="#cb40-512" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> time_point) <span class="co">#not accounting for timepoint</span></span>
<span id="cb40-513"><a href="#cb40-513" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb40-514"><a href="#cb40-514" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb40-515"><a href="#cb40-515" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb40-516"><a href="#cb40-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-517"><a href="#cb40-517" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-518"><a href="#cb40-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-519"><a href="#cb40-519" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-520"><a href="#cb40-520" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-521"><a href="#cb40-521" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-522"><a href="#cb40-522" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-Wald"</span>,</span>
<span id="cb40-523"><a href="#cb40-523" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb40-524"><a href="#cb40-524" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb40-525"><a href="#cb40-525" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb40-526"><a href="#cb40-526" aria-hidden="true" tabindex="-1"></a>res4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-527"><a href="#cb40-527" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb40-528"><a href="#cb40-528" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-529"><a href="#cb40-529" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-530"><a href="#cb40-530" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb40-531"><a href="#cb40-531" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> genotype)</span>
<span id="cb40-532"><a href="#cb40-532" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb40-533"><a href="#cb40-533" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> res5 <span class="sc">%&gt;%</span></span>
<span id="cb40-534"><a href="#cb40-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-535"><a href="#cb40-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-536"><a href="#cb40-536" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-537"><a href="#cb40-537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-538"><a href="#cb40-538" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-539"><a href="#cb40-539" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-540"><a href="#cb40-540" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-LRT"</span>,</span>
<span id="cb40-541"><a href="#cb40-541" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb40-542"><a href="#cb40-542" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb40-543"><a href="#cb40-543" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb40-544"><a href="#cb40-544" aria-hidden="true" tabindex="-1"></a>res5 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-545"><a href="#cb40-545" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res4,res5)</span>
<span id="cb40-546"><a href="#cb40-546" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb40-547"><a href="#cb40-547" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb40-548"><a href="#cb40-548" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb40-549"><a href="#cb40-549" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb40-550"><a href="#cb40-550" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb40-551"><a href="#cb40-551" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb40-552"><a href="#cb40-552" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span>
<span id="cb40-553"><a href="#cb40-553" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-554"><a href="#cb40-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-555"><a href="#cb40-555" aria-hidden="true" tabindex="-1"></a>DEGs with and without accounting for genotype yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to timepoints.</span>
<span id="cb40-556"><a href="#cb40-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-557"><a href="#cb40-557" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Visualization of timepoint-specific DEGs using DESeq2</span></span>
<span id="cb40-558"><a href="#cb40-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-559"><a href="#cb40-559" aria-hidden="true" tabindex="-1"></a>I am going with res4 object that uses Wald test to identify main effects due to genotype. Wald test DEGs are all represented in the LRT DEGs as determined by the Upset plot shown above.</span>
<span id="cb40-560"><a href="#cb40-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-561"><a href="#cb40-561" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Identification of timepoint-specific DEGs}</span></span>
<span id="cb40-562"><a href="#cb40-562" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-563"><a href="#cb40-563" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-564"><a href="#cb40-564" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-565"><a href="#cb40-565" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genotype-specific DEGs agnostic of the sampling timepoint"</span></span>
<span id="cb40-566"><a href="#cb40-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-567"><a href="#cb40-567" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb40-568"><a href="#cb40-568" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res4,</span>
<span id="cb40-569"><a href="#cb40-569" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb40-570"><a href="#cb40-570" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb40-571"><a href="#cb40-571" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res4<span class="sc">$</span>gene,</span>
<span id="cb40-572"><a href="#cb40-572" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb40-573"><a href="#cb40-573" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-574"><a href="#cb40-574" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"T12 vs T0 comparison [Single-factor design]"</span>,</span>
<span id="cb40-575"><a href="#cb40-575" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb40-576"><a href="#cb40-576" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb40-577"><a href="#cb40-577" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb40-578"><a href="#cb40-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-579"><a href="#cb40-579" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb40-580"><a href="#cb40-580" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res4<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb40-581"><a href="#cb40-581" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-582"><a href="#cb40-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb40-583"><a href="#cb40-583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-584"><a href="#cb40-584" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb40-585"><a href="#cb40-585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb40-586"><a href="#cb40-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb40-587"><a href="#cb40-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb40-588"><a href="#cb40-588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb40-589"><a href="#cb40-589" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-590"><a href="#cb40-590" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb40-591"><a href="#cb40-591" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb40-592"><a href="#cb40-592" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span>
<span id="cb40-593"><a href="#cb40-593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-594"><a href="#cb40-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-595"><a href="#cb40-595" aria-hidden="true" tabindex="-1"></a>The timepoint based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</span>
<span id="cb40-596"><a href="#cb40-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-597"><a href="#cb40-597" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Expression pattern of top time-specific DEGs</span></span>
<span id="cb40-598"><a href="#cb40-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-599"><a href="#cb40-599" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Expression pattern of top time-specific DEGs}</span></span>
<span id="cb40-600"><a href="#cb40-600" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb40-601"><a href="#cb40-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-602"><a href="#cb40-602" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-603"><a href="#cb40-603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-604"><a href="#cb40-604" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb40-605"><a href="#cb40-605" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb40-606"><a href="#cb40-606" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb40-607"><a href="#cb40-607" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">time_point =</span> <span class="fu">c</span>(<span class="st">"T0"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"T12"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb40-608"><a href="#cb40-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-609"><a href="#cb40-609" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-610"><a href="#cb40-610" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-611"><a href="#cb40-611" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-612"><a href="#cb40-612" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb40-613"><a href="#cb40-613" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb40-614"><a href="#cb40-614" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb40-615"><a href="#cb40-615" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb40-616"><a href="#cb40-616" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span>,</span>
<span id="cb40-617"><a href="#cb40-617" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb40-618"><a href="#cb40-618" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>)</span>
<span id="cb40-619"><a href="#cb40-619" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb40-620"><a href="#cb40-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-621"><a href="#cb40-621" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb40-622"><a href="#cb40-622" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-623"><a href="#cb40-623" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-624"><a href="#cb40-624" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-625"><a href="#cb40-625" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb40-626"><a href="#cb40-626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-627"><a href="#cb40-627" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb40-628"><a href="#cb40-628" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb40-629"><a href="#cb40-629" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb40-630"><a href="#cb40-630" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb40-631"><a href="#cb40-631" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb40-632"><a href="#cb40-632" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb40-633"><a href="#cb40-633" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span> </span>
<span id="cb40-634"><a href="#cb40-634" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb40-635"><a href="#cb40-635" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb40-636"><a href="#cb40-636" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb40-637"><a href="#cb40-637" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span>
<span id="cb40-638"><a href="#cb40-638" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-639"><a href="#cb40-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-640"><a href="#cb40-640" aria-hidden="true" tabindex="-1"></a>The timepoint based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</span>
<span id="cb40-641"><a href="#cb40-641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-642"><a href="#cb40-642" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-643"><a href="#cb40-643" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathway analysis using the timepoint-specific DEGs (Main effects) using ORA</span></span>
<span id="cb40-644"><a href="#cb40-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-647"><a href="#cb40-647" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-648"><a href="#cb40-648" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb40-649"><a href="#cb40-649" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb40-650"><a href="#cb40-650" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb40-651"><a href="#cb40-651" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb40-652"><a href="#cb40-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-653"><a href="#cb40-653" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb40-654"><a href="#cb40-654" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res4 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"T12"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-655"><a href="#cb40-655" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-656"><a href="#cb40-656" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-657"><a href="#cb40-657" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-658"><a href="#cb40-658" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res4))<span class="sc">$</span>result</span>
<span id="cb40-659"><a href="#cb40-659" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-660"><a href="#cb40-660" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb40-661"><a href="#cb40-661" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-662"><a href="#cb40-662" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"T12"</span></span>
<span id="cb40-663"><a href="#cb40-663" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-664"><a href="#cb40-664" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-665"><a href="#cb40-665" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res4 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"T0"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-666"><a href="#cb40-666" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-667"><a href="#cb40-667" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-668"><a href="#cb40-668" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-669"><a href="#cb40-669" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res4))<span class="sc">$</span>result</span>
<span id="cb40-670"><a href="#cb40-670" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-671"><a href="#cb40-671" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-672"><a href="#cb40-672" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-673"><a href="#cb40-673" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"T0"</span></span>
<span id="cb40-674"><a href="#cb40-674" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-675"><a href="#cb40-675" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-676"><a href="#cb40-676" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb40-677"><a href="#cb40-677" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-678"><a href="#cb40-678" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-679"><a href="#cb40-679" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb40-680"><a href="#cb40-680" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-681"><a href="#cb40-681" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-682"><a href="#cb40-682" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-683"><a href="#cb40-683" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-684"><a href="#cb40-684" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb40-685"><a href="#cb40-685" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-686"><a href="#cb40-686" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-687"><a href="#cb40-687" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-688"><a href="#cb40-688" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-689"><a href="#cb40-689" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb40-690"><a href="#cb40-690" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-691"><a href="#cb40-691" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-692"><a href="#cb40-692" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-693"><a href="#cb40-693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-694"><a href="#cb40-694" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1, <span class="at">plot_type =</span> <span class="st">"dot"</span>)</span>
<span id="cb40-695"><a href="#cb40-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-696"><a href="#cb40-696" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-697"><a href="#cb40-697" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-698"><a href="#cb40-698" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway enrichment stats across multiple databases and genotype conditions"</span></span>
<span id="cb40-699"><a href="#cb40-699" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Significant pathways (labelled in blue) across timepoints when using ORA and multiple pathway databases"</span></span>
<span id="cb40-700"><a href="#cb40-700" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span>
<span id="cb40-701"><a href="#cb40-701" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb40-702"><a href="#cb40-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb40-703"><a href="#cb40-703" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb40-704"><a href="#cb40-704" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-705"><a href="#cb40-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-706"><a href="#cb40-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-707"><a href="#cb40-707" aria-hidden="true" tabindex="-1"></a>**Only the T0-enriched DEGs are associated with Reactome pathways significantly when using ORA approach along with the universe argument. T0-enriched pathways are mainly collagen and ECM related.**</span>
<span id="cb40-708"><a href="#cb40-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-709"><a href="#cb40-709" aria-hidden="true" tabindex="-1"></a>Let's see if change of approach from ORA to GSEA lead to different results.</span>
<span id="cb40-710"><a href="#cb40-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-711"><a href="#cb40-711" aria-hidden="true" tabindex="-1"></a>ORA method is easy to conduct, but it will lose information when genes’ differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by fold change cutoff. Unlike ORA, FCS tools do not set a threshold for differentially expressed genes. Instead, it gives each detected gene a differential expression score and then evaluates whether the scores are more positive or negative than expected by chance for each gene set.</span>
<span id="cb40-712"><a href="#cb40-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-713"><a href="#cb40-713" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathway analysis using the genotype-specific DEGs (Main effects) using GSEA</span></span>
<span id="cb40-714"><a href="#cb40-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-715"><a href="#cb40-715" aria-hidden="true" tabindex="-1"></a><span class="in">```{r GSEA approach using the res4 output}</span></span>
<span id="cb40-716"><a href="#cb40-716" aria-hidden="true" tabindex="-1"></a>rankedGenes <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb40-717"><a href="#cb40-717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(gene, padj, log2FoldChange) <span class="sc">%&gt;%</span></span>
<span id="cb40-718"><a href="#cb40-718" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue) <span class="sc">*</span> <span class="fu">sign</span>(log2FoldChange),</span>
<span id="cb40-719"><a href="#cb40-719" aria-hidden="true" tabindex="-1"></a>         <span class="at">entrez =</span> <span class="fu">transId</span>(gene, <span class="at">transTo =</span> <span class="st">"entrez"</span>, <span class="at">keepNA =</span> T, <span class="at">unique =</span> T, <span class="at">org =</span> <span class="st">"mouse"</span>)<span class="sc">$</span>entrezid) <span class="sc">%&gt;%</span></span>
<span id="cb40-720"><a href="#cb40-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(rank) <span class="sc">&amp;</span> <span class="fu">is.finite</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb40-721"><a href="#cb40-721" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb40-722"><a href="#cb40-722" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(rank, gene)</span>
<span id="cb40-723"><a href="#cb40-723" aria-hidden="true" tabindex="-1"></a><span class="co">#head(rankedGenes)</span></span>
<span id="cb40-724"><a href="#cb40-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-725"><a href="#cb40-725" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb40-726"><a href="#cb40-726" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb40-727"><a href="#cb40-727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-728"><a href="#cb40-728" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb40-729"><a href="#cb40-729" aria-hidden="true" tabindex="-1"></a>  gsea_results1 <span class="ot">=</span> <span class="fu">safely</span>(genGSEA)(<span class="at">genelist =</span> rankedGenes,</span>
<span id="cb40-730"><a href="#cb40-730" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">p_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-731"><a href="#cb40-731" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">q_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-732"><a href="#cb40-732" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]])<span class="sc">$</span>result</span>
<span id="cb40-733"><a href="#cb40-733" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-734"><a href="#cb40-734" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(gsea_results1<span class="sc">$</span>gsea_df)) {</span>
<span id="cb40-735"><a href="#cb40-735" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-736"><a href="#cb40-736" aria-hidden="true" tabindex="-1"></a>    barplots_1[[i]] <span class="ot">&lt;-</span> <span class="fu">plotGSEA</span>(gsea_results1, <span class="at">plot_type =</span> <span class="st">"bar"</span>, <span class="at">colour =</span> <span class="fu">c</span>(<span class="st">"navyblue"</span>, <span class="st">"orange"</span>))</span>
<span id="cb40-737"><a href="#cb40-737" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb40-738"><a href="#cb40-738" aria-hidden="true" tabindex="-1"></a>    df1 <span class="ot">=</span> <span class="fu">rbind_common</span>(df1, gsea_results1<span class="sc">$</span>gsea_df <span class="sc">%&gt;%</span> </span>
<span id="cb40-739"><a href="#cb40-739" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">mutate</span>(<span class="at">enrichment =</span> <span class="fu">ifelse</span>(NES <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>),</span>
<span id="cb40-740"><a href="#cb40-740" aria-hidden="true" tabindex="-1"></a>                                <span class="at">geneset =</span> gene_set[[i]]<span class="sc">$</span>type))</span>
<span id="cb40-741"><a href="#cb40-741" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-742"><a href="#cb40-742" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-743"><a href="#cb40-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-744"><a href="#cb40-744" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-745"><a href="#cb40-745" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-746"><a href="#cb40-746" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway table across multiple databases and genotype conditions"</span></span>
<span id="cb40-747"><a href="#cb40-747" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Signficant pathways (p.adjust &lt; 0.01) identified using GSEA using multiple databases"</span></span>
<span id="cb40-748"><a href="#cb40-748" aria-hidden="true" tabindex="-1"></a>df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span>
<span id="cb40-749"><a href="#cb40-749" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.01</span>)) <span class="sc">&gt;=</span> <span class="dv">1</span>) {</span>
<span id="cb40-750"><a href="#cb40-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment, <span class="at">legend_labels =</span> <span class="st">"-log10(p.adjust)"</span></span>
<span id="cb40-751"><a href="#cb40-751" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb40-752"><a href="#cb40-752" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-753"><a href="#cb40-753" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-754"><a href="#cb40-754" aria-hidden="true" tabindex="-1"></a>All the significant pathways (p.adjust &lt; 0.01) are labelled in blue color. </span>
<span id="cb40-755"><a href="#cb40-755" aria-hidden="true" tabindex="-1"></a>To beautify the output and the plots, I resorted to a stricter cutoff for pathways i.e. p.adjust &lt; 0.01 instead of p.adjust &lt; 0.05</span>
<span id="cb40-756"><a href="#cb40-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-757"><a href="#cb40-757" aria-hidden="true" tabindex="-1"></a>**Conclusion: T0-enriched DEGs fell into ECM, collagen and EMT related pathways when using multiple pathway databases (KEGG, GO-BP, MSigDB hallmark and Reactome) when using GSEA approach. GSEA approach also confirmed the findings from the ORA approach. In addition, T12 associated pathways belong to mitochondria, respiration and Oxidative phosphorylation pathway terms. The statistically robust approach GSEA is able to identify more pathways compared to ORA which is surprising.**</span>
<span id="cb40-758"><a href="#cb40-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-759"><a href="#cb40-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-760"><a href="#cb40-760" aria-hidden="true" tabindex="-1"></a><span class="fu">### Interaction effects</span></span>
<span id="cb40-761"><a href="#cb40-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-762"><a href="#cb40-762" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Comparison of Wald versus LRT test - interaction effects</span></span>
<span id="cb40-763"><a href="#cb40-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-764"><a href="#cb40-764" aria-hidden="true" tabindex="-1"></a>**DEGs can be defined genes passing the threshold: abs(logFC) \&gt; 1 &amp; padj \&lt; 0.01**</span>
<span id="cb40-765"><a href="#cb40-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-766"><a href="#cb40-766" aria-hidden="true" tabindex="-1"></a>This is useful to detect interaction between genotype and timepoint. </span>
<span id="cb40-767"><a href="#cb40-767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-768"><a href="#cb40-768" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Comparison of Wald test and LRT - interaction effects}</span></span>
<span id="cb40-769"><a href="#cb40-769" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-770"><a href="#cb40-770" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-771"><a href="#cb40-771" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with Wald test [Main effects due to genotype]"</span></span>
<span id="cb40-772"><a href="#cb40-772" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with paired design and with using LRT [Main effects due to genotype]"</span></span>
<span id="cb40-773"><a href="#cb40-773" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to genotype]"</span></span>
<span id="cb40-774"><a href="#cb40-774" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-775"><a href="#cb40-775" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-776"><a href="#cb40-776" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">*</span> time_point) <span class="co">#interaction term</span></span>
<span id="cb40-777"><a href="#cb40-777" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb40-778"><a href="#cb40-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-779"><a href="#cb40-779" aria-hidden="true" tabindex="-1"></a><span class="co">#vd &lt;- VisualizeDesign(sampleData = coldata,</span></span>
<span id="cb40-780"><a href="#cb40-780" aria-hidden="true" tabindex="-1"></a> <span class="co">#                     designFormula = ~ genotype * time_point)</span></span>
<span id="cb40-781"><a href="#cb40-781" aria-hidden="true" tabindex="-1"></a><span class="co">#print(vd$plotlist) #can visualize the model matrix and the various coefficients</span></span>
<span id="cb40-782"><a href="#cb40-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-783"><a href="#cb40-783" aria-hidden="true" tabindex="-1"></a>res6 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">name =</span> <span class="st">"genotypeWT.time_pointT12"</span>)</span>
<span id="cb40-784"><a href="#cb40-784" aria-hidden="true" tabindex="-1"></a>res6 <span class="ot">&lt;-</span> res6 <span class="sc">%&gt;%</span></span>
<span id="cb40-785"><a href="#cb40-785" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-786"><a href="#cb40-786" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-787"><a href="#cb40-787" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-788"><a href="#cb40-788" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-789"><a href="#cb40-789" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-790"><a href="#cb40-790" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-791"><a href="#cb40-791" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"Interaction-Wald"</span>,</span>
<span id="cb40-792"><a href="#cb40-792" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb40-793"><a href="#cb40-793" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb40-794"><a href="#cb40-794" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb40-795"><a href="#cb40-795" aria-hidden="true" tabindex="-1"></a>res6 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-796"><a href="#cb40-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-797"><a href="#cb40-797" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb40-798"><a href="#cb40-798" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-799"><a href="#cb40-799" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-800"><a href="#cb40-800" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">*</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb40-801"><a href="#cb40-801" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">*</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point)</span>
<span id="cb40-802"><a href="#cb40-802" aria-hidden="true" tabindex="-1"></a>res7 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">name =</span> <span class="st">"genotypeWT.time_pointT12"</span>)</span>
<span id="cb40-803"><a href="#cb40-803" aria-hidden="true" tabindex="-1"></a>res7 <span class="ot">&lt;-</span> res7 <span class="sc">%&gt;%</span></span>
<span id="cb40-804"><a href="#cb40-804" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-805"><a href="#cb40-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-806"><a href="#cb40-806" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-807"><a href="#cb40-807" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-808"><a href="#cb40-808" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-809"><a href="#cb40-809" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-810"><a href="#cb40-810" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"Interaction-LRT"</span>,</span>
<span id="cb40-811"><a href="#cb40-811" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb40-812"><a href="#cb40-812" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb40-813"><a href="#cb40-813" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb40-814"><a href="#cb40-814" aria-hidden="true" tabindex="-1"></a>res7 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-815"><a href="#cb40-815" aria-hidden="true" tabindex="-1"></a>res8 <span class="ot">=</span> <span class="fu">rbind_common</span>(res6,res7)</span>
<span id="cb40-816"><a href="#cb40-816" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb40-817"><a href="#cb40-817" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb40-818"><a href="#cb40-818" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res8, <span class="at">f =</span> <span class="fu">list</span>(res8<span class="sc">$</span>comparison, res8<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb40-819"><a href="#cb40-819" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb40-820"><a href="#cb40-820" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb40-821"><a href="#cb40-821" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb40-822"><a href="#cb40-822" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span>
<span id="cb40-823"><a href="#cb40-823" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-824"><a href="#cb40-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-825"><a href="#cb40-825" aria-hidden="true" tabindex="-1"></a>DEGs are common irrespective of Wald or LRT test used to obtain interaction effects.</span>
<span id="cb40-826"><a href="#cb40-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-827"><a href="#cb40-827" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Visualization of interaction-specific DEGs using DESeq2</span></span>
<span id="cb40-828"><a href="#cb40-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-829"><a href="#cb40-829" aria-hidden="true" tabindex="-1"></a>I am going with res6 object that uses Wald test to identify main effects due to genotype. Wald test DEGs are all represented in the LRT DEGs as determined by the previous Upset plot</span>
<span id="cb40-830"><a href="#cb40-830" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-831"><a href="#cb40-831" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Identification of interaction-specific DEGs}</span></span>
<span id="cb40-832"><a href="#cb40-832" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-833"><a href="#cb40-833" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-834"><a href="#cb40-834" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-835"><a href="#cb40-835" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genotype-specific DEGs agnostic of the sampling timepoint"</span></span>
<span id="cb40-836"><a href="#cb40-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-837"><a href="#cb40-837" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb40-838"><a href="#cb40-838" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res6,</span>
<span id="cb40-839"><a href="#cb40-839" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb40-840"><a href="#cb40-840" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb40-841"><a href="#cb40-841" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res6<span class="sc">$</span>gene,</span>
<span id="cb40-842"><a href="#cb40-842" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb40-843"><a href="#cb40-843" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb40-844"><a href="#cb40-844" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"comparison [Interaction effect]"</span>,</span>
<span id="cb40-845"><a href="#cb40-845" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb40-846"><a href="#cb40-846" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb40-847"><a href="#cb40-847" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb40-848"><a href="#cb40-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-849"><a href="#cb40-849" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb40-850"><a href="#cb40-850" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res6<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb40-851"><a href="#cb40-851" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-852"><a href="#cb40-852" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb40-853"><a href="#cb40-853" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-854"><a href="#cb40-854" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb40-855"><a href="#cb40-855" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb40-856"><a href="#cb40-856" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb40-857"><a href="#cb40-857" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb40-858"><a href="#cb40-858" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb40-859"><a href="#cb40-859" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-860"><a href="#cb40-860" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb40-861"><a href="#cb40-861" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb40-862"><a href="#cb40-862" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span>
<span id="cb40-863"><a href="#cb40-863" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-864"><a href="#cb40-864" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-865"><a href="#cb40-865" aria-hidden="true" tabindex="-1"></a>The timepoint based DEGs do not have specific pattern because enrichment can be at any genotype and timepoint combination</span>
<span id="cb40-866"><a href="#cb40-866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-867"><a href="#cb40-867" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Expression pattern of top interaction-specific DEGs</span></span>
<span id="cb40-868"><a href="#cb40-868" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-869"><a href="#cb40-869" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Expression pattern of top interaction-specific DEGs}</span></span>
<span id="cb40-870"><a href="#cb40-870" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res6 <span class="sc">%&gt;%</span></span>
<span id="cb40-871"><a href="#cb40-871" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-872"><a href="#cb40-872" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-873"><a href="#cb40-873" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-874"><a href="#cb40-874" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb40-875"><a href="#cb40-875" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb40-876"><a href="#cb40-876" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb40-877"><a href="#cb40-877" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">time_point =</span> <span class="fu">c</span>(<span class="st">"T0"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"T12"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb40-878"><a href="#cb40-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-879"><a href="#cb40-879" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-880"><a href="#cb40-880" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-881"><a href="#cb40-881" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-882"><a href="#cb40-882" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb40-883"><a href="#cb40-883" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb40-884"><a href="#cb40-884" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb40-885"><a href="#cb40-885" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb40-886"><a href="#cb40-886" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span>,</span>
<span id="cb40-887"><a href="#cb40-887" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb40-888"><a href="#cb40-888" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>)</span>
<span id="cb40-889"><a href="#cb40-889" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb40-890"><a href="#cb40-890" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-891"><a href="#cb40-891" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res6 <span class="sc">%&gt;%</span></span>
<span id="cb40-892"><a href="#cb40-892" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-893"><a href="#cb40-893" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-894"><a href="#cb40-894" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-895"><a href="#cb40-895" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb40-896"><a href="#cb40-896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-897"><a href="#cb40-897" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb40-898"><a href="#cb40-898" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb40-899"><a href="#cb40-899" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb40-900"><a href="#cb40-900" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb40-901"><a href="#cb40-901" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb40-902"><a href="#cb40-902" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb40-903"><a href="#cb40-903" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span> </span>
<span id="cb40-904"><a href="#cb40-904" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb40-905"><a href="#cb40-905" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb40-906"><a href="#cb40-906" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb40-907"><a href="#cb40-907" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span>
<span id="cb40-908"><a href="#cb40-908" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-909"><a href="#cb40-909" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-910"><a href="#cb40-910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-911"><a href="#cb40-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-912"><a href="#cb40-912" aria-hidden="true" tabindex="-1"></a><span class="fu">## Paired contrast analysis - Difference between WT and KO at each timepoint</span></span>
<span id="cb40-913"><a href="#cb40-913" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-914"><a href="#cb40-914" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Identification of DEGs between WT and KO for every timepoint}</span></span>
<span id="cb40-915"><a href="#cb40-915" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-916"><a href="#cb40-916" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-917"><a href="#cb40-917" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01) by comparing KO versus WT at time T0"</span></span>
<span id="cb40-918"><a href="#cb40-918" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01) by comparing KO versus WT at time T12"</span></span>
<span id="cb40-919"><a href="#cb40-919" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Upset plot showing overlap between DEGs"</span></span>
<span id="cb40-920"><a href="#cb40-920" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">paste0</span>(coldata<span class="sc">$</span>genotype, <span class="st">"_"</span>, coldata<span class="sc">$</span>time_point)</span>
<span id="cb40-921"><a href="#cb40-921" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb40-922"><a href="#cb40-922" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb40-923"><a href="#cb40-923" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> group)</span>
<span id="cb40-924"><a href="#cb40-924" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb40-925"><a href="#cb40-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-926"><a href="#cb40-926" aria-hidden="true" tabindex="-1"></a><span class="co">#Comparison between KO and WT at time T0</span></span>
<span id="cb40-927"><a href="#cb40-927" aria-hidden="true" tabindex="-1"></a>res9 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"KO_T0"</span>, <span class="st">"WT_T0"</span>))</span>
<span id="cb40-928"><a href="#cb40-928" aria-hidden="true" tabindex="-1"></a>res9 <span class="ot">&lt;-</span> res9 <span class="sc">%&gt;%</span></span>
<span id="cb40-929"><a href="#cb40-929" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-930"><a href="#cb40-930" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-931"><a href="#cb40-931" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-932"><a href="#cb40-932" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-933"><a href="#cb40-933" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-934"><a href="#cb40-934" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-935"><a href="#cb40-935" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Time0-LRT"</span>,</span>
<span id="cb40-936"><a href="#cb40-936" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO_T0"</span>,</span>
<span id="cb40-937"><a href="#cb40-937" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT_T0"</span>,</span>
<span id="cb40-938"><a href="#cb40-938" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb40-939"><a href="#cb40-939" aria-hidden="true" tabindex="-1"></a>res9 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-940"><a href="#cb40-940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-941"><a href="#cb40-941" aria-hidden="true" tabindex="-1"></a><span class="co">#Comparison between KO and WT at time T12</span></span>
<span id="cb40-942"><a href="#cb40-942" aria-hidden="true" tabindex="-1"></a>res10 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"KO_T12"</span>, <span class="st">"WT_T12"</span>))</span>
<span id="cb40-943"><a href="#cb40-943" aria-hidden="true" tabindex="-1"></a>res10 <span class="ot">&lt;-</span> res10 <span class="sc">%&gt;%</span></span>
<span id="cb40-944"><a href="#cb40-944" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-945"><a href="#cb40-945" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-946"><a href="#cb40-946" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb40-947"><a href="#cb40-947" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb40-948"><a href="#cb40-948" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb40-949"><a href="#cb40-949" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb40-950"><a href="#cb40-950" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Time12-LRT"</span>,</span>
<span id="cb40-951"><a href="#cb40-951" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO_T12"</span>,</span>
<span id="cb40-952"><a href="#cb40-952" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT_T12"</span>,</span>
<span id="cb40-953"><a href="#cb40-953" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb40-954"><a href="#cb40-954" aria-hidden="true" tabindex="-1"></a>res10 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb40-955"><a href="#cb40-955" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-956"><a href="#cb40-956" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res9,res10)</span>
<span id="cb40-957"><a href="#cb40-957" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb40-958"><a href="#cb40-958" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb40-959"><a href="#cb40-959" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb40-960"><a href="#cb40-960" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb40-961"><a href="#cb40-961" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span>
<span id="cb40-962"><a href="#cb40-962" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-963"><a href="#cb40-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-964"><a href="#cb40-964" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathway analysis for genotype-enriched DEGs for timepoint T0</span></span>
<span id="cb40-965"><a href="#cb40-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-966"><a href="#cb40-966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-969"><a href="#cb40-969" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-970"><a href="#cb40-970" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb40-971"><a href="#cb40-971" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb40-972"><a href="#cb40-972" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb40-973"><a href="#cb40-973" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb40-974"><a href="#cb40-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-975"><a href="#cb40-975" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb40-976"><a href="#cb40-976" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res9 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"KO"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-977"><a href="#cb40-977" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-978"><a href="#cb40-978" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-979"><a href="#cb40-979" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-980"><a href="#cb40-980" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res9))<span class="sc">$</span>result</span>
<span id="cb40-981"><a href="#cb40-981" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-982"><a href="#cb40-982" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb40-983"><a href="#cb40-983" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-984"><a href="#cb40-984" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"KO"</span></span>
<span id="cb40-985"><a href="#cb40-985" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-986"><a href="#cb40-986" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-987"><a href="#cb40-987" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res9 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-988"><a href="#cb40-988" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-989"><a href="#cb40-989" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-990"><a href="#cb40-990" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-991"><a href="#cb40-991" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res9))<span class="sc">$</span>result</span>
<span id="cb40-992"><a href="#cb40-992" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-993"><a href="#cb40-993" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-994"><a href="#cb40-994" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-995"><a href="#cb40-995" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"WT"</span></span>
<span id="cb40-996"><a href="#cb40-996" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-997"><a href="#cb40-997" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-998"><a href="#cb40-998" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb40-999"><a href="#cb40-999" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1000"><a href="#cb40-1000" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-1001"><a href="#cb40-1001" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb40-1002"><a href="#cb40-1002" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-1003"><a href="#cb40-1003" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-1004"><a href="#cb40-1004" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-1005"><a href="#cb40-1005" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-1006"><a href="#cb40-1006" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb40-1007"><a href="#cb40-1007" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-1008"><a href="#cb40-1008" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-1009"><a href="#cb40-1009" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-1010"><a href="#cb40-1010" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-1011"><a href="#cb40-1011" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb40-1012"><a href="#cb40-1012" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1013"><a href="#cb40-1013" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1014"><a href="#cb40-1014" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1015"><a href="#cb40-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1016"><a href="#cb40-1016" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-1017"><a href="#cb40-1017" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-1018"><a href="#cb40-1018" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway enrichment stats across multiple databases and genotype conditions"</span></span>
<span id="cb40-1019"><a href="#cb40-1019" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Significant pathways (labelled in blue) across timepoints when using ORA and multiple pathway databases"</span></span>
<span id="cb40-1020"><a href="#cb40-1020" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(df1)) { df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>) }</span>
<span id="cb40-1021"><a href="#cb40-1021" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(df1)) {</span>
<span id="cb40-1022"><a href="#cb40-1022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb40-1023"><a href="#cb40-1023" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb40-1024"><a href="#cb40-1024" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1025"><a href="#cb40-1025" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1026"><a href="#cb40-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1027"><a href="#cb40-1027" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathway analysis for genotype-enriched DEGs for timepoint T12</span></span>
<span id="cb40-1028"><a href="#cb40-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1031"><a href="#cb40-1031" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1032"><a href="#cb40-1032" aria-hidden="true" tabindex="-1"></a><span class="co"># perform pathway analysis using multiple pathway genesets</span></span>
<span id="cb40-1033"><a href="#cb40-1033" aria-hidden="true" tabindex="-1"></a><span class="co">#empty lists for adding in the pathway results</span></span>
<span id="cb40-1034"><a href="#cb40-1034" aria-hidden="true" tabindex="-1"></a>barplots_1 <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb40-1035"><a href="#cb40-1035" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb40-1036"><a href="#cb40-1036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1037"><a href="#cb40-1037" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene_set)) {</span>
<span id="cb40-1038"><a href="#cb40-1038" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res10 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"KO"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-1039"><a href="#cb40-1039" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-1040"><a href="#cb40-1040" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-1041"><a href="#cb40-1041" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-1042"><a href="#cb40-1042" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res10))<span class="sc">$</span>result</span>
<span id="cb40-1043"><a href="#cb40-1043" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1044"><a href="#cb40-1044" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1)) {</span>
<span id="cb40-1045"><a href="#cb40-1045" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-1046"><a href="#cb40-1046" aria-hidden="true" tabindex="-1"></a>    go_results1<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"KO"</span></span>
<span id="cb40-1047"><a href="#cb40-1047" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1048"><a href="#cb40-1048" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1049"><a href="#cb40-1049" aria-hidden="true" tabindex="-1"></a>  go_results2 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(res10 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene),</span>
<span id="cb40-1050"><a href="#cb40-1050" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> gene_set[[i]],</span>
<span id="cb40-1051"><a href="#cb40-1051" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-1052"><a href="#cb40-1052" aria-hidden="true" tabindex="-1"></a>                               <span class="at">q_cutoff =</span> <span class="dv">1</span>, <span class="co">#to retrieve the whole output</span></span>
<span id="cb40-1053"><a href="#cb40-1053" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> <span class="fu">rownames</span>(res10))<span class="sc">$</span>result</span>
<span id="cb40-1054"><a href="#cb40-1054" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1055"><a href="#cb40-1055" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-1056"><a href="#cb40-1056" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>geneset <span class="ot">=</span> gene_set[[i]]<span class="sc">$</span>type</span>
<span id="cb40-1057"><a href="#cb40-1057" aria-hidden="true" tabindex="-1"></a>    go_results2<span class="sc">$</span>enrichment <span class="ot">=</span> <span class="st">"WT"</span></span>
<span id="cb40-1058"><a href="#cb40-1058" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1059"><a href="#cb40-1059" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1060"><a href="#cb40-1060" aria-hidden="true" tabindex="-1"></a>  df1 <span class="ot">&lt;-</span> <span class="fu">rbind_common</span>(df1, go_results1, go_results2)</span>
<span id="cb40-1061"><a href="#cb40-1061" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1062"><a href="#cb40-1062" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(go_results1) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(go_results2)) {</span>
<span id="cb40-1063"><a href="#cb40-1063" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results1,</span>
<span id="cb40-1064"><a href="#cb40-1064" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-1065"><a href="#cb40-1065" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-1066"><a href="#cb40-1066" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-1067"><a href="#cb40-1067" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-1068"><a href="#cb40-1068" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">plotEnrich</span>(go_results2,</span>
<span id="cb40-1069"><a href="#cb40-1069" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot_type =</span> <span class="st">"bar"</span>,</span>
<span id="cb40-1070"><a href="#cb40-1070" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n_term =</span> <span class="dv">10</span>,</span>
<span id="cb40-1071"><a href="#cb40-1071" aria-hidden="true" tabindex="-1"></a>                   <span class="at">term_metric =</span> <span class="st">"FoldEnrich"</span>,</span>
<span id="cb40-1072"><a href="#cb40-1072" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stats_metric =</span> <span class="st">"qvalue"</span>)</span>
<span id="cb40-1073"><a href="#cb40-1073" aria-hidden="true" tabindex="-1"></a>  barplots_1[[i]] <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2</span>
<span id="cb40-1074"><a href="#cb40-1074" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1075"><a href="#cb40-1075" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1076"><a href="#cb40-1076" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb40-1077"><a href="#cb40-1077" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1078"><a href="#cb40-1078" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-1079"><a href="#cb40-1079" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-1080"><a href="#cb40-1080" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Pathway enrichment stats across multiple databases and genotype conditions"</span></span>
<span id="cb40-1081"><a href="#cb40-1081" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Significant pathways (labelled in blue) across timepoints when using ORA and multiple pathway databases"</span></span>
<span id="cb40-1082"><a href="#cb40-1082" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(df1)) { df1 <span class="sc">%&gt;%</span> <span class="fu">head</span>() <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>) }</span>
<span id="cb40-1083"><a href="#cb40-1083" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(df1)) {</span>
<span id="cb40-1084"><a href="#cb40-1084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">tidyheatmap</span>(<span class="at">df =</span> df1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p.adjust =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(geneset) <span class="sc">%&gt;%</span> <span class="fu">remove_rownames</span>(), <span class="at">color_legend_n =</span> <span class="dv">1</span>, <span class="at">color_legend_min =</span> <span class="dv">0</span>, <span class="at">color_legend_max =</span> <span class="fl">1.3</span>,<span class="at">rows =</span> Description, <span class="at">columns =</span> enrichment, <span class="at">values =</span> p.adjust, <span class="at">annotation_row =</span> geneset, <span class="at">gaps_row =</span> geneset, <span class="at">gaps_col =</span> enrichment</span>
<span id="cb40-1085"><a href="#cb40-1085" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb40-1086"><a href="#cb40-1086" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb40-1087"><a href="#cb40-1087" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1088"><a href="#cb40-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1089"><a href="#cb40-1089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1090"><a href="#cb40-1090" aria-hidden="true" tabindex="-1"></a><span class="fu">## Heirarchical clustering of all the DEGs</span></span>
<span id="cb40-1091"><a href="#cb40-1091" aria-hidden="true" tabindex="-1"></a>Main effect DEGs are in res1 (Genotype), res4 (Timepoint) objects whereas interaction specific DEGs are in res6 objects. res9 and res10 objects contain DEGs between KO and WT comparison for timepoints T0 and T12 respectively. </span>
<span id="cb40-1092"><a href="#cb40-1092" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1095"><a href="#cb40-1095" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1096"><a href="#cb40-1096" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(res1 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res4 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res6 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res9 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene), res10 <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)))</span>
<span id="cb40-1097"><a href="#cb40-1097" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb40-1098"><a href="#cb40-1098" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb40-1099"><a href="#cb40-1099" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">genotype =</span> <span class="fu">c</span>(<span class="st">"WT"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"KO"</span> <span class="ot">=</span> <span class="st">"blue"</span>)))</span>
<span id="cb40-1100"><a href="#cb40-1100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1101"><a href="#cb40-1101" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-1102"><a href="#cb40-1102" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-1103"><a href="#cb40-1103" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb40-1104"><a href="#cb40-1104" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb40-1105"><a href="#cb40-1105" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb40-1106"><a href="#cb40-1106" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb40-1107"><a href="#cb40-1107" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> F,</span>
<span id="cb40-1108"><a href="#cb40-1108" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> T,</span>
<span id="cb40-1109"><a href="#cb40-1109" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> T,</span>
<span id="cb40-1110"><a href="#cb40-1110" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_km =</span> <span class="dv">2</span>,</span>
<span id="cb40-1111"><a href="#cb40-1111" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_km =</span> <span class="dv">4</span>,</span>
<span id="cb40-1112"><a href="#cb40-1112" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),</span>
<span id="cb40-1113"><a href="#cb40-1113" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>),</span>
<span id="cb40-1114"><a href="#cb40-1114" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span></span>
<span id="cb40-1115"><a href="#cb40-1115" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb40-1116"><a href="#cb40-1116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1117"><a href="#cb40-1117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1118"><a href="#cb40-1118" aria-hidden="true" tabindex="-1"></a>From the heatmap, we see that clustering of samples based on the genotype then the timepoint. Genotype effects drive major source of variation compared to the timepoint. I roughly divided the DEGs obtained from various comparisons into 4 clusters -</span>
<span id="cb40-1119"><a href="#cb40-1119" aria-hidden="true" tabindex="-1"></a>Cluster 2 has genes active in T0 in both genotypes</span>
<span id="cb40-1120"><a href="#cb40-1120" aria-hidden="true" tabindex="-1"></a>Cluster 1 has genes active in KO genotype in both timepoints</span>
<span id="cb40-1121"><a href="#cb40-1121" aria-hidden="true" tabindex="-1"></a>Cluster 4 has genes active in WT genotype in both timepoints</span>
<span id="cb40-1122"><a href="#cb40-1122" aria-hidden="true" tabindex="-1"></a>Cluster 3 has genes active in T12 in both genotypes</span>
<span id="cb40-1123"><a href="#cb40-1123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1124"><a href="#cb40-1124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1125"><a href="#cb40-1125" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plot the expression pattern of the DEGs obtained from all the above comparisons</span></span>
<span id="cb40-1126"><a href="#cb40-1126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1129"><a href="#cb40-1129" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1130"><a href="#cb40-1130" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb40-1131"><a href="#cb40-1131" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb40-1132"><a href="#cb40-1132" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Grouping all the DEGs into clusters based on expression pattern trends"</span></span>
<span id="cb40-1133"><a href="#cb40-1133" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Clusters and their associated genes"</span></span>
<span id="cb40-1134"><a href="#cb40-1134" aria-hidden="true" tabindex="-1"></a>ma <span class="ot">=</span> <span class="fu">assay</span>(<span class="fu">vst</span>(dds))[significant_genes,]</span>
<span id="cb40-1135"><a href="#cb40-1135" aria-hidden="true" tabindex="-1"></a>coldata1 <span class="ot">=</span> coldata <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-1136"><a href="#cb40-1136" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">paste0</span>(genotype,<span class="st">"_"</span>, time_point),</span>
<span id="cb40-1137"><a href="#cb40-1137" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">group =</span> <span class="fu">factor</span>(group))</span>
<span id="cb40-1138"><a href="#cb40-1138" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">degPatterns</span>(ma, </span>
<span id="cb40-1139"><a href="#cb40-1139" aria-hidden="true" tabindex="-1"></a>                        <span class="at">metadata =</span> coldata1, </span>
<span id="cb40-1140"><a href="#cb40-1140" aria-hidden="true" tabindex="-1"></a>                        <span class="at">time =</span> <span class="st">"group"</span>, <span class="at">plot =</span> T)</span>
<span id="cb40-1141"><a href="#cb40-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1142"><a href="#cb40-1142" aria-hidden="true" tabindex="-1"></a>clusters<span class="sc">$</span>df <span class="sc">%&gt;%</span> <span class="fu">gt_preview</span>(<span class="at">top_n =</span> <span class="dv">10</span>)</span>
<span id="cb40-1143"><a href="#cb40-1143" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1144"><a href="#cb40-1144" aria-hidden="true" tabindex="-1"></a> The output is the list of genes along with cluster IDs that facilitate filtering of genes with a particular expression pattern trend. </span>
<span id="cb40-1145"><a href="#cb40-1145" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb40-1146"><a href="#cb40-1146" aria-hidden="true" tabindex="-1"></a><span class="fu">### Pathway analysis of various clusters obtained from the above analysis</span></span>
<span id="cb40-1147"><a href="#cb40-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1150"><a href="#cb40-1150" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb40-1151"><a href="#cb40-1151" aria-hidden="true" tabindex="-1"></a>hg_gs <span class="ot">=</span> geneset<span class="sc">::</span><span class="fu">getMsigdb</span>(<span class="at">org =</span> <span class="st">"human"</span>, <span class="at">category =</span> <span class="st">"H"</span>)</span>
<span id="cb40-1152"><a href="#cb40-1152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1153"><a href="#cb40-1153" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb40-1154"><a href="#cb40-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1155"><a href="#cb40-1155" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">unique</span>(clusters<span class="sc">$</span>df<span class="sc">$</span>cluster), <span class="cf">function</span>(x) {</span>
<span id="cb40-1156"><a href="#cb40-1156" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb40-1157"><a href="#cb40-1157" aria-hidden="true" tabindex="-1"></a>  go_results1 <span class="ot">=</span> <span class="fu">safely</span>(genORA)(clusters<span class="sc">$</span>df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cluster <span class="sc">==</span> x) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes),</span>
<span id="cb40-1158"><a href="#cb40-1158" aria-hidden="true" tabindex="-1"></a>                               <span class="at">geneset =</span> hg_gs,</span>
<span id="cb40-1159"><a href="#cb40-1159" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#p_cutoff = 1, #to retrieve the whole output</span></span>
<span id="cb40-1160"><a href="#cb40-1160" aria-hidden="true" tabindex="-1"></a>                               <span class="co">#q_cutoff = 1, #to retrieve the whole output</span></span>
<span id="cb40-1161"><a href="#cb40-1161" aria-hidden="true" tabindex="-1"></a>                               <span class="at">universe =</span> clusters<span class="sc">$</span>df <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genes))<span class="sc">$</span>result</span>
<span id="cb40-1162"><a href="#cb40-1162" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb40-1163"><a href="#cb40-1163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-1164"><a href="#cb40-1164" aria-hidden="true" tabindex="-1"></a><span class="fu">list_rbind</span>(df1, <span class="at">names_to =</span> <span class="st">"cluster"</span>) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb40-1165"><a href="#cb40-1165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb40-1166"><a href="#cb40-1166" aria-hidden="true" tabindex="-1"></a>No pathways associated with each of the cluster genes. </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>