<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>SampleAnalysis_Mansingh2024_matrix</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/clipboard/clipboard.min.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/quarto.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/popper.min.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/anchor.min.js"></script>
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="DGE_pathwayanalysis_Mansingh2024countmatrix_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#loading-the-required-libraries-and-importing-required-files" id="toc-loading-the-required-libraries-and-importing-required-files" class="nav-link active" data-scroll-target="#loading-the-required-libraries-and-importing-required-files">Loading the required libraries and importing required files</a>
  <ul class="collapse">
  <li><a href="#loading-the-required-libraries" id="toc-loading-the-required-libraries" class="nav-link" data-scroll-target="#loading-the-required-libraries">Loading the required libraries</a></li>
  <li><a href="#importing-the-counts-matrix" id="toc-importing-the-counts-matrix" class="nav-link" data-scroll-target="#importing-the-counts-matrix">Importing the counts matrix</a></li>
  <li><a href="#loading-the-metadata-file" id="toc-loading-the-metadata-file" class="nav-link" data-scroll-target="#loading-the-metadata-file">Loading the metadata file</a></li>
  </ul></li>
  <li><a href="#creation-and-exploration-of-dds-object" id="toc-creation-and-exploration-of-dds-object" class="nav-link" data-scroll-target="#creation-and-exploration-of-dds-object">Creation and exploration of dds object</a>
  <ul class="collapse">
  <li><a href="#pca-plot" id="toc-pca-plot" class="nav-link" data-scroll-target="#pca-plot">PCA plot</a></li>
  <li><a href="#sample-correlation-heatmap-using-vst-counts" id="toc-sample-correlation-heatmap-using-vst-counts" class="nav-link" data-scroll-target="#sample-correlation-heatmap-using-vst-counts">Sample correlation heatmap using vst counts</a></li>
  </ul></li>
  <li><a href="#differential-gene-expression-analysis" id="toc-differential-gene-expression-analysis" class="nav-link" data-scroll-target="#differential-gene-expression-analysis">Differential gene expression analysis</a>
  <ul class="collapse">
  <li><a href="#tissue-specific-degs-the-main-effect" id="toc-tissue-specific-degs-the-main-effect" class="nav-link" data-scroll-target="#tissue-specific-degs-the-main-effect">Tissue-specific DEGs (the main effect)</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">SampleAnalysis_Mansingh2024_matrix</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Aim:</strong> To perform differential gene expression analysis for different contrasts</p>
<section id="loading-the-required-libraries-and-importing-required-files" class="level2">
<h2 class="anchored" data-anchor-id="loading-the-required-libraries-and-importing-required-files">Loading the required libraries and importing required files</h2>
<section id="loading-the-required-libraries" class="level3">
<h3 class="anchored" data-anchor-id="loading-the-required-libraries">Loading the required libraries</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co">#for data wrangling</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2) <span class="co">#for DEG analysis</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genekitr) <span class="co">#for pathway analysis</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr) <span class="co">#for data import</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(eatTools) <span class="co">#for rbind functions</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(strex) <span class="co">#for string related functions</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr) <span class="co">#for beautiful tables</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt) <span class="co">#for customized table layouts</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap) <span class="co">#for beautiful heatmaps</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyheatmaps) <span class="co">#for heatmap generation from dataframe</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co">#for adding multiple figures to a plot</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer) <span class="co">#for color schemas</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) <span class="co">#for plotting</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix) <span class="co">#to explore complicated design in the DESeq2 formula</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="importing-the-counts-matrix" class="level3">
<h3 class="anchored" data-anchor-id="importing-the-counts-matrix">Importing the counts matrix</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#loading the length-scaled counts tsv file</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"~/Documents/BiostateAI/data/Mansingh2024_expression_matrix.txt"</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span>T) </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#converting NA into 0 and round off the counts</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>cts[<span class="fu">is.na</span>(cts)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.numeric, <span class="fu">list</span>(round))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="loading-the-metadata-file" class="level3">
<h3 class="anchored" data-anchor-id="loading-the-metadata-file">Loading the metadata file</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">str_detect</span>( <span class="fu">colnames</span>(cts) , <span class="st">'CTRL'</span> )  )</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">revalue</span>( genotype , <span class="fu">c</span>( <span class="st">'TRUE'</span><span class="ot">=</span><span class="st">'WT'</span> , <span class="st">'FALSE'</span> <span class="ot">=</span> <span class="st">'KO'</span>  )  ) <span class="co">#TRUE is WT, FALSE is CTRL</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>time_point <span class="ot">=</span> <span class="fu">factor</span>( <span class="fu">substring</span>(<span class="fu">colnames</span>(cts), <span class="fu">regexpr</span>(<span class="st">"_Z"</span>,<span class="fu">colnames</span>(cts)) <span class="sc">+</span> <span class="dv">2</span>) , <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>,<span class="st">'T4'</span>,<span class="st">'T8'</span>,<span class="st">'T12'</span>,<span class="st">'T16'</span>,<span class="st">'T20'</span>))</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> <span class="fu">sapply</span>( <span class="fu">strsplit</span>( <span class="fu">colnames</span>( cts ) , <span class="st">"_"</span>) , <span class="st">'['</span> , <span class="dv">5</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>replicate <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(cts) , <span class="dv">1</span>, <span class="fu">nchar</span>( <span class="st">"HL3YYBGX5"</span> ) ) </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( <span class="at">sample =</span> <span class="fu">colnames</span>(cts) ,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">genotype =</span> genotype ,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">time_point =</span> time_point,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mouse =</span> mouse,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">replicate =</span> replicate</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>                       )</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> coldata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(time_point <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>time_point <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>time_point, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">select</span>(coldata<span class="sc">$</span>sample)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Metadata file"</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Sample counts for genotype and timepoint combinations"</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="zqkzeabdlb" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#zqkzeabdlb table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#zqkzeabdlb thead, #zqkzeabdlb tbody, #zqkzeabdlb tfoot, #zqkzeabdlb tr, #zqkzeabdlb td, #zqkzeabdlb th {
  border-style: none;
}

#zqkzeabdlb p {
  margin: 0;
  padding: 0;
}

#zqkzeabdlb .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#zqkzeabdlb .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#zqkzeabdlb .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#zqkzeabdlb .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#zqkzeabdlb .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zqkzeabdlb .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zqkzeabdlb .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zqkzeabdlb .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#zqkzeabdlb .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#zqkzeabdlb .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#zqkzeabdlb .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#zqkzeabdlb .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#zqkzeabdlb .gt_spanner_row {
  border-bottom-style: hidden;
}

#zqkzeabdlb .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#zqkzeabdlb .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#zqkzeabdlb .gt_from_md > :first-child {
  margin-top: 0;
}

#zqkzeabdlb .gt_from_md > :last-child {
  margin-bottom: 0;
}

#zqkzeabdlb .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#zqkzeabdlb .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#zqkzeabdlb .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#zqkzeabdlb .gt_row_group_first td {
  border-top-width: 2px;
}

#zqkzeabdlb .gt_row_group_first th {
  border-top-width: 2px;
}

#zqkzeabdlb .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zqkzeabdlb .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#zqkzeabdlb .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#zqkzeabdlb .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zqkzeabdlb .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zqkzeabdlb .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#zqkzeabdlb .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#zqkzeabdlb .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#zqkzeabdlb .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zqkzeabdlb .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zqkzeabdlb .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zqkzeabdlb .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zqkzeabdlb .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zqkzeabdlb .gt_left {
  text-align: left;
}

#zqkzeabdlb .gt_center {
  text-align: center;
}

#zqkzeabdlb .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#zqkzeabdlb .gt_font_normal {
  font-weight: normal;
}

#zqkzeabdlb .gt_font_bold {
  font-weight: bold;
}

#zqkzeabdlb .gt_font_italic {
  font-style: italic;
}

#zqkzeabdlb .gt_super {
  font-size: 65%;
}

#zqkzeabdlb .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#zqkzeabdlb .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#zqkzeabdlb .gt_indent_1 {
  text-indent: 5px;
}

#zqkzeabdlb .gt_indent_2 {
  text-indent: 10px;
}

#zqkzeabdlb .gt_indent_3 {
  text-indent: 15px;
}

#zqkzeabdlb .gt_indent_4 {
  text-indent: 20px;
}

#zqkzeabdlb .gt_indent_5 {
  text-indent: 25px;
}

#zqkzeabdlb .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#zqkzeabdlb div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="sample">sample</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="genotype">genotype</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="time_point">time_point</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="mouse">mouse</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="replicate">replicate</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_6_4__23_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">23</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_7__19_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">19</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_2__6_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">6</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_8__22_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">22</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_2__6_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">6</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_6_4__23_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">23</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_5_5__2_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">2</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_7__19_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">19</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_5_5__2_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">2</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_2__6_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">6</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_5_6__5_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">5</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_5_6__5_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">5</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_8__22_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">22</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_5_6__5_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">5</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_1__3_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">3</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_1__1_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">1</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_8__24_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">24</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_7__21_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">21</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_6_3__20_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">20</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_7__21_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">21</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_2__4_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">4</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_7__21_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">21</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_7_1__3_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">3</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_8__24_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">24</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_5_5__2_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">2</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_6_3__20_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">20</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_8__22_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">22</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_7__19_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">19</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_1__1_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">1</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_7_8__24_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">24</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_4_2__4_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">4</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_4_1__1_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">1</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YTBGX5_6_3__20_CTRL_ZT12</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">20</td>
<td headers="replicate" class="gt_row gt_left">HL3YTBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_4_2__4_MKO_ZT0</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">4</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL3YYBGX5_6_4__23_MKO_ZT12</td>
<td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="mouse" class="gt_row gt_right">23</td>
<td headers="replicate" class="gt_row gt_left">HL3YYBGX5</td></tr>
    <tr><td headers="sample" class="gt_row gt_left">HL7CHBGX5_7_1__3_CTRL_ZT0</td>
<td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="mouse" class="gt_row gt_right">3</td>
<td headers="replicate" class="gt_row gt_left">HL7CHBGX5</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(genotype,time_point) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="dhwsrvmcze" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#dhwsrvmcze table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#dhwsrvmcze thead, #dhwsrvmcze tbody, #dhwsrvmcze tfoot, #dhwsrvmcze tr, #dhwsrvmcze td, #dhwsrvmcze th {
  border-style: none;
}

#dhwsrvmcze p {
  margin: 0;
  padding: 0;
}

#dhwsrvmcze .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#dhwsrvmcze .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#dhwsrvmcze .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#dhwsrvmcze .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#dhwsrvmcze .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dhwsrvmcze .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dhwsrvmcze .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dhwsrvmcze .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#dhwsrvmcze .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#dhwsrvmcze .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#dhwsrvmcze .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#dhwsrvmcze .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#dhwsrvmcze .gt_spanner_row {
  border-bottom-style: hidden;
}

#dhwsrvmcze .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#dhwsrvmcze .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#dhwsrvmcze .gt_from_md > :first-child {
  margin-top: 0;
}

#dhwsrvmcze .gt_from_md > :last-child {
  margin-bottom: 0;
}

#dhwsrvmcze .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#dhwsrvmcze .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#dhwsrvmcze .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#dhwsrvmcze .gt_row_group_first td {
  border-top-width: 2px;
}

#dhwsrvmcze .gt_row_group_first th {
  border-top-width: 2px;
}

#dhwsrvmcze .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhwsrvmcze .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#dhwsrvmcze .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#dhwsrvmcze .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dhwsrvmcze .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhwsrvmcze .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#dhwsrvmcze .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#dhwsrvmcze .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#dhwsrvmcze .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dhwsrvmcze .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dhwsrvmcze .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhwsrvmcze .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dhwsrvmcze .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dhwsrvmcze .gt_left {
  text-align: left;
}

#dhwsrvmcze .gt_center {
  text-align: center;
}

#dhwsrvmcze .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#dhwsrvmcze .gt_font_normal {
  font-weight: normal;
}

#dhwsrvmcze .gt_font_bold {
  font-weight: bold;
}

#dhwsrvmcze .gt_font_italic {
  font-style: italic;
}

#dhwsrvmcze .gt_super {
  font-size: 65%;
}

#dhwsrvmcze .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#dhwsrvmcze .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#dhwsrvmcze .gt_indent_1 {
  text-indent: 5px;
}

#dhwsrvmcze .gt_indent_2 {
  text-indent: 10px;
}

#dhwsrvmcze .gt_indent_3 {
  text-indent: 15px;
}

#dhwsrvmcze .gt_indent_4 {
  text-indent: 20px;
}

#dhwsrvmcze .gt_indent_5 {
  text-indent: 25px;
}

#dhwsrvmcze .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#dhwsrvmcze div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="genotype">genotype</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="time_point">time_point</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
    <tr><td headers="genotype" class="gt_row gt_center">KO</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
    <tr><td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T0</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
    <tr><td headers="genotype" class="gt_row gt_center">WT</td>
<td headers="time_point" class="gt_row gt_center">T12</td>
<td headers="n" class="gt_row gt_right">9</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
</section>
</section>
<section id="creation-and-exploration-of-dds-object" class="level2">
<h2 class="anchored" data-anchor-id="creation-and-exploration-of-dds-object">Creation and exploration of dds object</h2>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#creation of dds object without prefiltering</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#paired design tissue + time if need to adjust for confounders (time in this case)</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(<span class="at">sampleData =</span> coldata,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">designFormula =</span> <span class="sc">~</span> genotype)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#print(vd$plotlist)</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">relevel</span>(dds<span class="sc">$</span>genotype, <span class="at">ref =</span> <span class="st">"WT"</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds) <span class="co">#Main effects for genotype #Expression affected by genotype not correcting for  the timepoint effects</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>)) <span class="co">#FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram without prefiltering of counts"</span>)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Conversion%20to%20a%20DESeq%20object-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">p-value histogram without prefiltering of counts</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Conversion%20to%20a%20DESeq%20object-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Genes with high p-values do seem to have lower gene expression</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>p-value histogram distribution looks bimodal. Therefore, the counts needs prefiltering before DEG analysis. High expressing count genes fall into low p-value bin which is great.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dds &lt;- dds[rowSums(counts(dds) &gt; 10,)] #opted out</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dds &lt;- DESeq(dds)</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#resultsNames(dds)</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- results(dds, contrast = c("genotype", "FALSE", "TRUE")) #FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "p-value histogram after prefiltering of counts"</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genes with high p-values do seem to have lower gene expression"</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram without prefiltering of counts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Prefiltering%20of%20counts%20using%20the%20dds%20object-1.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Prefiltering%20of%20counts%20using%20the%20dds%20object-2.png" class="img-fluid"></p>
</div>
</div>
<p>Various contrasts can be identified using <strong>resultNames(dds)</strong> function</p>
<p>The p-value histogram after prefiltering of counts is not looking good. I havent done prefiltering. Supposedly, DESeq2 takes care of the low counts while identifying DEGs. Refer this link <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/" class="uri">http://varianceexplained.org/statistics/interpreting-pvalue-histogram/</a> for more info.</p>
<section id="pca-plot" class="level3">
<h3 class="anchored" data-anchor-id="pca-plot">PCA plot</h3>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>vstcounts <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> T)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>normalized_cts <span class="ot">&lt;-</span> <span class="fu">assay</span>(vstcounts) <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>pca_prcomp <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(normalized_cts), <span class="at">center =</span> T, <span class="at">scale =</span> F)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co">#names(pca_prcomp)</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>PC1_and_PC2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">1</span>],</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">PC2 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">type =</span> <span class="fu">rownames</span>(pca_prcomp<span class="sc">$</span>x))</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co">#using the complete normalized matrix</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(PC1_and_PC2, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">col=</span>type)) <span class="sc">+</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> type), <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>()</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co">#using variable features</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vstcounts, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/PCA%20plot-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">PCA with sample IDs</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/PCA%20plot-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">PCA when colored by genotype and timepoint</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="sample-correlation-heatmap-using-vst-counts" class="level3">
<h3 class="anchored" data-anchor-id="sample-correlation-heatmap-using-vst-counts">Sample correlation heatmap using vst counts</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>sampleCor <span class="ot">&lt;-</span> <span class="fu">cor</span>(normalized_cts)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(normalized_cts))</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDist)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>blueColours <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Blues"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(blueColours))(<span class="dv">255</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(sampleDistMatrix,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clustering_distance_cols =</span> sampleDist, </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Sample%20correlation%20heatmap%20using%20vst%20counts-1.png" class="img-fluid"></p>
</div>
</div>
</section>
</section>
<section id="differential-gene-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="differential-gene-expression-analysis">Differential gene expression analysis</h2>
<section id="tissue-specific-degs-the-main-effect" class="level3">
<h3 class="anchored" data-anchor-id="tissue-specific-degs-the-main-effect">Tissue-specific DEGs (the main effect)</h3>
<p>Here, I am identifying tissue-specific (liver versus heart) DEGs. These DEGs would be liver- or heart-enriched.</p>
<section id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype" class="level4">
<h4 class="anchored" data-anchor-id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-genotype">Comparison of Wald versus LRT test DEGs [Main effects due to genotype]</h4>
<p><strong>DEGs can be defined genes passing the threshold: abs(logFC) &gt; 1 &amp; padj &lt; 0.01</strong></p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#not accounting for timepoint</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Wald"</span>,</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>res1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> time_point)</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> res2 <span class="sc">%&gt;%</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-LRT"</span>,</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>res2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res1,res2)</span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="shzrwhrbgl" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#shzrwhrbgl table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#shzrwhrbgl thead, #shzrwhrbgl tbody, #shzrwhrbgl tfoot, #shzrwhrbgl tr, #shzrwhrbgl td, #shzrwhrbgl th {
  border-style: none;
}

#shzrwhrbgl p {
  margin: 0;
  padding: 0;
}

#shzrwhrbgl .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#shzrwhrbgl .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#shzrwhrbgl .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#shzrwhrbgl .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#shzrwhrbgl .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#shzrwhrbgl .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#shzrwhrbgl .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#shzrwhrbgl .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#shzrwhrbgl .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#shzrwhrbgl .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#shzrwhrbgl .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#shzrwhrbgl .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#shzrwhrbgl .gt_spanner_row {
  border-bottom-style: hidden;
}

#shzrwhrbgl .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#shzrwhrbgl .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#shzrwhrbgl .gt_from_md > :first-child {
  margin-top: 0;
}

#shzrwhrbgl .gt_from_md > :last-child {
  margin-bottom: 0;
}

#shzrwhrbgl .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#shzrwhrbgl .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#shzrwhrbgl .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#shzrwhrbgl .gt_row_group_first td {
  border-top-width: 2px;
}

#shzrwhrbgl .gt_row_group_first th {
  border-top-width: 2px;
}

#shzrwhrbgl .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#shzrwhrbgl .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#shzrwhrbgl .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#shzrwhrbgl .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#shzrwhrbgl .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#shzrwhrbgl .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#shzrwhrbgl .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#shzrwhrbgl .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#shzrwhrbgl .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#shzrwhrbgl .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#shzrwhrbgl .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#shzrwhrbgl .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#shzrwhrbgl .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#shzrwhrbgl .gt_left {
  text-align: left;
}

#shzrwhrbgl .gt_center {
  text-align: center;
}

#shzrwhrbgl .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#shzrwhrbgl .gt_font_normal {
  font-weight: normal;
}

#shzrwhrbgl .gt_font_bold {
  font-weight: bold;
}

#shzrwhrbgl .gt_font_italic {
  font-style: italic;
}

#shzrwhrbgl .gt_super {
  font-size: 65%;
}

#shzrwhrbgl .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#shzrwhrbgl .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#shzrwhrbgl .gt_indent_1 {
  text-indent: 5px;
}

#shzrwhrbgl .gt_indent_2 {
  text-indent: 10px;
}

#shzrwhrbgl .gt_indent_3 {
  text-indent: 15px;
}

#shzrwhrbgl .gt_indent_4 {
  text-indent: 20px;
}

#shzrwhrbgl .gt_indent_5 {
  text-indent: 25px;
}

#shzrwhrbgl .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#shzrwhrbgl div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Wald</td>
<td headers="enrichment" class="gt_row gt_left">KO</td>
<td headers="n" class="gt_row gt_right">163</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Wald</td>
<td headers="enrichment" class="gt_row gt_left">WT</td>
<td headers="n" class="gt_row gt_right">169</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-Wald</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18252</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test [Main effects due to genotype]</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="iqlboxcyqh" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#iqlboxcyqh table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#iqlboxcyqh thead, #iqlboxcyqh tbody, #iqlboxcyqh tfoot, #iqlboxcyqh tr, #iqlboxcyqh td, #iqlboxcyqh th {
  border-style: none;
}

#iqlboxcyqh p {
  margin: 0;
  padding: 0;
}

#iqlboxcyqh .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#iqlboxcyqh .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#iqlboxcyqh .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#iqlboxcyqh .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#iqlboxcyqh .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#iqlboxcyqh .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#iqlboxcyqh .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#iqlboxcyqh .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#iqlboxcyqh .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#iqlboxcyqh .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#iqlboxcyqh .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#iqlboxcyqh .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#iqlboxcyqh .gt_spanner_row {
  border-bottom-style: hidden;
}

#iqlboxcyqh .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#iqlboxcyqh .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#iqlboxcyqh .gt_from_md > :first-child {
  margin-top: 0;
}

#iqlboxcyqh .gt_from_md > :last-child {
  margin-bottom: 0;
}

#iqlboxcyqh .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#iqlboxcyqh .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#iqlboxcyqh .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#iqlboxcyqh .gt_row_group_first td {
  border-top-width: 2px;
}

#iqlboxcyqh .gt_row_group_first th {
  border-top-width: 2px;
}

#iqlboxcyqh .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#iqlboxcyqh .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#iqlboxcyqh .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#iqlboxcyqh .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#iqlboxcyqh .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#iqlboxcyqh .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#iqlboxcyqh .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#iqlboxcyqh .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#iqlboxcyqh .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#iqlboxcyqh .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#iqlboxcyqh .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#iqlboxcyqh .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#iqlboxcyqh .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#iqlboxcyqh .gt_left {
  text-align: left;
}

#iqlboxcyqh .gt_center {
  text-align: center;
}

#iqlboxcyqh .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#iqlboxcyqh .gt_font_normal {
  font-weight: normal;
}

#iqlboxcyqh .gt_font_bold {
  font-weight: bold;
}

#iqlboxcyqh .gt_font_italic {
  font-style: italic;
}

#iqlboxcyqh .gt_super {
  font-size: 65%;
}

#iqlboxcyqh .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#iqlboxcyqh .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#iqlboxcyqh .gt_indent_1 {
  text-indent: 5px;
}

#iqlboxcyqh .gt_indent_2 {
  text-indent: 10px;
}

#iqlboxcyqh .gt_indent_3 {
  text-indent: 15px;
}

#iqlboxcyqh .gt_indent_4 {
  text-indent: 20px;
}

#iqlboxcyqh .gt_indent_5 {
  text-indent: 25px;
}

#iqlboxcyqh .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#iqlboxcyqh div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-LRT</td>
<td headers="enrichment" class="gt_row gt_left">KO</td>
<td headers="n" class="gt_row gt_right">177</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-LRT</td>
<td headers="enrichment" class="gt_row gt_left">WT</td>
<td headers="n" class="gt_row gt_right">182</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">KO_vs_WT-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18225</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test [Main effects due to genotype]</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Comparison%20of%20Wald%20test%20and%20LRT%20-%20genotype-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">DEGs obtained with Wald test [Main effects due to genotype]</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>DEGs with and without accounting for timepoint yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to genotype.</p>
</section>
<section id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point" class="level4">
<h4 class="anchored" data-anchor-id="comparison-of-wald-versus-lrt-test-degs-main-effects-due-to-time_point">Comparison of Wald versus LRT test DEGs [Main effects due to time_point]</h4>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> time_point) <span class="co">#not accounting for timepoint</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-Wald"</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>res4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> genotype)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> res5 <span class="sc">%&gt;%</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-LRT"</span>,</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>res5 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res4,res5)</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="amhwktizzo" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#amhwktizzo table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#amhwktizzo thead, #amhwktizzo tbody, #amhwktizzo tfoot, #amhwktizzo tr, #amhwktizzo td, #amhwktizzo th {
  border-style: none;
}

#amhwktizzo p {
  margin: 0;
  padding: 0;
}

#amhwktizzo .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#amhwktizzo .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#amhwktizzo .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#amhwktizzo .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#amhwktizzo .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#amhwktizzo .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#amhwktizzo .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#amhwktizzo .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#amhwktizzo .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#amhwktizzo .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#amhwktizzo .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#amhwktizzo .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#amhwktizzo .gt_spanner_row {
  border-bottom-style: hidden;
}

#amhwktizzo .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#amhwktizzo .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#amhwktizzo .gt_from_md > :first-child {
  margin-top: 0;
}

#amhwktizzo .gt_from_md > :last-child {
  margin-bottom: 0;
}

#amhwktizzo .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#amhwktizzo .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#amhwktizzo .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#amhwktizzo .gt_row_group_first td {
  border-top-width: 2px;
}

#amhwktizzo .gt_row_group_first th {
  border-top-width: 2px;
}

#amhwktizzo .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#amhwktizzo .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#amhwktizzo .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#amhwktizzo .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#amhwktizzo .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#amhwktizzo .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#amhwktizzo .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#amhwktizzo .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#amhwktizzo .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#amhwktizzo .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#amhwktizzo .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#amhwktizzo .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#amhwktizzo .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#amhwktizzo .gt_left {
  text-align: left;
}

#amhwktizzo .gt_center {
  text-align: center;
}

#amhwktizzo .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#amhwktizzo .gt_font_normal {
  font-weight: normal;
}

#amhwktizzo .gt_font_bold {
  font-weight: bold;
}

#amhwktizzo .gt_font_italic {
  font-style: italic;
}

#amhwktizzo .gt_super {
  font-size: 65%;
}

#amhwktizzo .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#amhwktizzo .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#amhwktizzo .gt_indent_1 {
  text-indent: 5px;
}

#amhwktizzo .gt_indent_2 {
  text-indent: 10px;
}

#amhwktizzo .gt_indent_3 {
  text-indent: 15px;
}

#amhwktizzo .gt_indent_4 {
  text-indent: 20px;
}

#amhwktizzo .gt_indent_5 {
  text-indent: 25px;
}

#amhwktizzo .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#amhwktizzo div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-Wald</td>
<td headers="enrichment" class="gt_row gt_left">T0</td>
<td headers="n" class="gt_row gt_right">231</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-Wald</td>
<td headers="enrichment" class="gt_row gt_left">T12</td>
<td headers="n" class="gt_row gt_right">113</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-Wald</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18240</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test [Main effects due to time_point]</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">

<div id="odlqqtmccy" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#odlqqtmccy table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#odlqqtmccy thead, #odlqqtmccy tbody, #odlqqtmccy tfoot, #odlqqtmccy tr, #odlqqtmccy td, #odlqqtmccy th {
  border-style: none;
}

#odlqqtmccy p {
  margin: 0;
  padding: 0;
}

#odlqqtmccy .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#odlqqtmccy .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#odlqqtmccy .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#odlqqtmccy .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#odlqqtmccy .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#odlqqtmccy .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#odlqqtmccy .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#odlqqtmccy .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#odlqqtmccy .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#odlqqtmccy .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#odlqqtmccy .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#odlqqtmccy .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#odlqqtmccy .gt_spanner_row {
  border-bottom-style: hidden;
}

#odlqqtmccy .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#odlqqtmccy .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#odlqqtmccy .gt_from_md > :first-child {
  margin-top: 0;
}

#odlqqtmccy .gt_from_md > :last-child {
  margin-bottom: 0;
}

#odlqqtmccy .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#odlqqtmccy .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#odlqqtmccy .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#odlqqtmccy .gt_row_group_first td {
  border-top-width: 2px;
}

#odlqqtmccy .gt_row_group_first th {
  border-top-width: 2px;
}

#odlqqtmccy .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#odlqqtmccy .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#odlqqtmccy .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#odlqqtmccy .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#odlqqtmccy .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#odlqqtmccy .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#odlqqtmccy .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#odlqqtmccy .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#odlqqtmccy .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#odlqqtmccy .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#odlqqtmccy .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#odlqqtmccy .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#odlqqtmccy .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#odlqqtmccy .gt_left {
  text-align: left;
}

#odlqqtmccy .gt_center {
  text-align: center;
}

#odlqqtmccy .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#odlqqtmccy .gt_font_normal {
  font-weight: normal;
}

#odlqqtmccy .gt_font_bold {
  font-weight: bold;
}

#odlqqtmccy .gt_font_italic {
  font-style: italic;
}

#odlqqtmccy .gt_super {
  font-size: 65%;
}

#odlqqtmccy .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#odlqqtmccy .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#odlqqtmccy .gt_indent_1 {
  text-indent: 5px;
}

#odlqqtmccy .gt_indent_2 {
  text-indent: 10px;
}

#odlqqtmccy .gt_indent_3 {
  text-indent: 15px;
}

#odlqqtmccy .gt_indent_4 {
  text-indent: 20px;
}

#odlqqtmccy .gt_indent_5 {
  text-indent: 25px;
}

#odlqqtmccy .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#odlqqtmccy div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="comparison">comparison</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="enrichment">enrichment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">T0</td>
<td headers="n" class="gt_row gt_right">242</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">T12</td>
<td headers="n" class="gt_row gt_right">124</td></tr>
    <tr><td headers="comparison" class="gt_row gt_left">T12_vs_T0-LRT</td>
<td headers="enrichment" class="gt_row gt_left">non-sig</td>
<td headers="n" class="gt_row gt_right">18763</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<p>DEGs obtained with Wald test [Main effects due to time_point]</p>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Comparison%20of%20Wald%20test%20and%20LRT%20-%20Timepoint-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">DEGs obtained with Wald test [Main effects due to time_point]</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
<p>DEGs with and without accounting for genotype yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to timepoints.</p>
</section>
<section id="visualization-of-tissue-specific-degs-using-deseq2" class="level4">
<h4 class="anchored" data-anchor-id="visualization-of-tissue-specific-degs-using-deseq2">Visualization of tissue-specific DEGs using DESeq2</h4>
<p>I am going with res1 object that uses Wald test to identify main effects due to genotype.</p>
<div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res1,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res1<span class="sc">$</span>gene,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"Liver vs Heart comparison [Paired-design]"</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res1<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20tissue-specific%20DEGs-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Identification%20of%20tissue-specific%20DEGs-2.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Tissue-specific DEGs agnostic of the sampling timepoint</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="expression-pattern-of-top-tissue-specific-degs" class="level4">
<h4 class="anchored" data-anchor-id="expression-pattern-of-top-tissue-specific-degs">Expression pattern of top tissue-specific DEGs</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&gt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">tissue =</span> <span class="fu">c</span>(<span class="st">"TRUE"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"FALSE"</span>, <span class="st">"blue"</span>)))</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20tissue-specific%20DEGs-1.png" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&gt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DGE_pathwayanalysis_Mansingh2024countmatrix_files/figure-html/Expression%20pattern%20of%20top%20tissue-specific%20DEGs-2.png" class="img-fluid"></p>
</div>
</div>
<!-- -->

</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb15" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "SampleAnalysis_Mansingh2024_matrix"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">    eval: true</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">    warning: false</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co">    message: false</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-height: 7</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-width: 7</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-format: png</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-dpi: 300</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>**Aim:** To perform differential gene expression analysis for different contrasts</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Loading the required libraries and importing required files</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="fu">### Loading the required libraries</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Loading libraries}</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co">#for data wrangling</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2) <span class="co">#for DEG analysis</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genekitr) <span class="co">#for pathway analysis</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr) <span class="co">#for data import</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(eatTools) <span class="co">#for rbind functions</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(strex) <span class="co">#for string related functions</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr) <span class="co">#for beautiful tables</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt) <span class="co">#for customized table layouts</span></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap) <span class="co">#for beautiful heatmaps</span></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyheatmaps) <span class="co">#for heatmap generation from dataframe</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co">#for adding multiple figures to a plot</span></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer) <span class="co">#for color schemas</span></span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) <span class="co">#for plotting</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix) <span class="co">#to explore complicated design in the DESeq2 formula</span></span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UpSetR)</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a><span class="fu">### Importing the counts matrix</span></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Importing the counts and conversion to a DESeq object}</span></span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a><span class="co">#loading the length-scaled counts tsv file</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"~/Documents/BiostateAI/data/Mansingh2024_expression_matrix.txt"</span>,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span>T) </span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a><span class="co">#converting NA into 0 and round off the counts</span></span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>cts[<span class="fu">is.na</span>(cts)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.numeric, <span class="fu">list</span>(round))</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a><span class="fu">### Loading the metadata file</span></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Import metadata file}</span></span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">str_detect</span>( <span class="fu">colnames</span>(cts) , <span class="st">'CTRL'</span> )  )</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">=</span> plyr<span class="sc">::</span><span class="fu">revalue</span>( genotype , <span class="fu">c</span>( <span class="st">'TRUE'</span><span class="ot">=</span><span class="st">'WT'</span> , <span class="st">'FALSE'</span> <span class="ot">=</span> <span class="st">'KO'</span>  )  ) <span class="co">#TRUE is WT, FALSE is CTRL</span></span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>time_point <span class="ot">=</span> <span class="fu">factor</span>( <span class="fu">substring</span>(<span class="fu">colnames</span>(cts), <span class="fu">regexpr</span>(<span class="st">"_Z"</span>,<span class="fu">colnames</span>(cts)) <span class="sc">+</span> <span class="dv">2</span>) , <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>,<span class="st">'T4'</span>,<span class="st">'T8'</span>,<span class="st">'T12'</span>,<span class="st">'T16'</span>,<span class="st">'T20'</span>))</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">=</span> <span class="fu">sapply</span>( <span class="fu">strsplit</span>( <span class="fu">colnames</span>( cts ) , <span class="st">"_"</span>) , <span class="st">'['</span> , <span class="dv">5</span>)</span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>replicate <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">colnames</span>(cts) , <span class="dv">1</span>, <span class="fu">nchar</span>( <span class="st">"HL3YYBGX5"</span> ) ) </span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( <span class="at">sample =</span> <span class="fu">colnames</span>(cts) ,</span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a>                       <span class="at">genotype =</span> genotype ,</span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a>                       <span class="at">time_point =</span> time_point,</span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mouse =</span> mouse,</span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a>                       <span class="at">replicate =</span> replicate</span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a>                       )</span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> coldata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(time_point <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>time_point <span class="ot">&lt;-</span> <span class="fu">factor</span>(coldata<span class="sc">$</span>time_point, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"T0"</span>, <span class="st">"T12"</span>))</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts <span class="sc">%&gt;%</span> <span class="fu">select</span>(coldata<span class="sc">$</span>sample)</span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 2</span></span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Metadata file"</span></span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Sample counts for genotype and timepoint combinations"</span></span>
<span id="cb15-82"><a href="#cb15-82" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb15-83"><a href="#cb15-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-84"><a href="#cb15-84" aria-hidden="true" tabindex="-1"></a>coldata <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(genotype,time_point) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb15-85"><a href="#cb15-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-86"><a href="#cb15-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-87"><a href="#cb15-87" aria-hidden="true" tabindex="-1"></a><span class="fu">## Creation and exploration of dds object</span></span>
<span id="cb15-88"><a href="#cb15-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-89"><a href="#cb15-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Conversion to a DESeq object}</span></span>
<span id="cb15-90"><a href="#cb15-90" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-91"><a href="#cb15-91" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-92"><a href="#cb15-92" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "p-value histogram without prefiltering of counts"</span></span>
<span id="cb15-93"><a href="#cb15-93" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genes with high p-values do seem to have lower gene expression"</span></span>
<span id="cb15-94"><a href="#cb15-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-95"><a href="#cb15-95" aria-hidden="true" tabindex="-1"></a><span class="co">#creation of dds object without prefiltering</span></span>
<span id="cb15-96"><a href="#cb15-96" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb15-97"><a href="#cb15-97" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb15-98"><a href="#cb15-98" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#paired design tissue + time if need to adjust for confounders (time in this case)</span></span>
<span id="cb15-99"><a href="#cb15-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-100"><a href="#cb15-100" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(<span class="at">sampleData =</span> coldata,</span>
<span id="cb15-101"><a href="#cb15-101" aria-hidden="true" tabindex="-1"></a>                      <span class="at">designFormula =</span> <span class="sc">~</span> genotype)</span>
<span id="cb15-102"><a href="#cb15-102" aria-hidden="true" tabindex="-1"></a><span class="co">#print(vd$plotlist)</span></span>
<span id="cb15-103"><a href="#cb15-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-104"><a href="#cb15-104" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">relevel</span>(dds<span class="sc">$</span>genotype, <span class="at">ref =</span> <span class="st">"WT"</span>)</span>
<span id="cb15-105"><a href="#cb15-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-106"><a href="#cb15-106" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds) <span class="co">#Main effects for genotype #Expression affected by genotype not correcting for  the timepoint effects</span></span>
<span id="cb15-107"><a href="#cb15-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-108"><a href="#cb15-108" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>)) <span class="co">#FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb15-109"><a href="#cb15-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-110"><a href="#cb15-110" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-111"><a href="#cb15-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb15-112"><a href="#cb15-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb15-113"><a href="#cb15-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb15-114"><a href="#cb15-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram without prefiltering of counts"</span>)</span>
<span id="cb15-115"><a href="#cb15-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-116"><a href="#cb15-116" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-117"><a href="#cb15-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-118"><a href="#cb15-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb15-119"><a href="#cb15-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-120"><a href="#cb15-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb15-121"><a href="#cb15-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb15-122"><a href="#cb15-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-123"><a href="#cb15-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-124"><a href="#cb15-124" aria-hidden="true" tabindex="-1"></a>p-value histogram distribution looks bimodal. Therefore, the counts needs prefiltering before DEG analysis. High expressing count genes fall into low p-value bin which is great. </span>
<span id="cb15-125"><a href="#cb15-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-126"><a href="#cb15-126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Prefiltering of counts using the dds object}</span></span>
<span id="cb15-127"><a href="#cb15-127" aria-hidden="true" tabindex="-1"></a><span class="co">#dds &lt;- dds[rowSums(counts(dds) &gt; 10,)] #opted out</span></span>
<span id="cb15-128"><a href="#cb15-128" aria-hidden="true" tabindex="-1"></a><span class="co">#dds &lt;- DESeq(dds)</span></span>
<span id="cb15-129"><a href="#cb15-129" aria-hidden="true" tabindex="-1"></a><span class="co">#resultsNames(dds)</span></span>
<span id="cb15-130"><a href="#cb15-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-131"><a href="#cb15-131" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- results(dds, contrast = c("genotype", "FALSE", "TRUE")) #FALSE is KO, WT is TRUE and is in the denominator</span></span>
<span id="cb15-132"><a href="#cb15-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-133"><a href="#cb15-133" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-134"><a href="#cb15-134" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-135"><a href="#cb15-135" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "p-value histogram after prefiltering of counts"</span></span>
<span id="cb15-136"><a href="#cb15-136" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Genes with high p-values do seem to have lower gene expression"</span></span>
<span id="cb15-137"><a href="#cb15-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-138"><a href="#cb15-138" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-139"><a href="#cb15-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span></span>
<span id="cb15-140"><a href="#cb15-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb15-141"><a href="#cb15-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb15-142"><a href="#cb15-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"p-value histogram without prefiltering of counts"</span>)</span>
<span id="cb15-143"><a href="#cb15-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-144"><a href="#cb15-144" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-145"><a href="#cb15-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-146"><a href="#cb15-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb15-147"><a href="#cb15-147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pvalue_bin =</span> <span class="fu">if_else</span>(pvalue <span class="sc">&gt;</span> <span class="fl">0.75</span>, <span class="st">"high"</span>, <span class="st">"low"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-148"><a href="#cb15-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pvalue_bin, <span class="at">y =</span> <span class="fu">log2</span>(baseMean))) <span class="sc">+</span> </span>
<span id="cb15-149"><a href="#cb15-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb15-150"><a href="#cb15-150" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-151"><a href="#cb15-151" aria-hidden="true" tabindex="-1"></a>Various contrasts can be identified using **resultNames(dds)** function</span>
<span id="cb15-152"><a href="#cb15-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-153"><a href="#cb15-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-154"><a href="#cb15-154" aria-hidden="true" tabindex="-1"></a>The p-value histogram after prefiltering of counts is not looking good. I havent done prefiltering. Supposedly, DESeq2 takes care of the low counts while identifying DEGs. Refer this link <span class="ot">&lt;http://varianceexplained.org/statistics/interpreting-pvalue-histogram/&gt;</span> for more info.</span>
<span id="cb15-155"><a href="#cb15-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-156"><a href="#cb15-156" aria-hidden="true" tabindex="-1"></a><span class="fu">### PCA plot</span></span>
<span id="cb15-157"><a href="#cb15-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-158"><a href="#cb15-158" aria-hidden="true" tabindex="-1"></a><span class="in">```{r PCA plot}</span></span>
<span id="cb15-159"><a href="#cb15-159" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-160"><a href="#cb15-160" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-161"><a href="#cb15-161" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "PCA with sample IDs"</span></span>
<span id="cb15-162"><a href="#cb15-162" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "PCA when colored by genotype and timepoint"</span></span>
<span id="cb15-163"><a href="#cb15-163" aria-hidden="true" tabindex="-1"></a>vstcounts <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> T)</span>
<span id="cb15-164"><a href="#cb15-164" aria-hidden="true" tabindex="-1"></a>normalized_cts <span class="ot">&lt;-</span> <span class="fu">assay</span>(vstcounts) <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb15-165"><a href="#cb15-165" aria-hidden="true" tabindex="-1"></a>pca_prcomp <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(normalized_cts), <span class="at">center =</span> T, <span class="at">scale =</span> F)</span>
<span id="cb15-166"><a href="#cb15-166" aria-hidden="true" tabindex="-1"></a><span class="co">#names(pca_prcomp)</span></span>
<span id="cb15-167"><a href="#cb15-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-168"><a href="#cb15-168" aria-hidden="true" tabindex="-1"></a>PC1_and_PC2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">1</span>],</span>
<span id="cb15-169"><a href="#cb15-169" aria-hidden="true" tabindex="-1"></a>                          <span class="at">PC2 =</span> pca_prcomp<span class="sc">$</span>x[,<span class="dv">2</span>],</span>
<span id="cb15-170"><a href="#cb15-170" aria-hidden="true" tabindex="-1"></a>                          <span class="at">type =</span> <span class="fu">rownames</span>(pca_prcomp<span class="sc">$</span>x))</span>
<span id="cb15-171"><a href="#cb15-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-172"><a href="#cb15-172" aria-hidden="true" tabindex="-1"></a><span class="co">#using the complete normalized matrix</span></span>
<span id="cb15-173"><a href="#cb15-173" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(PC1_and_PC2, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">col=</span>type)) <span class="sc">+</span></span>
<span id="cb15-174"><a href="#cb15-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb15-175"><a href="#cb15-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> type), <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb15-176"><a href="#cb15-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>()</span>
<span id="cb15-177"><a href="#cb15-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-178"><a href="#cb15-178" aria-hidden="true" tabindex="-1"></a><span class="co">#using variable features</span></span>
<span id="cb15-179"><a href="#cb15-179" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vstcounts, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>))</span>
<span id="cb15-180"><a href="#cb15-180" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-181"><a href="#cb15-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-182"><a href="#cb15-182" aria-hidden="true" tabindex="-1"></a><span class="fu">### Sample correlation heatmap using vst counts</span></span>
<span id="cb15-183"><a href="#cb15-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-184"><a href="#cb15-184" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Sample correlation heatmap using vst counts}</span></span>
<span id="cb15-185"><a href="#cb15-185" aria-hidden="true" tabindex="-1"></a>sampleCor <span class="ot">&lt;-</span> <span class="fu">cor</span>(normalized_cts)</span>
<span id="cb15-186"><a href="#cb15-186" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> <span class="fu">cor</span>(normalized_cts))</span>
<span id="cb15-187"><a href="#cb15-187" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDist)</span>
<span id="cb15-188"><a href="#cb15-188" aria-hidden="true" tabindex="-1"></a>blueColours <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Blues"</span>)</span>
<span id="cb15-189"><a href="#cb15-189" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(blueColours))(<span class="dv">255</span>)</span>
<span id="cb15-190"><a href="#cb15-190" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(sampleDistMatrix,</span>
<span id="cb15-191"><a href="#cb15-191" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clustering_distance_cols =</span> sampleDist, </span>
<span id="cb15-192"><a href="#cb15-192" aria-hidden="true" tabindex="-1"></a>                   <span class="at">color =</span> colors)</span>
<span id="cb15-193"><a href="#cb15-193" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-194"><a href="#cb15-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-195"><a href="#cb15-195" aria-hidden="true" tabindex="-1"></a><span class="fu">## Differential gene expression analysis</span></span>
<span id="cb15-196"><a href="#cb15-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-197"><a href="#cb15-197" aria-hidden="true" tabindex="-1"></a><span class="fu">### Tissue-specific DEGs (the main effect)</span></span>
<span id="cb15-198"><a href="#cb15-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-199"><a href="#cb15-199" aria-hidden="true" tabindex="-1"></a>Here, I am identifying tissue-specific (liver versus heart) DEGs. These DEGs would be liver- or heart-enriched. </span>
<span id="cb15-200"><a href="#cb15-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-201"><a href="#cb15-201" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Comparison of Wald versus LRT test DEGs [Main effects due to genotype]</span></span>
<span id="cb15-202"><a href="#cb15-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-203"><a href="#cb15-203" aria-hidden="true" tabindex="-1"></a>**DEGs can be defined genes passing the threshold: abs(logFC) &gt; 1 &amp; padj &lt; 0.01**</span>
<span id="cb15-204"><a href="#cb15-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-205"><a href="#cb15-205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Comparison of Wald test and LRT - genotype}</span></span>
<span id="cb15-206"><a href="#cb15-206" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-207"><a href="#cb15-207" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-208"><a href="#cb15-208" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with Wald test [Main effects due to genotype]"</span></span>
<span id="cb15-209"><a href="#cb15-209" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with paired design and with using LRT [Main effects due to genotype]"</span></span>
<span id="cb15-210"><a href="#cb15-210" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to genotype]"</span></span>
<span id="cb15-211"><a href="#cb15-211" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb15-212"><a href="#cb15-212" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb15-213"><a href="#cb15-213" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype) <span class="co">#not accounting for timepoint</span></span>
<span id="cb15-214"><a href="#cb15-214" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb15-215"><a href="#cb15-215" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb15-216"><a href="#cb15-216" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb15-217"><a href="#cb15-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-218"><a href="#cb15-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-219"><a href="#cb15-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb15-220"><a href="#cb15-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb15-221"><a href="#cb15-221" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb15-222"><a href="#cb15-222" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb15-223"><a href="#cb15-223" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-Wald"</span>,</span>
<span id="cb15-224"><a href="#cb15-224" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb15-225"><a href="#cb15-225" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb15-226"><a href="#cb15-226" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb15-227"><a href="#cb15-227" aria-hidden="true" tabindex="-1"></a>res1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb15-228"><a href="#cb15-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-229"><a href="#cb15-229" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb15-230"><a href="#cb15-230" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb15-231"><a href="#cb15-231" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb15-232"><a href="#cb15-232" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb15-233"><a href="#cb15-233" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> time_point)</span>
<span id="cb15-234"><a href="#cb15-234" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"KO"</span>, <span class="st">"WT"</span>))</span>
<span id="cb15-235"><a href="#cb15-235" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> res2 <span class="sc">%&gt;%</span></span>
<span id="cb15-236"><a href="#cb15-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-237"><a href="#cb15-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-238"><a href="#cb15-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb15-239"><a href="#cb15-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb15-240"><a href="#cb15-240" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb15-241"><a href="#cb15-241" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb15-242"><a href="#cb15-242" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"KO_vs_WT-LRT"</span>,</span>
<span id="cb15-243"><a href="#cb15-243" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"KO"</span>,</span>
<span id="cb15-244"><a href="#cb15-244" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"WT"</span>,</span>
<span id="cb15-245"><a href="#cb15-245" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb15-246"><a href="#cb15-246" aria-hidden="true" tabindex="-1"></a>res2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb15-247"><a href="#cb15-247" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res1,res2)</span>
<span id="cb15-248"><a href="#cb15-248" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb15-249"><a href="#cb15-249" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb15-250"><a href="#cb15-250" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb15-251"><a href="#cb15-251" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb15-252"><a href="#cb15-252" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb15-253"><a href="#cb15-253" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb15-254"><a href="#cb15-254" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span>
<span id="cb15-255"><a href="#cb15-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-256"><a href="#cb15-256" aria-hidden="true" tabindex="-1"></a>DEGs with and without accounting for timepoint yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to genotype. </span>
<span id="cb15-257"><a href="#cb15-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-258"><a href="#cb15-258" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Comparison of Wald versus LRT test DEGs [Main effects due to time_point]</span></span>
<span id="cb15-259"><a href="#cb15-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-260"><a href="#cb15-260" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Comparison of Wald test and LRT - Timepoint}</span></span>
<span id="cb15-261"><a href="#cb15-261" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-262"><a href="#cb15-262" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-263"><a href="#cb15-263" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with Wald test [Main effects due to time_point]"</span></span>
<span id="cb15-264"><a href="#cb15-264" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DEGs obtained with paired design and with using LRT [Main effects due to time_point]"</span></span>
<span id="cb15-265"><a href="#cb15-265" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to time_point]"</span></span>
<span id="cb15-266"><a href="#cb15-266" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb15-267"><a href="#cb15-267" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb15-268"><a href="#cb15-268" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> time_point) <span class="co">#not accounting for timepoint</span></span>
<span id="cb15-269"><a href="#cb15-269" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb15-270"><a href="#cb15-270" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb15-271"><a href="#cb15-271" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> res4 <span class="sc">%&gt;%</span></span>
<span id="cb15-272"><a href="#cb15-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-273"><a href="#cb15-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-274"><a href="#cb15-274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb15-275"><a href="#cb15-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb15-276"><a href="#cb15-276" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb15-277"><a href="#cb15-277" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb15-278"><a href="#cb15-278" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-Wald"</span>,</span>
<span id="cb15-279"><a href="#cb15-279" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb15-280"><a href="#cb15-280" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb15-281"><a href="#cb15-281" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>)) </span>
<span id="cb15-282"><a href="#cb15-282" aria-hidden="true" tabindex="-1"></a>res4 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb15-283"><a href="#cb15-283" aria-hidden="true" tabindex="-1"></a><span class="co">#LRT test</span></span>
<span id="cb15-284"><a href="#cb15-284" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts,</span>
<span id="cb15-285"><a href="#cb15-285" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> coldata,</span>
<span id="cb15-286"><a href="#cb15-286" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point) <span class="co">#accounting for timepoint in design</span></span>
<span id="cb15-287"><a href="#cb15-287" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">test =</span> <span class="st">"LRT"</span>, <span class="at">full =</span> <span class="sc">~</span> genotype <span class="sc">+</span> time_point, <span class="at">reduced =</span> <span class="sc">~</span> genotype)</span>
<span id="cb15-288"><a href="#cb15-288" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"time_point"</span>, <span class="st">"T12"</span>, <span class="st">"T0"</span>))</span>
<span id="cb15-289"><a href="#cb15-289" aria-hidden="true" tabindex="-1"></a>res5 <span class="ot">&lt;-</span> res5 <span class="sc">%&gt;%</span></span>
<span id="cb15-290"><a href="#cb15-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-291"><a href="#cb15-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-292"><a href="#cb15-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(log2FoldChange,padj) <span class="sc">%&gt;%</span></span>
<span id="cb15-293"><a href="#cb15-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DEG =</span> <span class="fu">case_when</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb15-294"><a href="#cb15-294" aria-hidden="true" tabindex="-1"></a>                         log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">"down"</span>,</span>
<span id="cb15-295"><a href="#cb15-295" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>),</span>
<span id="cb15-296"><a href="#cb15-296" aria-hidden="true" tabindex="-1"></a>         <span class="at">comparison =</span> <span class="st">"T12_vs_T0-LRT"</span>,</span>
<span id="cb15-297"><a href="#cb15-297" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="fu">case_when</span>(DEG <span class="sc">==</span> <span class="st">"up"</span> <span class="sc">~</span> <span class="st">"T12"</span>,</span>
<span id="cb15-298"><a href="#cb15-298" aria-hidden="true" tabindex="-1"></a>                                DEG <span class="sc">==</span> <span class="st">"down"</span> <span class="sc">~</span> <span class="st">"T0"</span>,</span>
<span id="cb15-299"><a href="#cb15-299" aria-hidden="true" tabindex="-1"></a>                                <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"non-sig"</span>))</span>
<span id="cb15-300"><a href="#cb15-300" aria-hidden="true" tabindex="-1"></a>res5 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">count</span>(comparison, enrichment) <span class="sc">%&gt;%</span> <span class="fu">gt</span>()</span>
<span id="cb15-301"><a href="#cb15-301" aria-hidden="true" tabindex="-1"></a>res3 <span class="ot">=</span> <span class="fu">rbind_common</span>(res4,res5)</span>
<span id="cb15-302"><a href="#cb15-302" aria-hidden="true" tabindex="-1"></a><span class="co">#split(res3, f = list(res3$comparison, res3$enrichment))</span></span>
<span id="cb15-303"><a href="#cb15-303" aria-hidden="true" tabindex="-1"></a><span class="co">#UpSetR::upset()</span></span>
<span id="cb15-304"><a href="#cb15-304" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVenn</span>(<span class="fu">map</span>(<span class="fu">split</span>(res3, <span class="at">f =</span> <span class="fu">list</span>(res3<span class="sc">$</span>comparison, res3<span class="sc">$</span>enrichment)), <span class="cf">function</span>(x) x <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)),</span>
<span id="cb15-305"><a href="#cb15-305" aria-hidden="true" tabindex="-1"></a>         <span class="at">use_venn =</span> F,</span>
<span id="cb15-306"><a href="#cb15-306" aria-hidden="true" tabindex="-1"></a>         <span class="at">main_text_size =</span> <span class="dv">15</span>,</span>
<span id="cb15-307"><a href="#cb15-307" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_text_size =</span> <span class="dv">8</span>,</span>
<span id="cb15-308"><a href="#cb15-308" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend_position =</span> <span class="st">"left"</span>)</span>
<span id="cb15-309"><a href="#cb15-309" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-310"><a href="#cb15-310" aria-hidden="true" tabindex="-1"></a>DEGs with and without accounting for genotype yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to timepoints.  </span>
<span id="cb15-311"><a href="#cb15-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-312"><a href="#cb15-312" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Visualization of tissue-specific DEGs using DESeq2</span></span>
<span id="cb15-313"><a href="#cb15-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-314"><a href="#cb15-314" aria-hidden="true" tabindex="-1"></a>I am going with res1 object that uses Wald test to identify main effects due to genotype. </span>
<span id="cb15-315"><a href="#cb15-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-316"><a href="#cb15-316" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Identification of tissue-specific DEGs}</span></span>
<span id="cb15-317"><a href="#cb15-317" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-318"><a href="#cb15-318" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-319"><a href="#cb15-319" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Number of identified DEGs (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb15-320"><a href="#cb15-320" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Tissue-specific DEGs agnostic of the sampling timepoint"</span></span>
<span id="cb15-321"><a href="#cb15-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-322"><a href="#cb15-322" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plot</span></span>
<span id="cb15-323"><a href="#cb15-323" aria-hidden="true" tabindex="-1"></a>EnhancedVolcano<span class="sc">::</span><span class="fu">EnhancedVolcano</span>(res1,</span>
<span id="cb15-324"><a href="#cb15-324" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb15-325"><a href="#cb15-325" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb15-326"><a href="#cb15-326" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">lab =</span> res1<span class="sc">$</span>gene,</span>
<span id="cb15-327"><a href="#cb15-327" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pCutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb15-328"><a href="#cb15-328" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">FCcutoff =</span> <span class="dv">1</span>,</span>
<span id="cb15-329"><a href="#cb15-329" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">"Liver vs Heart comparison [Paired-design]"</span>,</span>
<span id="cb15-330"><a href="#cb15-330" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">subtitle =</span> <span class="st">"Differential expression analysis using DESeq2"</span>,</span>
<span id="cb15-331"><a href="#cb15-331" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">labSize =</span> <span class="fl">2.0</span></span>
<span id="cb15-332"><a href="#cb15-332" aria-hidden="true" tabindex="-1"></a>                                 )</span>
<span id="cb15-333"><a href="#cb15-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-334"><a href="#cb15-334" aria-hidden="true" tabindex="-1"></a><span class="co">#test this code</span></span>
<span id="cb15-335"><a href="#cb15-335" aria-hidden="true" tabindex="-1"></a><span class="fu">as_tibble</span>(<span class="fu">counts</span>(dds[res1<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], ], <span class="at">normalize =</span> <span class="cn">TRUE</span>),</span>
<span id="cb15-336"><a href="#cb15-336" aria-hidden="true" tabindex="-1"></a>          <span class="at">rownames =</span> <span class="st">'gene'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-337"><a href="#cb15-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"sample"</span>, <span class="at">values_to =</span> <span class="st">"counts"</span>, <span class="sc">-</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb15-338"><a href="#cb15-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">as_tibble</span>(<span class="fu">colData</span>(dds), <span class="at">rownames =</span> <span class="st">"group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-339"><a href="#cb15-339" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %&gt;%</span></span>
<span id="cb15-340"><a href="#cb15-340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> counts, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb15-341"><a href="#cb15-341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">'identity'</span>, <span class="at">color =</span> <span class="st">"gray30"</span>) <span class="sc">+</span></span>
<span id="cb15-342"><a href="#cb15-342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb15-343"><a href="#cb15-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb15-344"><a href="#cb15-344" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-345"><a href="#cb15-345" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb15-346"><a href="#cb15-346" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb15-347"><a href="#cb15-347" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))</span>
<span id="cb15-348"><a href="#cb15-348" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-349"><a href="#cb15-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-350"><a href="#cb15-350" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Expression pattern of top tissue-specific DEGs</span></span>
<span id="cb15-351"><a href="#cb15-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-352"><a href="#cb15-352" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Expression pattern of top tissue-specific DEGs}</span></span>
<span id="cb15-353"><a href="#cb15-353" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb15-354"><a href="#cb15-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-355"><a href="#cb15-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&gt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-356"><a href="#cb15-356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-357"><a href="#cb15-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb15-358"><a href="#cb15-358" aria-hidden="true" tabindex="-1"></a>sig_mat <span class="ot">&lt;-</span> normalized_cts[significant_genes,]</span>
<span id="cb15-359"><a href="#cb15-359" aria-hidden="true" tabindex="-1"></a>col_anno <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> coldata <span class="sc">%&gt;%</span> <span class="fu">select</span>(genotype, time_point),</span>
<span id="cb15-360"><a href="#cb15-360" aria-hidden="true" tabindex="-1"></a>                             <span class="at">col =</span> <span class="fu">list</span>(<span class="at">tissue =</span> <span class="fu">c</span>(<span class="st">"TRUE"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"FALSE"</span>, <span class="st">"blue"</span>)))</span>
<span id="cb15-361"><a href="#cb15-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-362"><a href="#cb15-362" aria-hidden="true" tabindex="-1"></a><span class="co">#| layout-ncol: 1</span></span>
<span id="cb15-363"><a href="#cb15-363" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: </span></span>
<span id="cb15-364"><a href="#cb15-364" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| &gt; 1 &amp; padj &lt; 0.01)"</span></span>
<span id="cb15-365"><a href="#cb15-365" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "Boxplot of top 12 DEGs"</span></span>
<span id="cb15-366"><a href="#cb15-366" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sig_mat))),</span>
<span id="cb15-367"><a href="#cb15-367" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_annotation =</span> col_anno,</span>
<span id="cb15-368"><a href="#cb15-368" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> T,</span>
<span id="cb15-369"><a href="#cb15-369" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Scaled normalized</span><span class="sc">\n</span><span class="st">expression"</span></span>
<span id="cb15-370"><a href="#cb15-370" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb15-371"><a href="#cb15-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-372"><a href="#cb15-372" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> res1 <span class="sc">%&gt;%</span></span>
<span id="cb15-373"><a href="#cb15-373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-374"><a href="#cb15-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&gt;</span> <span class="fl">0.01</span>, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-375"><a href="#cb15-375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="fu">abs</span>(log2FoldChange), <span class="at">n =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-376"><a href="#cb15-376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(gene)</span>
<span id="cb15-377"><a href="#cb15-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-378"><a href="#cb15-378" aria-hidden="true" tabindex="-1"></a>boxplot_1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(significant_genes, <span class="cf">function</span>(x) {</span>
<span id="cb15-379"><a href="#cb15-379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> x,</span>
<span id="cb15-380"><a href="#cb15-380" aria-hidden="true" tabindex="-1"></a>             <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time_point"</span>),</span>
<span id="cb15-381"><a href="#cb15-381" aria-hidden="true" tabindex="-1"></a>             <span class="at">returnData =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb15-382"><a href="#cb15-382" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">paste</span>(genotype,time_point), <span class="at">y =</span> <span class="fu">log2</span>(count), <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>))) <span class="sc">+</span></span>
<span id="cb15-383"><a href="#cb15-383" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb15-384"><a href="#cb15-384" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Experimental group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Normalized counts"</span>) <span class="sc">+</span></span>
<span id="cb15-385"><a href="#cb15-385" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(x))</span>
<span id="cb15-386"><a href="#cb15-386" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb15-387"><a href="#cb15-387" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(boxplot_1, <span class="at">ncol =</span> <span class="dv">3</span>)                             </span>
<span id="cb15-388"><a href="#cb15-388" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>