---
title: "SampleAnalysis_Mansingh2024_matrix"
format:
  html:
    code-fold: true
    code-tools: true
    toc: true
    eval: true
    warning: false
    message: false
    fig-height: 7
    fig-width: 7
    fig-format: png
    fig-dpi: 300
editor: visual
editor_options: 
  chunk_output_type: console
---

**Aim:** To perform differential gene expression analysis for different contrasts

## Loading the required libraries and importing required files

### Loading the required libraries

```{r Loading libraries}
library(tidyverse) #for data wrangling
library(DESeq2) #for DEG analysis
library(genekitr) #for pathway analysis
library(readr) #for data import
library(eatTools) #for rbind functions
library(strex) #for string related functions
library(knitr) #for beautiful tables
library(gt) #for customized table layouts
library(ComplexHeatmap) #for beautiful heatmaps
library(tidyheatmaps) #for heatmap generation from dataframe
library(patchwork) #for adding multiple figures to a plot
library(RColorBrewer) #for color schemas
library(ggplot2) #for plotting
library(ExploreModelMatrix) #to explore complicated design in the DESeq2 formula
library(UpSetR)
```

### Importing the counts matrix

```{r Importing the counts and conversion to a DESeq object}
#loading the length-scaled counts tsv file
cts <- read.csv("~/Documents/BiostateAI/data/Mansingh2024_expression_matrix.txt",sep="\t", header=T) 

#converting NA into 0 and round off the counts
cts[is.na(cts)] = 0
cts <- cts %>% mutate_if(is.numeric, list(round))
```

### Loading the metadata file

```{r Import metadata file}
genotype = as.factor( str_detect( colnames(cts) , 'CTRL' )  )
genotype = plyr::revalue( genotype , c( 'TRUE'='WT' , 'FALSE' = 'KO'  )  ) #TRUE is WT, FALSE is CTRL

time_point = factor( substring(colnames(cts), regexpr("_Z",colnames(cts)) + 2) , levels = c("T0",'T4','T8','T12','T16','T20'))

mouse = sapply( strsplit( colnames( cts ) , "_") , '[' , 5)

replicate = substring(colnames(cts) , 1, nchar( "HL3YYBGX5" ) ) 

coldata <- data.frame( sample = colnames(cts) ,
                       genotype = genotype ,
                       time_point = time_point,
                       mouse = mouse,
                       replicate = replicate
                       )

coldata <- coldata %>% filter(time_point %in% c("T0", "T12"))
coldata$time_point <- factor(coldata$time_point, levels = c("T0", "T12"))
  
cts <- cts %>% select(coldata$sample)
#| layout-ncol: 2
#| fig-cap: 
#|   - "Metadata file"
#|   - "Sample counts for genotype and timepoint combinations"
coldata %>% gt()

coldata %>% dplyr::count(genotype,time_point) %>% gt()
```

## Creation and exploration of dds object

```{r Conversion to a DESeq object}
#| layout-ncol: 1
#| fig-cap: 
#|   - "p-value histogram without prefiltering of counts"
#|   - "Genes with high p-values do seem to have lower gene expression"

#creation of dds object without prefiltering
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ genotype) #paired design tissue + time if need to adjust for confounders (time in this case)

vd <- VisualizeDesign(sampleData = coldata,
                      designFormula = ~ genotype)
#print(vd$plotlist)

dds$genotype <- relevel(dds$genotype, ref = "WT")

dds <- DESeq(dds) #Main effects for genotype #Expression affected by genotype not correcting for  the timepoint effects

res <- results(dds, contrast = c("genotype", "KO", "WT")) #FALSE is KO, WT is TRUE and is in the denominator

res %>% as.data.frame() %>%
  arrange(padj) %>%
  ggplot(aes(x = pvalue)) +
  geom_histogram(color = "white", bins = 50) +
  ggtitle("p-value histogram without prefiltering of counts")

res %>% as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(pvalue)) %>%
  mutate(pvalue_bin = if_else(pvalue > 0.75, "high", "low")) %>%
  ggplot(aes(x = pvalue_bin, y = log2(baseMean))) + 
  geom_boxplot()
```

p-value histogram distribution looks bimodal. Therefore, the counts needs prefiltering before DEG analysis. High expressing count genes fall into low p-value bin which is great. 

```{r Prefiltering of counts using the dds object}
#dds <- dds[rowSums(counts(dds) > 10,)] #opted out
#dds <- DESeq(dds)
#resultsNames(dds)

#res <- results(dds, contrast = c("genotype", "FALSE", "TRUE")) #FALSE is KO, WT is TRUE and is in the denominator

#| layout-ncol: 1
#| fig-cap: 
#|   - "p-value histogram after prefiltering of counts"
#|   - "Genes with high p-values do seem to have lower gene expression"

res %>% as.data.frame() %>%
  arrange(padj) %>%
  ggplot(aes(x = pvalue)) +
  geom_histogram(color = "white", bins = 50) +
  ggtitle("p-value histogram without prefiltering of counts")

res %>% as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(pvalue)) %>%
  mutate(pvalue_bin = if_else(pvalue > 0.75, "high", "low")) %>%
  ggplot(aes(x = pvalue_bin, y = log2(baseMean))) + 
  geom_boxplot()
```
Various contrasts can be identified using **resultNames(dds)** function


The p-value histogram after prefiltering of counts is not looking good. I havent done prefiltering. Supposedly, DESeq2 takes care of the low counts while identifying DEGs. Refer this link <http://varianceexplained.org/statistics/interpreting-pvalue-histogram/> for more info.

### PCA plot

```{r PCA plot}
#| layout-ncol: 1
#| fig-cap: 
#|   - "PCA with sample IDs"
#|   - "PCA when colored by genotype and timepoint"
vstcounts <- vst(dds, blind = T)
normalized_cts <- assay(vstcounts) %>% as.matrix()
pca_prcomp <- prcomp(t(normalized_cts), center = T, scale = F)
#names(pca_prcomp)

PC1_and_PC2 <- data.frame(PC1 = pca_prcomp$x[,1],
                          PC2 = pca_prcomp$x[,2],
                          type = rownames(pca_prcomp$x))

#using the complete normalized matrix
ggplot(PC1_and_PC2, aes(x = PC1, y = PC2, col=type)) +
  geom_point() +
  geom_text(aes(label = type), hjust = 0, vjust = 0) +
  coord_fixed()

#using variable features
plotPCA(vstcounts, intgroup = c("genotype","time_point"))
```

### Sample correlation heatmap using vst counts

```{r Sample correlation heatmap using vst counts}
sampleCor <- cor(normalized_cts)
sampleDist <- as.dist(1 - cor(normalized_cts))
sampleDistMatrix <- as.matrix(sampleDist)
blueColours <- brewer.pal(9, "Blues")
colors <- colorRampPalette(rev(blueColours))(255)
pheatmap::pheatmap(sampleDistMatrix,
                   clustering_distance_cols = sampleDist, 
                   color = colors)
```

## Differential gene expression analysis

### Tissue-specific DEGs (the main effect)

Here, I am identifying tissue-specific (liver versus heart) DEGs. These DEGs would be liver- or heart-enriched. 

#### Comparison of Wald versus LRT test DEGs [Main effects due to genotype]

**DEGs can be defined genes passing the threshold: abs(logFC) > 1 & padj < 0.01**

```{r Comparison of Wald test and LRT - genotype}
#| layout-ncol: 1
#| fig-cap: 
#|   - "DEGs obtained with Wald test [Main effects due to genotype]"
#|   - "DEGs obtained with paired design and with using LRT [Main effects due to genotype]"
#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to genotype]"
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ genotype) #not accounting for timepoint
dds <- DESeq(dds)
res1 <- results(dds, contrast = c("genotype", "KO", "WT"))
res1 <- res1 %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  drop_na(log2FoldChange,padj) %>%
  mutate(DEG = case_when(log2FoldChange > 1 & padj < 0.01 ~ "up",
                         log2FoldChange < -1 & padj < 0.01 ~ "down",
                         TRUE ~ "non-sig"),
         comparison = "KO_vs_WT-Wald",
         enrichment = case_when(DEG == "up" ~ "KO",
                                DEG == "down" ~ "WT",
                                TRUE ~ "non-sig")) 
res1 %>% dplyr::count(comparison, enrichment) %>% gt()

#LRT test
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ genotype + time_point) #accounting for timepoint in design
dds <- DESeq(dds, test = "LRT", full = ~ genotype + time_point, reduced = ~ time_point)
res2 <- results(dds, contrast = c("genotype", "KO", "WT"))
res2 <- res2 %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  drop_na(log2FoldChange,padj) %>%
  mutate(DEG = case_when(log2FoldChange > 1 & padj < 0.01 ~ "up",
                         log2FoldChange < -1 & padj < 0.01 ~ "down",
                         TRUE ~ "non-sig"),
         comparison = "KO_vs_WT-LRT",
         enrichment = case_when(DEG == "up" ~ "KO",
                                DEG == "down" ~ "WT",
                                TRUE ~ "non-sig")) 
res2 %>% dplyr::count(comparison, enrichment) %>% gt()
res3 = rbind_common(res1,res2)
#split(res3, f = list(res3$comparison, res3$enrichment))
#UpSetR::upset()
plotVenn(map(split(res3, f = list(res3$comparison, res3$enrichment)), function(x) x %>% pull(gene)),
         use_venn = F,
         main_text_size = 15,
         legend_text_size = 8,
         legend_position = "left")
```
DEGs with and without accounting for timepoint yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to genotype. 

#### Comparison of Wald versus LRT test DEGs [Main effects due to time_point]

```{r Comparison of Wald test and LRT - Timepoint}
#| layout-ncol: 1
#| fig-cap: 
#|   - "DEGs obtained with Wald test [Main effects due to time_point]"
#|   - "DEGs obtained with paired design and with using LRT [Main effects due to time_point]"
#|   - "Upset plot showing the common DEGs between comparisons and enrichment directionality [Main effects due to time_point]"
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ time_point) #not accounting for timepoint
dds <- DESeq(dds)
res4 <- results(dds, contrast = c("time_point", "T12", "T0"))
res4 <- res4 %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  drop_na(log2FoldChange,padj) %>%
  mutate(DEG = case_when(log2FoldChange > 1 & padj < 0.01 ~ "up",
                         log2FoldChange < -1 & padj < 0.01 ~ "down",
                         TRUE ~ "non-sig"),
         comparison = "T12_vs_T0-Wald",
         enrichment = case_when(DEG == "up" ~ "T12",
                                DEG == "down" ~ "T0",
                                TRUE ~ "non-sig")) 
res4 %>% dplyr::count(comparison, enrichment) %>% gt()
#LRT test
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ genotype + time_point) #accounting for timepoint in design
dds <- DESeq(dds, test = "LRT", full = ~ genotype + time_point, reduced = ~ genotype)
res5 <- results(dds, contrast = c("time_point", "T12", "T0"))
res5 <- res5 %>%
  as.data.frame() %>%
  rownames_to_column("gene") %>%
  drop_na(log2FoldChange,padj) %>%
  mutate(DEG = case_when(log2FoldChange > 1 & padj < 0.01 ~ "up",
                         log2FoldChange < -1 & padj < 0.01 ~ "down",
                         TRUE ~ "non-sig"),
         comparison = "T12_vs_T0-LRT",
         enrichment = case_when(DEG == "up" ~ "T12",
                                DEG == "down" ~ "T0",
                                TRUE ~ "non-sig"))
res5 %>% dplyr::count(comparison, enrichment) %>% gt()
res3 = rbind_common(res4,res5)
#split(res3, f = list(res3$comparison, res3$enrichment))
#UpSetR::upset()
plotVenn(map(split(res3, f = list(res3$comparison, res3$enrichment)), function(x) x %>% pull(gene)),
         use_venn = F,
         main_text_size = 15,
         legend_text_size = 8,
         legend_position = "left")
```
DEGs with and without accounting for genotype yields similar DEGs between Wald and LRT tests suggesting that both designs are okay to detect main effects due to timepoints.  

#### Visualization of tissue-specific DEGs using DESeq2

I am going with res1 object that uses Wald test to identify main effects due to genotype. 

```{r Identification of tissue-specific DEGs}
#| layout-ncol: 1
#| fig-cap: 
#|   - "Number of identified DEGs (|logFC| > 1 & padj < 0.01)"
#|   - "Tissue-specific DEGs agnostic of the sampling timepoint"

#Volcano plot
EnhancedVolcano::EnhancedVolcano(res1,
                                 x = "log2FoldChange",
                                 y = "padj",
                                 lab = res1$gene,
                                 pCutoff = 0.01,
                                 FCcutoff = 1,
                                 title = "Liver vs Heart comparison [Paired-design]",
                                 subtitle = "Differential expression analysis using DESeq2",
                                 labSize = 2.0
                                 )

#test this code
as_tibble(counts(dds[res1$gene[1:2], ], normalize = TRUE),
          rownames = 'gene') %>%
  pivot_longer(names_to = "sample", values_to = "counts", -gene) %>%
  left_join(as_tibble(colData(dds), rownames = "group")) %>%
  #mutate(name = paste0(substr(Type, 1, 5), '_', Condition, '_', 1:3)) %>%
  ggplot(aes(x = group, y = counts, fill = genotype)) +
  geom_bar(stat = 'identity', color = "gray30") +
  facet_wrap( ~ gene, scales = "free") +
  theme(axis.text.x = element_text(size = 8, angle = 90),
        axis.title.x = element_blank(),
        legend.position = "right",
        legend.text = element_text(size = 7),
        legend.title = element_text(size = 7))
```

#### Expression pattern of top tissue-specific DEGs

```{r Expression pattern of top tissue-specific DEGs}
significant_genes <- res1 %>%
  as.data.frame() %>%
  filter(padj > 0.01, abs(log2FoldChange) > 1) %>%
  slice_max(abs(log2FoldChange), n = 50) %>%
  pull(gene)
sig_mat <- normalized_cts[significant_genes,]
col_anno = HeatmapAnnotation(df = coldata %>% select(genotype, time_point),
                             col = list(tissue = c("TRUE" = "red", "FALSE", "blue")))

#| layout-ncol: 1
#| fig-cap: 
#|   - "Heatmap of top 50 DEGs sorted by |logFC| (|logFC| > 1 & padj < 0.01)"
#|   - "Boxplot of top 12 DEGs"
Heatmap(t(scale(t(sig_mat))),
        top_annotation = col_anno,
        show_row_names = T,
        name = "Scaled normalized\nexpression"
        )

significant_genes <- res1 %>%
  as.data.frame() %>%
  filter(padj > 0.01, abs(log2FoldChange) > 1) %>%
  slice_max(abs(log2FoldChange), n = 12) %>%
  pull(gene)

boxplot_1 <- lapply(significant_genes, function(x) {
  plotCounts(dds, gene = x,
             intgroup = c("genotype","time_point"),
             returnData = T) %>%
    ggplot(aes(x = paste(genotype,time_point), y = log2(count), fill = c("genotype"))) +
    geom_boxplot() + geom_jitter(width = 0.1) + 
    xlab("Experimental group") + ylab("Normalized counts") +
    ggtitle(paste0(x))
    })
wrap_plots(boxplot_1, ncol = 3)                             
```